---
title: "GlobalPhosphoIsoformLevel"
author: "Alexis Chang"
date: "`r Sys.Date()`"
output: console
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


#libraries
```{r}
library(tidyverse)
library(janitor)
library(readr)
library(gridExtra)
library(ggrepel)
library(cowplot)
library(reshape2)
library(eulerr)
library(protti)

library(iq)
library(RColorBrewer)
#for PCA loadings
library(FactoMineR)

#for protti sample correlation and volcano plots
library(dendextend)
library(pheatmap)
library(seriation)
library(UpSetR)

library(gprofiler2)
```

##_b) functions
#alexis_theme()
```{r}
alexis_theme <- function() {
  theme(
    # panel.border = element_rect(colour = "blue", fill = NA, linetype = 2),
    panel.border = element_blank(),
    panel.background = element_blank(),
    panel.grid.major.x  = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(angle = 90, vjust = 0.25, hjust = 1),
    axis.line = element_line(colour = "black"),
    axis.text = element_text(colour = "black", face = "plain", family = "sans"),
    axis.title = element_text(colour = "black", family = "sans"),
    axis.ticks = element_line(colour = "black"),
    title = element_text(size = 8, hjust = 0.5),
    # legend at the bottom 6)
    legend.position = "right")   
}
```

#FASTAs
```{r}
human_fasta_2024 <- read_csv("raw_data/MainFig3to6_SuppFig9to15/sp_reviewed_UP000005640_2024_03_21_forR.csv")


#just to convert PTMSEA gene names to uniprot ID references
fasta_gene_names <- read_tsv(file = "raw_data/MainFig3to6_SuppFig9to15/uniprotkb_proteome_UP000005640_2024_09_04.tsv") %>%
  clean_names() %>% 
  #many gene names per uniprot reference, hope first gene name matches,
  #otherwise figure out if secondary matches improve join.
  separate(gene_names, into = c("gene", "gene2", "gene3", "gene4", "gene5", "gene6"), remove = FALSE)
```


#pathways
##PTMSEA
```{r}
EGFR1_pathway <- read_csv("raw_data/MainFig3to6_SuppFig9to15/EGFR1_phosphopathway_PTMSEA.csv") %>% clean_names() %>%
  left_join(y = fasta_gene_names, by = "gene", relationship = "many-to-many") %>%  #match on gene1 worked!
  select(-c(gene_names : gene6)) %>% 
  rename(reference = entry) %>% 
  mutate(
    ref = paste(reference, psite, sep = "_"),
    PTM_SEA = TRUE,
    mod_res = str_sub(psite, start = 1L, end = 1L))

```
##Phosphopedia
```{r}
ppdia_ErbB_signaling <- read_csv(file = "raw_data/MainFig3to6_SuppFig9to15/Phosphopedia_ErbB_gene_sites.csv") %>% 
  clean_names()
```



#raw data import

##comet
```{r}

comet <- read_csv("raw_data/MainFig3to6_SuppFig9to15/GlobalPhospho/AC080_all_comet/result.csv") %>%
  clean_names() %>% 
  separate(col = sample_name, into = c("experiment", "enrichment", "condition", "replicate", "R2P2_well", "MS_method", "inj_vol"), sep = "_", remove = FALSE) %>%
  filter(grepl("tc", condition) == FALSE) %>% 
  mutate(sample_id = paste(condition, replicate, sep = " ")) %>% 
  left_join(y = human_fasta_2024, by = "reference") %>% 
  rename(intensity = max_intensity_light_c2837) %>% 
  filter(reverse == FALSE) %>% 
  mutate(phospho = case_when(
    grepl("\\@", sequence) ~ "phospho",
    TRUE ~ "none"))
```

##ascore
```{r}
ascore <- read_csv("raw_data/MainFig3to6_SuppFig9to15/GlobalPhospho/AC080_all_ascore/result.csv") %>%
  clean_names() %>% 
  separate(col = sample_name, into = c("experiment", "enrichment", "condition", "replicate", "R2P2_well", "MS_method", "inj_vol"), sep = "_", remove = FALSE) %>% 
  filter(grepl("tc", condition) == FALSE) %>%
  mutate(sample_id = paste(condition, replicate, sep = " ")) %>% 
  left_join(y = human_fasta_2024, by = "reference") %>% 
  rename(intensity = max_intensity_light_c2837) %>% 
  filter(reverse == FALSE) %>% 
  mutate(phospho = case_when(
    grepl("\\@", sequence) ~ "phospho",
    TRUE ~ "none"))
```

####--parse phosphosite localization


Working from 'Ascore Extended' output from Zucchini. Avoids relying on "mod_location" for phospho localization.

```{r}
ascore.site2 <- ascore %>% 
  distinct() %>%
  
  mutate(seq_no_mods = str_remove_all(ascore_sequence, "[:punct:]"),
         seq_no_mods = str_remove_all(seq_no_mods, "[n]")) %>% ##remove n from n## mark of nAc modification. 
  mutate_at(c("a_score_1", "a_score_2", "a_score_3", "position_1", "position_2", "position_3"), funs(as.numeric)) %>% 
  mutate_at(c("a_score_1", "a_score_2", "a_score_3", "position_1", "position_2", "position_3"), ~replace_na(., 0)) %>%

  select(sample_name, condition, replicate, sample_id,  reference, gene,
         ascore_sequence, num_sites, a_score_1, a_score_2, a_score_3, position_1, position_2, position_3,
         seq_no_mods, intensity, q_score_c2837, num_scans_light_c2837, charge, x_corr,  msn_scan, retention_time, max_int_rt_light_c2837, #use for ascore cross check - pisoforms should align by MS1 feature.
         missed_cleavages, redundancy, proteins, full_protein_sequence, organism, protein_names, length) %>%
  
  
  #fix incorrectly localized MAPK1 pT site back to pY based on manual TQ inspection and RT:
  mutate(
    position_1 = case_when(
      sample_name == "AC080_gp_EGF3min_rep1_well4_90mDDA_6uL" & ascore_sequence == "VADPDHDHTGFLT@EYVATR" ~ 14,
    TRUE ~ position_1),

    ascore_sequence = case_when(
      sample_name == "AC080_gp_EGF3min_rep1_well4_90mDDA_6uL" & ascore_sequence == "VADPDHDHTGFLT@EYVATR" ~ "VADPDHDHTGFLTEY@VATR",
    TRUE ~ ascore_sequence)) %>%
  
  #code to identify if fix worked during troubleshooting:
  # filter(sample_name == "AC080_gp_EGF3min_rep1_well4_90mDDA_6uL") %>% 
  # filter(reference == "P28482")
  #resume original code
  
 
##extract modified residue
  mutate(
    mod_res1 = str_sub(seq_no_mods, start = position_1+1L, end = position_1 + 1L),
    mod_res2 = str_sub(seq_no_mods, start = position_2+1L, end = position_2 + 1L),
    mod_res3 = str_sub(seq_no_mods, start = position_3+1L, end = position_3 + 1L))


## pivot wider to match each p-site to an ascore and filter for STY
ascore.site2_longer <- ascore.site2 %>% 
  unite(ascore_mod_res1, c(a_score_1, mod_res1, position_1), sep = "_") %>% 
  unite(ascore_mod_res2, c(a_score_2, mod_res2, position_2), sep = "_") %>%
  unite(ascore_mod_res3, c(a_score_3, mod_res3, position_3), sep = "_") %>%
  pivot_longer(cols = c("ascore_mod_res1", "ascore_mod_res2", "ascore_mod_res3")) %>% 
  separate(value, into = c("ascore", "mod_res", "mod_position"), sep = "_") %>% 
  mutate(ascore= as.numeric(ascore)) 
  

```



####--filter for confidently localized phosphoisoforms

```{r}
ascore.site2_longer_stringent <- ascore.site2_longer %>% 
  filter(ascore >= 13) %>% 
  filter(grepl("[STY]", mod_res) == TRUE)
```

####--keep detected proteins
already joined to fasta above
```{r}
ascore_fasta <- ascore.site2_longer_stringent %>% 
  filter(!is.na(ascore_sequence))
```



####--get protein level site number

This code chunk includes the following update (from Sept 2023)

I made the z-substitute script to solve bug Sophie found.

The goal is to not have NA mods. Sophie found on Sept 20, 2023 that those mods are result of incorrect Isoleucine, Leucine assignment in peptide sequence.

Sophie implemented a very nice test of S, T, Y, or NA mods to see if parsing is complete and captures peptides with isoleucine or leucine that may have been swapped during searching.

```{r}
ascore_fasta_protein_mod_loc <- ascore_fasta %>% 
  mutate(
    mod_position = as.numeric(mod_position), ##mod_position parsed from ascore output above
    z_peptide = str_replace_all(seq_no_mods, "[IL]", "Z"),
    z_protein = str_replace_all(full_protein_sequence, "[IL]", "Z"),
    z_pept_position_in_z_protein = str_locate(z_protein, pattern = z_peptide), ##get peptide position from z-substituted peptide and protein
    mod_position_in_protein = z_pept_position_in_z_protein[,"start"] + mod_position, ##extract mod position from true protein sequence
    test_mod_position = str_sub(full_protein_sequence, start = mod_position_in_protein, end = mod_position_in_protein)) %>% 
  select(test_mod_position, mod_res, mod_position, ascore_sequence, everything()) %>% 
  unite(col = "mod_protein_location", c(mod_res, mod_position_in_protein), sep = "", remove = FALSE) %>% 
  unite(col = "isoform_cluster", c(reference, mod_protein_location), sep = "_", remove = FALSE) 
  
  
  #join to BCA quant for correlation analysis with plate protein yield
  # left_join(y = df_BCA, by = c("condition", "replicate", "density_numeric", "density")) 

#--------------------------------------------------
    ##tally up the number of rows that have NA vs. STY in test_mod_position
mod_pos_group <- ascore_fasta_protein_mod_loc %>% 
  select(test_mod_position) %>% 
  group_by(test_mod_position) %>% 
   summarize(
    num_phospeptides = n())
mod_pos_group

    ##all mods are on S, T or Y as hoped
  
#--------------------------------------------------
```

###---most common redundancy assignment per isoform_cluster
starting with dataframe that hasn't yet collapsed different ascore_sequences to merely isoform sites. thus this maintains different peptide observatinos with potentially different protein assignments/redundancy assignments. We want to take the most common redundancy assignment across all conditions and replicates. If we just took the max or min redundancy, this could be skewed to cases when a one-off peptide has a ragged end and is assigned to less peptides (min redundancy) although we may still be seeing conglomerate of multiple isoforms - aka I don't think one or two peptides (or minority of peptides) should be enough for statistical assertion of protein group assignment if majority of peptides show a different/broader assignment. Alternatively, in case of max redundancy assignment, majority of peptides could have a missed cleavage and come from a more specific / lower redundancy assignment and thus I would think it is reasonable to say the data displayed in shiny app can be the more specific redundancy assignment. Although, I think a case could be made to show the broadest redundancy possible for a site or isoform to represent all possibilities of the least specific peptide contributor.




##rollup PSM to precursor, then sum precursor intensities to p-peptide (aka isoform) and median normalize
may want to go to peptide isoform as well...!
```{r}
#keep single PSM per precursor:

ascore_fasta_protein_mod_loc_sum_isoform <- ascore_fasta_protein_mod_loc %>% 
  group_by(condition, replicate, sample_id, ascore_sequence, charge, ascore) %>% 
  filter(intensity == max(intensity)) %>% 
  ungroup() %>% 
  distinct(condition, replicate, sample_id, reference, ascore_sequence, mod_protein_location, intensity) %>% 


  ##SUM PRECURSOR INTENSITIES TO P-isoforms clusters + median normalize

 
  group_by(condition, replicate,  sample_id, reference, ascore_sequence) %>% #ascore_sequence = phospho peptide sequence = isoform
  mutate(
    isoform_in_peptide_number = paste("site" ,seq(1, n(), by = 1), sep = "" )) %>%
  ungroup %>% 
  
  pivot_wider(id_cols = c("condition", "replicate", "sample_id", "reference", "ascore_sequence", "intensity"),
              values_from = mod_protein_location,
              names_from = isoform_in_peptide_number) %>% 
  mutate(
    isoform_cluster = paste(site1, site2, site3, sep = " "),
    isoform_cluster = str_squish(str_replace_all(str_replace_na(isoform_cluster), "NA", ""))) %>% #nested turn logical to chr, then replace "NA" with empty character.
  group_by(condition, replicate,  sample_id, reference, isoform_cluster) %>%
  #sum PSM intensities of confident p-isoforms to individual p-isoforms
  mutate(
    sum_intensity_precursor_to_peptide = sum(intensity),
    log2_peptide_qty = log2(sum_intensity_precursor_to_peptide)) %>% 
  ungroup() %>% 
  
  #keep single summed intensity per p-site prior to median normalization
  distinct(condition, replicate, sample_id, reference, isoform_cluster, sum_intensity_precursor_to_peptide, log2_peptide_qty) %>% 
  
  
  #median normalize intensities summed to p-site
  mutate(
    global_median_intensity = median(log2_peptide_qty)) %>% 
  group_by(condition, replicate) %>% 
  mutate(
    sample_median_intensity = median(log2_peptide_qty)) %>% 
  ungroup() %>% 
  mutate(
    median_norm_intensity = log2_peptide_qty - sample_median_intensity + global_median_intensity,
    raw_median_norm_intensity = 2^median_norm_intensity) %>% 
  filter(!median_norm_intensity %in% c(Inf, -Inf)) %>% 
  filter(grepl("[STY]", isoform_cluster) ==TRUE) %>%  #require an S, T, or Y in isoform cluster
  distinct(condition, replicate, sample_id, reference, isoform_cluster, sum_intensity_precursor_to_peptide, median_norm_intensity, raw_median_norm_intensity) %>% 
  
  #attempt removal of duplicates by max intensity
  group_by(condition, replicate, sample_id, reference, isoform_cluster) %>% 
  filter(median_norm_intensity == max(median_norm_intensity)) %>% 
  ungroup()

```




###_identify mislocalized p-sites by max intensity MS1 feature_
Note! I can also write a function that assess spread of retention times between different phosphoisoforms to identify candidates of likely phospho mislocalization.

in progress code here to assess RT distribution 09/19/2024
```{r}

#keep single PSM per precursor:

RT_isoform_input <- ascore_fasta_protein_mod_loc %>% 
  group_by(condition, replicate, sample_id, ascore_sequence, charge, ascore) %>% 
  filter(intensity == max(intensity)) %>% 
  ungroup() %>% 
  distinct(condition, replicate, sample_id, reference, ascore_sequence, mod_protein_location, intensity, seq_no_mods, max_int_rt_light_c2837) %>% 


  ##SUM PRECURSOR INTENSITIES TO P-isoforms clusters + median normalize

 
  group_by(condition, replicate,  sample_id, reference, ascore_sequence) %>% #ascore_sequence = phospho peptide sequence = isoform
  mutate(
    isoform_in_peptide_number = paste("site" ,seq(1, n(), by = 1), sep = "" )) %>%
  ungroup %>% 
  
  pivot_wider(id_cols = c("condition", "replicate", "sample_id", "reference", "ascore_sequence", "intensity", "seq_no_mods", "max_int_rt_light_c2837"),
              values_from = mod_protein_location,
              names_from = isoform_in_peptide_number) %>% 
  mutate(
    isoform_cluster = paste(site1, site2, site3, sep = " "),
    isoform_cluster = str_squish(str_replace_all(str_replace_na(isoform_cluster), "NA", ""))) %>% #nested turn logical to chr, then replace "NA" with empty character.
  group_by(condition, replicate,  sample_id, reference, isoform_cluster) %>%
  #sum PSM intensities of confident p-isoforms to individual p-isoforms
  mutate(
    sum_intensity_precursor_to_peptide = sum(intensity),
    log2_peptide_qty = log2(sum_intensity_precursor_to_peptide)) %>% 
  ungroup() %>% 
  
  #keep single summed intensity per p-site prior to median normalization
  distinct(condition, replicate, sample_id, reference, ascore_sequence, isoform_cluster, sum_intensity_precursor_to_peptide, log2_peptide_qty, seq_no_mods, max_int_rt_light_c2837) %>% 
  
  
  #median normalize intensities summed to p-site
  mutate(
    global_median_intensity = median(log2_peptide_qty)) %>% 
  group_by(condition, replicate) %>% 
  mutate(
    sample_median_intensity = median(log2_peptide_qty)) %>% 
  ungroup() %>% 
  mutate(
    median_norm_intensity = log2_peptide_qty - sample_median_intensity + global_median_intensity,
    raw_median_norm_intensity = 2^median_norm_intensity) %>% 
  filter(!median_norm_intensity %in% c(Inf, -Inf)) %>% 
  filter(grepl("[STY]", isoform_cluster) ==TRUE) %>%  #require an S, T, or Y in isoform cluster
  distinct(condition, replicate, sample_id, reference, isoform_cluster, ascore_sequence, sum_intensity_precursor_to_peptide, median_norm_intensity, raw_median_norm_intensity, seq_no_mods, max_int_rt_light_c2837) %>% 
  
  #attempt removal of duplicates by max intensity
  group_by(condition, replicate, sample_id, reference, isoform_cluster) %>% 
  filter(median_norm_intensity == max(median_norm_intensity)) %>% 
  ungroup()

#keep single PSM per precursor:

testRT_ascore_fasta_protein_mod_loc_sum_isoform <- RT_isoform_input %>%
  mutate(seq_no_phospho = str_remove_all(ascore_sequence, "@")) %>% 
  group_by(condition, replicate, seq_no_phospho, isoform_cluster) %>% 
  mutate(max_RT = max(max_int_rt_light_c2837),
         min_RT = min(max_int_rt_light_c2837),
         range_RT = max_RT - min_RT)
  
plot_range_RT_isoform_cluster <-  ggplot(data = testRT_ascore_fasta_protein_mod_loc_sum_isoform) +
  geom_histogram(mapping = aes( x= range_RT)) +
  facet_wrap(facets = vars(condition))

plot_range_RT_isoform_cluster
```
 Majority of isoforms show 0 to small retention time differences, but some do show larger retention time differences, which could mean they are different isoforms or are peptide carryover.
 
 
 
```{r}
psite_maxRT_df <- ascore_fasta_protein_mod_loc %>% 
  ungroup() %>% 
  group_by(condition, replicate) %>% 
  distinct(ascore_sequence,  max_int_rt_light_c2837) %>%
  ungroup() %>%
  distinct(condition, replicate, ascore_sequence, max_int_rt_light_c2837) %>% 
  group_by(condition, replicate, ascore_sequence) %>% 
  mutate(n_distinct_maxintRT = n_distinct(max_int_rt_light_c2837)) %>% 
  ungroup()
```









#ANALYZE COMET
#1. COMET QC
##_a) ENRICHMENT EFFICIENCY
###__i) counts
df
```{r}
enrichment_efficiency_count_df <- comet %>% 
  distinct( condition, replicate,sequence, phospho) %>% 
  group_by( condition, replicate, phospho) %>% 
  summarize(
    num_phospho = n()) %>% 
  pivot_wider(names_from = phospho, values_from = num_phospho) %>% 
  mutate(
    total_pept = none + phospho,
    individual_phosphopept_enrich_efficiency = phospho / total_pept) %>% 
  ungroup() %>% 
  group_by( condition) %>% 
  mutate(
    avg_phosphopeptide_enrichment_efficiency = mean(individual_phosphopept_enrich_efficiency)) %>% 
  ungroup() %>% 
  mutate(
    sample_id = paste(condition, replicate, sep = "_") )


##view df of phosphopeptide enrichment efficiency ratios
enrichment_efficiency_count_df
```

plot total phosphopeptides per sample: 
```{r}
plot_ppept_detections <- ggplot() +
  geom_bar(data = (enrichment_efficiency_count_df %>%
                     mutate(condition = fct_relevel(condition, "EGF0min", "EGF1min", "EGF3min", "EGF5min", "EGF15min")) %>% 
                     ungroup() %>% 
                     group_by(condition) %>% 
                     mutate(
                       mean_phospho = mean(phospho)) %>% 
                     distinct(condition, mean_phospho)),
           
           mapping = aes(x = condition, y = mean_phospho),
           stat = "identity", alpha = 1, fill = "gray80") + 
  geom_jitter(data = enrichment_efficiency_count_df,
              mapping = aes(x = condition, y = phospho, fill = replicate), shape = 1,
              width = 0.25, show.legend = FALSE) +
  # geom_hline(yintercept = 12000, linewidth = 0.5, alpha = 0.5, linetype = 2) +
  # annotate(geom = "text",x = 7.5, y = 10000, label = ">12,000\np-peptides", size = 4, lineheight = 0.75) +
  theme_bw(14) +
  theme(legend.position = "none") +
  ylab("unqiue p-peptides") +
  # theme(axis.text.x = element_text(angle = 90, vjust = 0.25, hjust = 1)) +
  # scale_y_continuous(expand = c(0,0)) +
  expand_limits(y = c(0, 13000)) +
  alexis_theme()+
  # ylim(0, 15000) +
  scale_y_continuous(expand = c(0,0)) #removes whitespace below bars!
  ## ggtitle("Phosphopeptide enrichment efficiency")

plot_ppept_detections
ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/phospho_pept_counts_noAscore.png", plot = plot_ppept_detections, width = 10, height = 10, scale = 0.5)
ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/phospho_pept_counts_noAscore.pdf", plot = plot_ppept_detections, width = 10, height = 10, scale = 0.5)
```

plot
```{r}
plot_ppept_enrich_efficiency <- ggplot() +
  geom_bar(data = (enrichment_efficiency_count_df %>%
                     mutate(condition = fct_relevel(condition, "EGF0min", "EGF1min", "EGF3min", "EGF5min", "EGF15min")) %>%
                     distinct(condition, avg_phosphopeptide_enrichment_efficiency)),
           mapping = aes(x = condition, y = avg_phosphopeptide_enrichment_efficiency),
           stat = "identity", alpha = 1, fill = "gray80") + 
  geom_jitter(data = enrichment_efficiency_count_df,
              mapping = aes(x = condition, y = individual_phosphopept_enrich_efficiency, fill = replicate), shape = 1,
              width = 0.25, show.legend = FALSE) +
  geom_hline(yintercept = 0.96, linewidth = 1, alpha = 0.5, linetype = 2) +
  annotate(geom = "text",x = 3, y = 0.88, label = ">96%\npurity", size = 4, lineheight = 0.75) +
  theme_bw(14) +
  theme(legend.position = "none") +
  ylab("p-/all pept (count)") +
  # theme(axis.text.x = element_text(angle = 90, vjust = 0.25, hjust = 1)) +
  # scale_y_continuous(expand = c(0,0)) +
  expand_limits(y = c(0, 1)) +
  alexis_theme()+
  scale_y_continuous(expand = c(0,0))
  ## ggtitle("Phosphopeptide enrichment efficiency")

plot_ppept_enrich_efficiency
ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/phospho_pept_enrich_efficiency_counts.png", plot = plot_ppept_enrich_efficiency, width = 6, height = 8, scale = 0.5)
ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/phospho_pept_enrich_efficiency_counts.pdf", plot = plot_ppept_enrich_efficiency, width = 6, height = 9, scale = 0.5)
```


###__ii) intensity
df
```{r}
enrichment_efficiency_intensity_df <- comet %>% 
  distinct(sequence,  condition, replicate, phospho, intensity) %>% 
  group_by(sequence,  condition, replicate, phospho) %>% 
  
##filter by max intensity for every peptide
  filter(intensity  == max(intensity)) %>% 
  ungroup() %>% 
  group_by(phospho, condition, replicate) %>% 
  
##summarize intensity of all peptides and just p-peptides
  summarise(all_peptides_intensity = sum(intensity)) %>% 
  pivot_wider(values_from = all_peptides_intensity, names_from = phospho) %>% 
  
##grouped mutate for taking mean of summed intensities for each replicate within each condition  
  group_by(condition) %>%   
  mutate(mean_phos = mean(phospho),
         mean_non = mean(none)) %>% 
  ungroup() %>% 
  mutate(mean_ratio = mean_phos / (mean_non + mean_phos),
         rep_ratio = phospho / (none + phospho)) %>% 
  ungroup() %>% 
  mutate(
    sample_id = paste(condition, replicate, sep = "_") )
  
  
  enrichment_efficiency_intensity_df
```



plot
```{r}
##relative to intensity of all peptides (with and without phospho)

plot_ratio_phos_intensity <- ggplot(data = (enrichment_efficiency_intensity_df %>%
                                              mutate(condition = fct_relevel(condition, "EGF0min", "EGF1min", "EGF3min", "EGF5min", "EGF15min")))) + 
  geom_bar(aes(x = condition, y = mean_ratio),
           stat = "identity",
           position = "dodge",
           fill = "skyblue2") +
  geom_point(aes(x = condition,
                 y = rep_ratio,
                 fill = replicate),
             position = position_jitterdodge(dodge.width = 0.35), alpha = 1, size =2, shape = 1, show.legend = FALSE) +
  geom_hline(yintercept = 0.977, linewidth = 1, alpha = 0.5, linetype = 2) +
  annotate(geom = "text",x = 3, y = 0.87, label = "> 97%\npurity", size = 4) +
  alexis_theme()+
  scale_y_continuous(expand = c(0,0))+ 
  expand_limits(y = c(0, 1.005)) +
  # theme_bw(14) +
  # theme(axis.text.x = element_text(angle=90, vjust = 0.25, hjust = 1), legend.position = "none") +
  # theme(legend.position = "none") +
  ylab("p-/all pept (intensity)") + xlab("conditions")

plot_ratio_phos_intensity
ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/phospho_pept_intensity_ratio.png", plot = plot_ratio_phos_intensity, width = 6, height = 8, scale = 0.4)
ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/phospho_pept_intensity_ratio.pdf", plot = plot_ratio_phos_intensity, width = 6, height = 9, scale = 0.5)
```
##_d)  OVERALL ITENSITY
```{r}
plot_psm_intensity_distribution_reps <- ggplot(data = comet %>%
                                                 mutate(condition = fct_relevel(condition, "EGF0min", "EGF1min", "EGF3min", "EGF5min", "EGF15min"))) +
  geom_boxplot(mapping = aes(x = condition, y = log2(intensity), fill = as.factor(replicate)),
               outlier.shape = 21,
               outlier.alpha = 0.3) +
  theme_bw(18) +
  theme(legend.position = "none") +

  scale_fill_brewer(palette = "PuBu") +
  theme(axis.text.x = element_text(angle=90, vjust = 0.25, hjust = 1), legend.position = "none") +
  ylab(expression(log[2]~(intensity))) + xlab("lysate and elution type") +
  ggtitle("Intensity of all PSMs")

plot_psm_intensity_distribution_reps


ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/all_psm_intensity_distribution_reps.png", plot = plot_psm_intensity_distribution_reps, width = 16, height = 16, scale = 0.4)
ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/all_psm_intensity_distribution_reps.pdf", plot = plot_psm_intensity_distribution_reps, width = 16, height = 16, scale = 0.4)
```
pink replicate perhaps is lower intensity related to bead amount.

```{r}
# plot_psm_intensity_distribution_reps <- ggplot(data = comet ) +
#   geom_boxplot(mapping = aes(x = condition, y = log2(intensity), fill = as.factor(replicate)),outliers = FALSE) +
#   theme_bw(10) +
#   theme(legend.position = "none") +
# 
#   scale_fill_brewer(palette = "Purples") +
#   theme(axis.text.x = element_text(angle=90, vjust = 0.25, hjust = 1), legend.position = "none") +
#   ylab(expression(log[2]~(intensity))) + xlab("lysate and elution type") +
#   ggtitle("Intensity of all PSMs")
# 
# plot_psm_intensity_distribution_reps
# 
# 
# ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/plot_psm_intensity_distribution_reps.png", plot = plot_psm_intensity_distribution_reps, width = 10, height = 10, scale = 0.4)
# ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/plot_psm_intensity_distribution_reps.pdf", plot = plot_psm_intensity_distribution_reps, width = 10, height = 10, scale = 0.4)
```


#2. ANALYZE ASCORE
##a. n distinct p isoforms 
###__i) by condition:

---filter for distinct phospho isoforms

```{r}
distinct_p_isoforms_condition <- ascore_fasta_protein_mod_loc_sum_isoform %>% 
  distinct(reference,  condition, isoform_cluster) %>% 
  mutate(condition = fct_relevel(condition, "EGF0min", "EGF1min", "EGF3min", "EGF5min", "EGF15min")) %>% 
  group_by(condition) %>% 
  mutate(n_isoforms = n()) %>% 
  ungroup()
  # group_by(condition,
  #          # replicate, sample_id,
  #          reference, ascore_sequence) %>% 
  # mutate(
  #   isoform_in_peptide_number = paste("site" ,seq(1, n(), by = 1), sep = "" )) %>%
  # ungroup %>% 
  # 
  # pivot_wider(id_cols = c("reference", "ascore_sequence", "condition"), values_from = mod_protein_location, names_from = isoform_in_peptide_number) %>% 
  # mutate(
  #   isoform_cluster = paste(site1, site2, site3, sep = " "),
  #   isoform_cluster = str_replace_all(str_replace_na(isoform_cluster), "NA", "")) #nested turn logical to chr, then replace "NA" with empty character.


##this collapses all  replicates into a single set of unique p isoforms.
```

####------write to csv 
write to csv distinct pSTY isoforms per condition

```{r}
write_csv(x = distinct_p_isoforms_condition, file = "modified_data/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/pSTYisoforms_ascore13_condition.csv", col_names = TRUE)
```

####------plot
plot number of distinct phospho isoforms 
-look at relative fraction of STY and compare to human ratios.

```{r}
plot_distinct_p_isoforms_condition <- ggplot( ) +
  geom_bar(data = distinct_p_isoforms_condition %>% distinct(condition, reference, isoform_cluster),
           mapping = aes(x = condition)) +
  geom_text(data = distinct_p_isoforms_condition %>% distinct(condition, n_isoforms),
            mapping = aes(x = condition, label = n_isoforms, y = n_isoforms - 500), size = 3, color = "white", fontface = "bold") +
  # scale_fill_viridis_d() +
  ylab("unique phospho isoforms") +
  alexis_theme()+
  expand_limits(y = c(0, 15000)) +
  scale_y_continuous(expand = c(0,0))
  # theme(axis.text.x = element_text(angle = 90, vjust = 0.25, hjust = 1))

plot_distinct_p_isoforms_condition

ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/distinct_p_isoforms_condition.png", plot = plot_distinct_p_isoforms_condition, width = 8, height = 8, scale = 0.4)
```

###__ii) by replicate

df
```{r}
distinct_p_isoforms_reps <- ascore_fasta_protein_mod_loc_sum_isoform %>% 
  ungroup() %>% 
  # filter(replicate %in% c("01", "02", "03", "1", "2", "3")) %>% #distinct(replicate) - successfully downsampled to 3 reps 
  distinct(condition, replicate, sample_id, reference, isoform_cluster) %>% 
  mutate(condition = fct_relevel(condition, "EGF0min", "EGF1min", "EGF3min", "EGF5min", "EGF15min")) %>% 
  group_by(condition, replicate, sample_id) %>% 
  mutate(n_isoforms = n()) %>% 
  ungroup() %>% 
  mutate(sample_id = fct_relevel(sample_id,
                                  "EGF0min rep1","EGF0min rep2",
                                  "EGF0min rep3","EGF0min rep4",
                                  "EGF0min rep5","EGF0min rep6",
                                  "EGF1min rep1","EGF1min rep2",
                                  "EGF1min rep3","EGF1min rep4",
                                  "EGF1min rep5","EGF1min rep6",
                                  "EGF3min rep1","EGF3min rep2",
                                  "EGF3min rep3","EGF3min rep4",
                                  "EGF3min rep5","EGF3min rep6",
                                  "EGF5min rep1","EGF5min rep2",
                                  "EGF5min rep3","EGF5min rep4",
                                  "EGF5min rep5","EGF5min rep6",
                                  "EGF15min rep1","EGF15min rep2",
                                  "EGF15min rep3","EGF15min rep4",
                                  "EGF15min rep5","EGF15min rep6"))
##this collapses all  replicates into a single set of unique p isoforms.
```


write to csv
```{r}
write_csv(x = distinct_p_isoforms_reps, file = "modified_data/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/pSTYisoforms_ascore13_reps.csv", col_names = TRUE)
```

plot
```{r}
plot_distinct_p_isoforms_reps <- ggplot() +  
                                       
  geom_bar(data = distinct_p_isoforms_reps %>%
             distinct(sample_id, reference, isoform_cluster),
           mapping = aes(x = sample_id)) +
  geom_text(data = distinct_p_isoforms_reps %>% distinct(  sample_id, n_isoforms),
            mapping = aes(x = sample_id, label = n_isoforms, y = n_isoforms - 500 ), size = 3, color = "white", fontface = "bold", angle = 90) +
  ylab("unique phospho isoforms")  +
  alexis_theme()+
  # expand_limits(y = c(0, 10000)) +
  scale_y_continuous(expand = c(0,0))
  # theme(axis.text.x = element_text(angle = 90, vjust = 0.25, hjust = 1))

plot_distinct_p_isoforms_reps

ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/distinct_p_isoforms_reps.png", plot = plot_distinct_p_isoforms_reps, width = 20, height = 10, scale = 0.4)
ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/distinct_p_isoforms_reps.pdf", plot = plot_distinct_p_isoforms_reps, width = 20, height = 12, scale = 0.4)
```


table and error bar df 
```{r}
summarize_distinct_pisoforms_replicates <- distinct_p_isoforms_reps %>%
  ungroup() %>% 
  group_by(condition, replicate, sample_id, isoform_cluster) %>% 
  summarize(
    n_pisoforms = n()) %>% 
  ungroup()

summarize_distinct_pisoforms_replicates



#errorbar ------------------------------------------------------------
df_geom_errorbar <- summarize_distinct_pisoforms_replicates %>%
  group_by(condition, isoform_cluster) %>%
  mutate(
    min_pisoforms = min(n_pisoforms), 
    max_pisoforms = max(n_pisoforms), 
    mid_pisoforms = mean(n_pisoforms)) %>% 
  ungroup() %>% 
  distinct(condition, isoform_cluster, min_pisoforms, max_pisoforms) #could also include mid pisoforms later if needed

df_geom_errorbar
```


#3.) isoform OVERLAPS



##a). UPSET overlaps

###__i) replicate
all pSTY isoforms
```{r}
df_overlaps_reps <- ascore_fasta_protein_mod_loc_sum_isoform %>% 
  ungroup() %>% 
  distinct(condition, replicate, sample_id, isoform_cluster) %>% 
  mutate(isoform_present = 1) %>% 
  pivot_wider(names_from = sample_id, values_fill = 0, values_from = isoform_present, id_cols = isoform_cluster)

df_overlaps_reps_less <- df_overlaps_reps %>% 
  select(-isoform_cluster) %>% 
  mutate_all(.funs = as.numeric) %>% 
  as.matrix() %>% 
  as.data.frame()

upset_plot_reps <- upset(data = df_overlaps_reps_less, nsets = 30, order.by = "freq", text.scale = 3, nintersects = 10) 
upset_plot_reps


png("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/upset_plot_reps.png", width = 1000, height = 600)
print(upset_plot_reps)
dev.off()


pdf("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/upset_plot_reps.pdf", width = 12, height = 8)
print(upset_plot_reps)
dev.off()
```

###__i) condition
all pSTY isoforms
```{r}
df_overlaps_condition <- ascore_fasta_protein_mod_loc_sum_isoform %>% 
  ungroup() %>% 
  distinct(condition,  isoform_cluster) %>% 
  mutate(isoform_present = 1) %>% 
  pivot_wider(names_from = condition, values_fill = 0, values_from = isoform_present, id_cols = isoform_cluster)

df_overlaps_condition_less <- df_overlaps_condition %>% 
  select(-isoform_cluster) %>% 
  mutate_all(.funs = as.numeric) %>% 
  as.matrix() %>% 
  as.data.frame()

upset_plot_condition <- upset(data = df_overlaps_condition_less, nsets = 7, order.by = "freq", text.scale = 3, nintersects = 15) 
upset_plot_condition


png("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/upset_plot_condition.png", width = 1000, height = 600)
print(upset_plot_condition)
dev.off()


pdf("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/upset_plot_condition.pdf", width = 12, height = 8)
print(upset_plot_condition)
dev.off()
```
15 min has most unique pSTY isoforms. could this reflect increase in downstream STY signaling at the longer timepoint?

##b). isoform overlaps per condition
```{r}
df_overlaps_condition_gt3reps <- ascore_fasta_protein_mod_loc_sum_isoform %>% 
  ungroup() %>% 
  group_by(condition, reference, isoform_cluster) %>%
  filter(n_distinct(replicate) >= 3) %>% 
  ungroup() %>% 
  distinct(condition,reference, isoform_cluster) %>% 
  mutate(isoform_present = 1) %>% 
  pivot_wider(names_from = condition, values_fill = 0, values_from = isoform_present, id_cols = c(isoform_cluster, reference))

df_overlaps_condition_gt3reps_less <- df_overlaps_condition_gt3reps %>% 
  select(-isoform_cluster) %>%
  mutate_all(.funs = as.numeric) %>% 
  as.matrix() %>% #trick for formatting fxns within protti.
  as.data.frame()

# df_overlaps_more <- cbind(df_overlaps_less, isoform_ids) %>%  as.matrix()

upset_plot_pisoforms_gt3reps <- upset(data = df_overlaps_condition_gt3reps_less, nsets = 5, order.by = "freq", text.scale = 3, nintersects = 10)
upset_plot_pisoforms_gt3reps


png("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/upset_plot_pisoforms_gt3reps.png", width = 2000, height = 600)
print(upset_plot_pisoforms_gt3reps)
dev.off()


pdf("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/upset_plot_pisoforms_gt3reps.pdf", width = 12, height = 8)
print(upset_plot_pisoforms_gt3reps)
dev.off()

```
these overlaps seem more realistic to highlight biological signaling nuances given timepoint overlaps. just 15min with most unique, then all other stimulation (less indicative perhaps, then 0 distinct of all EGF stimulation timepoints which could reflect removal of phosphates following stimulation.) Need volcano plot

###__i). gene isoforms unique to condition
```{r}
# df_unique_isoforms_genes_condition <- df_overlaps_condition_gt3reps %>%
#   # separate(col = isoform_cluster, into = c("reference", "isoform"), sep = "_", remove = FALSE) %>%
#   left_join(y = fasta_gene_names %>% rename(reference = entry), by = "reference") %>%
#   mutate(
#     condition_specific = case_when(
#       EGF0min == 1 & EGF1min == 0 & EGF3min == 0 & EGF5min == 0 & EGF15min == 0 ~ "control only",
#       EGF0min == 0 & EGF1min == 1 & EGF3min == 0 & EGF5min == 0 & EGF15min == 0 ~ "EGF1min only",
#       EGF0min == 0 & EGF1min == 0 & EGF3min == 1 & EGF5min == 0 & EGF15min == 0 ~ "EGF3min only",
#       EGF0min == 0 & EGF1min == 0 & EGF3min == 0 & EGF5min == 1 & EGF15min == 0 ~ "EGF5min only",
#       EGF0min == 0 & EGF1min == 0 & EGF3min == 0 & EGF5min == 0 & EGF15min == 1 ~ "EGF15min only",
# 
#       EGF0min == 1 & EGF1min == 1 & EGF3min == 1 & EGF5min == 1 & EGF15min == 1 ~ "all conditions",
#       TRUE ~ "combo")) %>%
# 
#   full_join(y = ppdia_ErbB_signaling %>% mutate(pathway = "ErbB2_Phosphopedia"), by = c("gene"), suffix = c(".mydata", ".ppdia"), relationship = "many-to-many") %>%
#   select(gene, isoform_cluster, condition_specific, EGF0min, EGF1min, EGF3min, EGF5min, EGF15min, pathway,everything())


```
EGFR Y1197 is target of antibody specific for EGF stimulation in vitro of A431 cells, so this target is as expected an hoped.

####---- plot

###---pheatmap
make matrix
```{r}
# pheatmap_df <- df_unique_isoforms_genes_condition %>%
#   
#   #remove duplicate observed pisoforms if multiple reference pisoforms present
#   # group_by(gene, isoform_cluster) %>% 
#   # filter(n_distinct(isoform_cluster) == 1) %>% 
#   # ungroup() %>% 
#  
#   filter(!is.na(isoform.ppdia)) %>%
#   mutate(gene_isoform = paste(gene, "OBS:", isoform.mydata, "isoform_cluster:", isoform.ppdia, sep = " ")) %>%
#   select(gene_isoform, EGF0min:EGF15min) %>% 
#   mutate_at(vars(EGF0min:EGF15min), as.numeric) %>% 
#   filter(!is.na(EGF0min)) 
# 
# # assign df colums to matrix values then row names:
# pheatmap_matrix <- as.matrix(pheatmap_df[,2:6])
# 
# rownames(pheatmap_matrix) <- as.matrix(pheatmap_df[,1])

```

annotation df
```{r}

```


pheatmap plot
```{r}
# ErbB2_pprotein_heatmap <- pheatmap(pheatmap_matrix, cluster_cols = FALSE, cluster_rows = TRUE,
#                                    legend = TRUE, legend_breaks = c(0, 1),
#                                    cutree_rows = 6, cutree_cols = 2,
#                                    fontsize_row = 9, fontsize_col = 10, angle_col = 270, border_color = TRUE)
# 
# ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/ErbB2_pprotein_heatmap.png", plot = ErbB2_pprotein_heatmap, height = 40, width = 15, scale = 0.3)
```
here is a more dialed representation of present vs. not present proteins and associated isoform refrencesby removing duplicate observed isoforms for different reference isoforms.

Notably, I am missing many pY isoforms with global phospho enrichment. I would like to plot the number of pY isoforms in ErbB2 signaling that I do vs. don't detect (relative to annotated reference isoforms) relative to number of pST isoforms I do and don't detect relative to reference isoforms. 
Also, I would like to plot overlap of isoforms in ErbB2 and EGFR and Pi3k signaling to better understand the variation in p-isoforms among reference pathways. Also look in PSP and commercial phospho specific antibodies marketed for tracing ErbB and EGFR signaling. These antibodies will be wayyy more expensive than global and SB enrichment if I can show they enrich isoforms reproducibly and sensitively.

```{r}
# longer_ErbB_signaling <- df_unique_isoforms_genes_condition %>% 
#   pivot_longer(cols = c(EGF0min:EGF15min), names_to = c("condition")) %>% 
#   select(gene, isoform.mydata, isoform.ppdia, condition, value) %>% 
#   mutate(gene_isoform = paste(gene, "OBS:", isoform.mydata, "isoform_cluster:", isoform.ppdia, sep = " ")) %>% 
#   select(gene_isoform, condition, value) %>% 
#   rename(isoform_detected = value)
# 
# ErbB2_pprotein_gt3reps_condition <- ggplot(longer_ErbB_signaling, aes(x = condition, y = gene_isoform, fill = isoform_detected)) +
#   geom_tile(color = "black", show.legend = FALSE) +
#   scale_x_discrete(position = "top", expand = c(0,0)) +
#   scale_y_discrete(expand = c(0,0)) +
#   labs(x = "", y = "", fill = "") +
#   alexis_theme()
#   # facet_grid(~group) +
#   # coord_fixed(ratio = 0.5) +
#   # theme(
#   #         legend.position = "none",
#   #         panel.spacing = unit(0, "lines"), 
#   #         strip.background = element_blank(),
#   #         strip.placement = "outside")
# 
# ErbB2_pprotein_gt3reps_condition
```





##c). intensity per condition overlap
variation in isoform localizations correlates with intensity
```{r}
isoform_overlaps_condition <- df_overlaps_condition %>% 
  ungroup() %>% 
  pivot_longer(cols = c( EGF0min, EGF1min, EGF3min, EGF5min, EGF15min), names_to = c("condition")) %>% 
  group_by(isoform_cluster) %>% 
  mutate(
    n_detections = sum(value)) %>% 
  ungroup()  
  # filter(replicate %in% c(1, 2, 3)) %>% 
  
ascore_isoform_overlaps <- ascore_fasta_protein_mod_loc_sum_isoform %>% 
  inner_join(y = isoform_overlaps_condition %>%  distinct(isoform_cluster, n_detections), by = c("isoform_cluster")) %>% 
  mutate(n_detections = as.factor(n_detections))


```

```{r}
#plot
  ## intensity by condition overlaps---------------------------------------------
intensity_distribution_isoform_overlaps <- ggplot(data = ascore_isoform_overlaps) +
  geom_violin(mapping = aes(x = n_detections, y = median_norm_intensity, fill = condition), draw_quantiles = c(0.25, 0.5, 0.75), show.legend = FALSE) +
  alexis_theme() +
  theme(strip.text = element_text(size = 8)) +
  facet_wrap(facets = vars(condition), nrow = 3)

intensity_distribution_isoform_overlaps
ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/intensity_distribution_isoform_overlaps.png", plot = intensity_distribution_isoform_overlaps, width = 16, height = 14, scale = 0.4)


```



#4. PROTTI
##_a) CVs
###__i) not normalized
```{r}
all_isoform_CVs <- qc_cvs(data = ascore_fasta_protein_mod_loc_sum_isoform,
                            grouping = isoform_cluster,
                            intensity = sum_intensity_precursor_to_peptide,
                            condition = condition,
                            plot = FALSE)

all_isoform_CVs_plot <- qc_cvs(data = ascore_fasta_protein_mod_loc_sum_isoform,
                            grouping = isoform_cluster,
                            intensity = sum_intensity_precursor_to_peptide,
                            condition = condition,
                            plot_style = "violin",
                            plot = TRUE)

#return
all_isoform_CVs
all_isoform_CVs_plot

ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/all_isoform_CVs_plotNotNorm.png", plot = all_isoform_CVs_plot, width = 18, height = 14, scale = 0.4)
```
median %CV of 21% is quite good for phospho. overall the R2P2 procedure seemed to work well regarding reproducibility

###__ii) normalized
```{r}
all_isoform_CVs_NORM <- qc_cvs(data = ascore_fasta_protein_mod_loc_sum_isoform,
                            grouping = isoform_cluster,
                            intensity = raw_median_norm_intensity,
                            condition = condition,
                            plot = FALSE)

all_isoform_CVs_plot_NORM <- qc_cvs(data = ascore_fasta_protein_mod_loc_sum_isoform,
                            grouping = isoform_cluster,
                            intensity = raw_median_norm_intensity,
                            condition = condition,
                            plot_style = "violin",
                            plot = TRUE)

#return
all_isoform_CVs_NORM
all_isoform_CVs_plot_NORM

ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/all_isoform_CVs_plot_Norm.png", plot = all_isoform_CVs_plot_NORM, width = 18, height = 14, scale = 0.4)
```
Hmmm, this is a bit odd to me. Why would the isoforms show higher variation than unique psites? Are the isoforms more difficult to accurately detect?

Will I be able to detect any isoforms with pY pulldown?

####----intensity normalization checks


QC median normalization of intensities
```{r}
#check median normalization by plotting.

##NOT normalized intensities --------------------------------------------------------
check_all_isoform_intensity_before_normalization <- qc_intensity_distribution(
  data = ascore_fasta_protein_mod_loc_sum_isoform,
  sample = sample_id, 
  grouping = isoform_cluster,
  intensity_log2 = sum_intensity_precursor_to_peptide,
  plot_style = "boxplot")

check_all_isoform_intensity_before_normalization

ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/check_all_isoform_intensity_before_normalization.pdf", plot = check_all_isoform_intensity_before_normalization, width = 15, height = 15, scale = 0.5)

ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/check_all_isoform_intensity_before_normalization.png", plot = check_all_isoform_intensity_before_normalization, width = 15, height = 15, scale = 0.5)

##YES normalized intensities --------------------------------------------------------
check_all_isoform_intensity_after_normalization <- qc_intensity_distribution(
  data = ascore_fasta_protein_mod_loc_sum_isoform,
  sample = sample_id, 
  grouping = isoform_cluster,
  intensity_log2 = median_norm_intensity,
  plot_style = "boxplot")

check_all_isoform_intensity_after_normalization

ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/check_all_isoform_intensity_after_normalization.pdf", plot = check_all_isoform_intensity_after_normalization, width = 15, height = 15, scale = 0.5)

ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/check_all_isoform_intensity_after_normalization.png", plot = check_all_isoform_intensity_after_normalization, width = 15, height = 15, scale = 0.5)

```

###---test (collapse multiple entries)

```{r}
test <- ascore_fasta_protein_mod_loc_sum_isoform %>%
  unite(col = "reference_isoform_cluster", c("reference", "isoform_cluster"), sep = " ", remove = TRUE) %>%
  group_by(condition, replicate, sample_id, reference_isoform_cluster) %>%
   
  filter(median_norm_intensity == max(median_norm_intensity)) %>%
  ungroup() %>% 
  select(-replicate)
  # mutate(n_isoform_intensities_obs = n())
```



##_c) MISSINGNESS
```{r}
ascore_med_norm_missing <- assign_missingness(
    data = test,
    sample = sample_id, 
    grouping = reference_isoform_cluster, 
    intensity = median_norm_intensity,
    condition = condition, 
    ref_condition = "all",
    completeness_MAR = 0.7, #require 5 of 6 reps to have to be considered condition specific signature
    completeness_MNAR = 0.15 ) #any time obs is in 1 or less samples per condition, the isoform is considered biologically missing.
```
##_b) hierarchical clustering
```{r}
#needed to use missingness assigned df here, not exactly sure why... I think not great overlaps.
all_isoform_correlation <- qc_sample_correlation(
  data = ascore_med_norm_missing,
  sample = sample_id,
  condition = condition,
  grouping = reference_isoform_cluster,
  intensity_log2 = median_norm_intensity,
  interactive = FALSE)


all_isoform_correlation

ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/all_isoform_correlation.png", plot = all_isoform_correlation, width = 20, height = 20, scale = 0.4)
ggsave("output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/all_isoform_correlation.pdf", plot = all_isoform_correlation, width = 16, height = 12, scale = 0.4)
```

Sweet! even when samples were randomized correlation is good and reveals condition specific grouping!


##_d) DIFFERENTIAL ABUNDANCE

```{r}
isoform_diff_abundance <- ascore_med_norm_missing %>% 
  calculate_diff_abundance(
    sample = sample_id,
    condition = condition, 
    grouping = reference_isoform_cluster, 
    intensity_log2 = median_norm_intensity,
    missingness = missingness,
    comparison = comparison,
    filter_NA_missingness = TRUE,
    method = "moderated_t-test" )
```
##_d2) VOLCANO PLOT
```{r}
isoform_volcano <- volcano_plot(
  data = isoform_diff_abundance,
  grouping = reference_isoform_cluster,
  log2FC = diff,
  significance = pval,
  significance_cutoff = c(0.05, "adj_pval"),
  method = "significant",
  facet_by = comparison)

isoform_volcano
ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/isoform_volcano.png", plot = isoform_volcano, width = 30, height = 18, scale = 0.4)
```
It is so cool to see the cell biology (phospho signaling) change so quickly upon a perturbation and to change in a way expected.

I am curious to see what these isoforms are.

Really nice to see the magnitude of number of pisoforms changes to correlate with duration of EGF stimulation!!

I want to next plot this with my custom volcano formatting (i.e. opaque points) and label with 'gene + site' as well as quantify amount of pisoforms changing per condition as a percentage. 

Also, I think I learn by getting my hands dirty, and getting excited about this experiment makes me want to revisit Mario's yeast perturbation paper to see how he dug into a preliminary analysis such as volcano plot upon stress (x100+)


#####-----overall percent p-isoforms changing per cell type
```{r}
percent_pisoforms_diff_regulated <- isoform_diff_abundance %>%
  distinct(comparison, reference_isoform_cluster, diff, adj_pval) %>%
  group_by(comparison) %>%
  mutate(total_pisoforms = n_distinct(reference_isoform_cluster)) %>%
  ungroup() %>%
  mutate(
    regulated_isoform = case_when(
      adj_pval <= 0.05 & abs(diff) >= 1 ~ "regulated",
      TRUE ~ "not regulated")) %>%
  group_by(comparison, regulated_isoform) %>%
  mutate(n_regulated_pisoforms = n()) %>%
  ungroup() %>%
  distinct(comparison, total_pisoforms, regulated_isoform, n_regulated_pisoforms) %>%
  pivot_wider(id_cols = c(comparison, total_pisoforms), names_from = regulated_isoform, values_from = n_regulated_pisoforms) %>%
  group_by(comparison) %>%
  mutate(percent_regulated = regulated / total_pisoforms *100)
  

percent_pisoforms_diff_regulated
```
Wow! really cool to see the amount of isoform regulation correlate with extent of EGF stimulation! Suggesting effect correlates with phospho signature in some way at least. now to see which phosphoisoforms... (remember I need to adjust for protein abundance differences if any pair with regulated phosphoisoforms) - also just see if these could be a measurment issue.
Also, only a small percentage of pisoforms detected appear regulated. this is alright, but is it the full truth?

#####-----percent regulated 
```{r}
percent_isoform_regulated <- isoform_diff_abundance %>% 
  distinct(comparison, diff, adj_pval, reference_isoform_cluster) %>% 
  group_by(comparison) %>% 
  mutate(total_pisoforms = n_distinct(reference_isoform_cluster)) %>%
  ungroup() %>%
  mutate(
    regulated_isoform = case_when(
      adj_pval <= 0.05 & abs(diff) >= 1 ~ "regulated",
      TRUE ~ "not regulated"),
    regulated_isoform_dir = case_when(
      regulated_isoform == "regulated" & diff >= 1 ~ "up",
      regulated_isoform == "regulated" & diff <= -1 ~ "down",
      regulated_isoform == "not regulated" ~ "not regulated")) %>%
  group_by(comparison, regulated_isoform_dir) %>% 
  mutate(n_regulated_pisoforms = n()) %>% 
  ungroup() %>% 
  distinct(comparison, total_pisoforms, regulated_isoform, n_regulated_pisoforms, regulated_isoform_dir) %>% 
  pivot_wider(id_cols= c(comparison, total_pisoforms), names_from = regulated_isoform_dir, values_from = n_regulated_pisoforms) %>% 
  group_by(comparison) %>% 
  mutate(percent_upregulated = up / total_pisoforms*100,
         percent_downregulated = down / total_pisoforms*100)
  

percent_isoform_regulated
```
####----my volcano plot

flipped 15 min vs. 0 min below, skip this 0 min vs. 15 min immediately underneath
```{r}

# # EGF15min vs. EGF0min (aka control/ untreated) ------------------------------------------------------
# 
# my_isoform_volcano_plot_EGF0min_vs_15min <- ggplot() +
#   geom_vline(xintercept = c(-1, 1), color = "darkgrey", alpha = 0.5, size =1, linetype = 2)+
#   geom_hline(yintercept = -log10(0.05), color = "darkgrey", alpha = 0.5, size = 1, linetype = 2) +
#   geom_point(data = isoform_diff_abundance %>%
#                filter(comparison == "EGF0min_vs_EGF15min") %>% 
#                # filter(abs(diff) < 1) %>% 
#                filter(adj_pval >0.05),
#              mapping = aes(x = diff, y = -log10(adj_pval)), color = "grey", alpha = 0.1, shape =19) +
#   geom_point(data = isoform_diff_abundance %>%
#                filter(comparison == "EGF0min_vs_EGF15min") %>% 
#                filter(abs(diff) < 1) %>%
#                filter(adj_pval <0.05),
#              mapping = aes(x = diff, y = -log10(adj_pval)), color = "grey", alpha = 0.1, shape =19) +
#   geom_point(data = isoform_diff_abundance %>%
#                filter(comparison == "EGF0min_vs_EGF15min") %>%
#                filter(diff < -1) %>% 
#                filter(adj_pval <= 0.05),
#              mapping = aes(x = diff, y = -log10(adj_pval)), color = "blue", alpha = 0.1, shape =19) +
#   geom_point(data = isoform_diff_abundance %>%
#                filter(comparison == "EGF0min_vs_EGF15min") %>% 
#                filter(diff > 1) %>% 
#                filter(adj_pval <= 0.05),
#              mapping = aes(x = diff, y = -log10(adj_pval)), color = "red", alpha = 0.1, shape =19) +
#   theme_bw(18) +
#   facet_wrap(facets = vars(comparison)) +
#   scale_x_continuous(name = "Log2(fold change)", breaks = seq(-8, 8, 2), limits = c(-8, 8)) +
#   scale_y_continuous(name = "-Log10(adj. p-value)", breaks = seq(0,20,5), limits = c(0, 20))
# 
# my_isoform_volcano_plot_EGF0min_vs_15min
# 
# ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/my_isoform_volcano_plot_EGF0min_vs_15min.png", plot = my_isoform_volcano_plot_EGF0min_vs_15min, width = 10, height = 8, scale = 0.4)
# 
# ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/my_isoform_volcano_plot_EGF0min_vs_15min.pdf", plot = my_isoform_volcano_plot_EGF0min_vs_15min, width = 10, height = 8, scale = 0.4)
```
####----STY color + labels
could be nice to color by pSTY, and to add gene names to volcano with regulated isoforms. Also write up and down regulated percentages.
```{r}
isoform_diff_abundance_gene <- isoform_diff_abundance %>%
  mutate(reference = str_sub(reference_isoform_cluster, start = 1L, end = 6L)) %>% 
  # separate(col = isoform_cluster, into = c("reference", "modres_position"), sep = "_", remove = FALSE) %>% 
  # mutate(mod_res = str_sub(modres_position, start  = 1L, end = 1L),
  #        position = as.numeric(str_sub(modres_position, start = 2L))) %>%
  left_join(y = (fasta_gene_names %>% select(entry, gene, protein_names, keyword_id) %>% rename(reference = entry)), by = "reference") %>% 
    #flip the 0 min and 15 min comparison
  mutate(diff = case_when(
    comparison == "EGF0min_vs_EGF15min" ~ -diff,
    TRUE ~ diff)) %>% 
  mutate(comparison = case_when(
    comparison == "EGF0min_vs_EGF15min" ~ "EGF15min_vs_EGF0min", 
    TRUE ~ comparison)) 
```

##--15 min vs. 0 min
```{r}
# EGF15min vs. EGF0min (aka control/ untreated) ------------------------------------------------------

my_isoform_volcano_plot_EGF15min_vs_0min_STYcolor <- ggplot() +
  geom_vline(xintercept = c(-1, 1), color = "darkgrey", alpha = 0.5, size =1, linetype = 2)+
  geom_hline(yintercept = -log10(0.05), color = "darkgrey", alpha = 0.5, size = 1, linetype = 2) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF15min_vs_EGF0min") %>% 
               # filter(abs(diff) < 1) %>% 
               filter(adj_pval >0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "grey", alpha = 0.3, shape =19) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF15min_vs_EGF0min") %>% 
               filter(abs(diff) < 1) %>%
               filter(adj_pval <0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "grey", alpha = 0.3, shape =19) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF15min_vs_EGF0min") %>%
               filter(diff < -1) %>% 
               filter(adj_pval <= 0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "red", alpha = 0.3, shape =19) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF15min_vs_EGF0min") %>% 
               filter(diff > 1) %>% 
               filter(adj_pval <= 0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "blue", alpha = 0.3, shape =19) +
  facet_wrap(facets = vars(comparison)) +
  # geom_text_repel(data = isoform_diff_abundance_gene %>%
  #              filter(comparison == "EGF15min_vs_EGF0min") %>% 
  #              filter(diff >= 1) %>% 
  #              filter(adj_pval <= 0.05),
  #              mapping = aes(x = diff, y = -log10(adj_pval), label = gene),
  #              max.overlaps = 200,
  #              segment.color = NA,
  #              # position = position_stack( hjust = 6),
  #              nudge_x = 8,
  #              force_pull   = 0, # do not pull toward data points
  #              # xlim = c(-Inf, Inf), ylim = c(-Inf, Inf),
  #              xlim = c(0, 10), ylim = c(-0.5, 21),
  #              direction = "y",
  #              # vjust = 1,
  #              hjust = "right",
  #              size = 2) +
  # geom_text_repel(data = isoform_diff_abundance_gene %>%
  #              filter(comparison == "EGF15min_vs_EGF0min") %>% 
  #              filter(diff <= -1.5) %>% 
  #              filter(adj_pval <= 0.01),
  #              mapping = aes(x = diff, y = -log10(adj_pval), label = gene),
  #              max.overlaps = 200,
  #              segment.color = NA,
  #              # position = position_stack( hjust = 6),
  #              nudge_x = -8,
  #              direction = "y",
  #              force_pull   = 0, # do not pull toward data points
  #              # xlim = c(-Inf, Inf), ylim = c(-Inf, Inf),
  #              xlim = c(-10, 0), ylim = c(-0.5, 21),
  #              # vjust = 1,
  #              hjust = "left",
  #              size = 2) +
  alexis_theme() +
  coord_cartesian(clip = 'off') +
  
  scale_x_continuous(name = "Log2(fold change)", breaks = seq(-6, 6, 2), limits = c(-6, 6)) +
  scale_y_continuous(name = "-Log10(adj. p-value)", breaks = seq(0,16,4), limits = c(0, 16))

my_isoform_volcano_plot_EGF15min_vs_0min_STYcolor

ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/my_isoform_volcano_plot_EGF15min_vs_0min_STYcolor.png", plot = my_isoform_volcano_plot_EGF15min_vs_0min_STYcolor, width = 10, height = 8, scale = 0.4)

ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/my_isoform_volcano_plot_EGF15min_vs_0min_STYcolor.pdf", plot = my_isoform_volcano_plot_EGF15min_vs_0min_STYcolor, width = 15, height = 10, scale = 0.4)
```
###--5 min vs. 0 min
```{r}
# EGF5min vs. EGF0min (aka control/ untreated) ------------------------------------------------------

my_isoform_volcano_plot_EGF5min_vs_0min_STYcolor <- ggplot() +
  geom_vline(xintercept = c(-1, 1), color = "darkgrey", alpha = 0.5, size =1, linetype = 2)+
  geom_hline(yintercept = -log10(0.05), color = "darkgrey", alpha = 0.5, size = 1, linetype = 2) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF5min_vs_EGF0min") %>% 
               # filter(abs(diff) < 1) %>% 
               filter(adj_pval >0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "grey", alpha = 0.3, shape =19) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF5min_vs_EGF0min") %>% 
               filter(abs(diff) < 1) %>%
               filter(adj_pval <0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "grey", alpha = 0.3, shape =19) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF5min_vs_EGF0min") %>%
               filter(diff < -1) %>% 
               filter(adj_pval <= 0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "red", alpha = 0.3, shape =19) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF5min_vs_EGF0min") %>% 
               filter(diff > 1.0) %>% 
               filter(adj_pval <= 0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "blue", alpha = 0.3, shape =19) +
  facet_wrap(facets = vars(comparison)) +
  # geom_text_repel(data = isoform_diff_abundance_gene %>%
  #              filter(comparison == "EGF5min_vs_EGF0min") %>%
  #              filter(diff >= 1) %>%
  #              filter(adj_pval <= 0.05),
  #              mapping = aes(x = diff, y = -log10(adj_pval), label = gene),
  #              max.overlaps = 200,
  #              segment.color = NA,
  #              # position = position_stack( hjust = 6),
  #              nudge_x = 8,
  #              force_pull   = 0, # do not pull toward data points
  #              # xlim = c(-Inf, Inf), ylim = c(-Inf, Inf),
  #              xlim = c(0, 10), ylim = c(-0.5, 21),
  #              direction = "y",
  #              # vjust = 1,
  #              hjust = "right",
  #              size = 2) +
  # geom_text_repel(data = isoform_diff_abundance_gene %>%
  #              filter(comparison == "EGF5min_vs_EGF0min") %>%
  #              filter(diff <= -1.5) %>%
  #              filter(adj_pval <= 0.01),
  #              mapping = aes(x = diff, y = -log10(adj_pval), label = gene),
  #              max.overlaps = 200,
  #              segment.color = NA,
  #              # position = position_stack( hjust = 6),
  #              nudge_x = -8,
  #              direction = "y",
  #              force_pull   = 0, # do not pull toward data points
  #              # xlim = c(-Inf, Inf), ylim = c(-Inf, Inf),
  #              xlim = c(-10, 0), ylim = c(-0.5, 21),
  #              # vjust = 1,
  #              hjust = "left",
  #              size = 2) +
  alexis_theme() +
  coord_cartesian(clip = 'off') +

  scale_x_continuous(name = "Log2(fold change)", breaks = seq(-6, 6, 2), limits = c(-6, 6)) +
  scale_y_continuous(name = "-Log10(adj. p-value)", breaks = seq(0,16,4), limits = c(0, 16))

my_isoform_volcano_plot_EGF5min_vs_0min_STYcolor

ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/my_isoform_volcano_plot_EGF5min_vs_0min_STYcolor.png", plot = my_isoform_volcano_plot_EGF5min_vs_0min_STYcolor, width = 15, height = 10, scale = 0.4)

ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/my_isoform_volcano_plot_EGF5min_vs_0min_STYcolor.pdf", plot = my_isoform_volcano_plot_EGF5min_vs_0min_STYcolor, width = 15, height = 10, scale = 0.4)
```

###--3 min vs. 0 min
```{r}
# EGF3min vs. EGF0min (aka control/ untreated) ------------------------------------------------------

my_isoform_volcano_plot_EGF3min_vs_0min_STYcolor <- ggplot() +
  geom_vline(xintercept = c(-1, 1), color = "darkgrey", alpha = 0.5, size =1, linetype = 2)+
  geom_hline(yintercept = -log10(0.05), color = "darkgrey", alpha = 0.5, size = 1, linetype = 2) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF3min_vs_EGF0min") %>% 
               # filter(abs(diff) < 1) %>% 
               filter(adj_pval >0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "grey", alpha = 0.3, shape =19) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF3min_vs_EGF0min") %>% 
               filter(abs(diff) < 1) %>%
               filter(adj_pval <0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "grey", alpha = 0.3, shape =19) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF3min_vs_EGF0min") %>%
               filter(diff < -1) %>% 
               filter(adj_pval <= 0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "red", alpha = 0.3, shape =19) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF3min_vs_EGF0min") %>% 
               filter(diff > 1) %>% 
               filter(adj_pval <= 0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "blue", alpha = 0.3, shape =19) +
  facet_wrap(facets = vars(comparison)) +
  # geom_text_repel(data = isoform_diff_abundance_gene %>%
  #              filter(comparison == "EGF3min_vs_EGF0min") %>%
  #              filter(diff >= 1) %>%
  #              filter(adj_pval <= 0.05),
  #              mapping = aes(x = diff, y = -log10(adj_pval), label = gene),
  #              max.overlaps = 200,
  #              segment.color = NA,
  #              # position = position_stack( hjust = 6),
  #              nudge_x = 8,
  #              force_pull   = 0, # do not pull toward data points
  #              # xlim = c(-Inf, Inf), ylim = c(-Inf, Inf),
  #              xlim = c(0, 10), ylim = c(-0.5, 21),
  #              direction = "y",
  #              # vjust = 1,
  #              hjust = "right",
  #              size = 2) +
  # geom_text_repel(data = isoform_diff_abundance_gene %>%
  #              filter(comparison == "EGF3min_vs_EGF0min") %>%
  #              filter(diff <= -1.5) %>%
  #              filter(adj_pval <= 0.01),
  #              mapping = aes(x = diff, y = -log10(adj_pval), label = gene),
  #              max.overlaps = 200,
  #              segment.color = NA,
  #              # position = position_stack( hjust = 6),
  #              nudge_x = -8,
  #              direction = "y",
  #              force_pull   = 0, # do not pull toward data points
  #              # xlim = c(-Inf, Inf), ylim = c(-Inf, Inf),
  #              xlim = c(-10, 0), ylim = c(-0.5, 21),
  #              # vjust = 1,
  #              hjust = "left",
  #              size = 2) +
  alexis_theme() +
  coord_cartesian(clip = 'off') +

  scale_x_continuous(name = "Log2(fold change)", breaks = seq(-6, 6, 2), limits = c(-6, 6)) +
  scale_y_continuous(name = "-Log10(adj. p-value)", breaks = seq(0,16,4), limits = c(0, 16))

my_isoform_volcano_plot_EGF3min_vs_0min_STYcolor

ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/my_isoform_volcano_plot_EGF3min_vs_0min_STYcolor.png", plot = my_isoform_volcano_plot_EGF3min_vs_0min_STYcolor, width = 15, height = 10, scale = 0.4)

ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/my_isoform_volcano_plot_EGF3min_vs_0min_STYcolor.pdf", plot = my_isoform_volcano_plot_EGF3min_vs_0min_STYcolor, width = 15, height = 10, scale = 0.4)
```
Wow. many less isoforms are differentially regulated at 3 min. Likely due to sparse isoform data. MAybe the ascore> 13 filter for all sites drops off some of those isoforms if only 1 - 2 of the sites are confidently detected in some samples.

###--1 min vs. 0 min
```{r}
# EGF1min vs. EGF0min (aka control/ untreated) ------------------------------------------------------

my_isoform_volcano_plot_EGF1min_vs_0min_STYcolor <- ggplot() +
  geom_vline(xintercept = c(-1, 1), color = "darkgrey", alpha = 0.5, size =1, linetype = 2)+
  geom_hline(yintercept = -log10(0.05), color = "darkgrey", alpha = 0.5, size = 1, linetype = 2) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF1min_vs_EGF0min") %>% 
               # filter(abs(diff) < 1) %>% 
               filter(adj_pval >0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "grey", alpha = 0.3, shape =19) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF1min_vs_EGF0min") %>% 
               filter(abs(diff) < 1) %>%
               filter(adj_pval <0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "grey", alpha = 0.3, shape =19) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF1min_vs_EGF0min") %>%
               filter(diff < -1) %>% 
               filter(adj_pval <= 0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "red", alpha = 0.3, shape =19) +
  geom_point(data = isoform_diff_abundance_gene %>%
               filter(comparison == "EGF1min_vs_EGF0min") %>% 
               filter(diff > 1) %>% 
               filter(adj_pval <= 0.05),
             mapping = aes(x = diff, y = -log10(adj_pval)), color = "blue", alpha = 0.3, shape =19) +
  facet_wrap(facets = vars(comparison)) +
  # geom_text_repel(data = isoform_diff_abundance_gene %>%
  #              filter(comparison == "EGF1min_vs_EGF0min") %>%
  #              filter(diff >= 1) %>%
  #              filter(adj_pval <= 0.05),
  #              mapping = aes(x = diff, y = -log10(adj_pval), label = gene),
  #              max.overlaps = 200,
  #              segment.color = NA,
  #              # position = position_stack( hjust = 6),
  #              nudge_x = 8,
  #              force_pull   = 0, # do not pull toward data points
  #              # xlim = c(-Inf, Inf), ylim = c(-Inf, Inf),
  #              xlim = c(0, 10), ylim = c(-0.5, 21),
  #              direction = "y",
  #              # vjust = 1,
  #              hjust = "right",
  #              size = 2) +
  # geom_text_repel(data = isoform_diff_abundance_gene %>%
  #              filter(comparison == "EGF1min_vs_EGF0min") %>%
  #              filter(diff <= -1.5) %>%
  #              filter(adj_pval <= 0.01),
  #              mapping = aes(x = diff, y = -log10(adj_pval), label = gene),
  #              max.overlaps = 200,
  #              segment.color = NA,
  #              # position = position_stack( hjust = 6),
  #              nudge_x = -8,
  #              direction = "y",
  #              force_pull   = 0, # do not pull toward data points
  #              # xlim = c(-Inf, Inf), ylim = c(-Inf, Inf),
  #              xlim = c(-10, 0), ylim = c(-0.5, 21),
  #              # vjust = 1,
  #              hjust = "left",
  #              size = 2) +
  alexis_theme() +
  coord_cartesian(clip = 'off') +

  scale_x_continuous(name = "Log2(fold change)", breaks = seq(-6, 6, 2), limits = c(-6, 6)) +
  scale_y_continuous(name = "-Log10(adj. p-value)", breaks = seq(0,16,4), limits = c(0, 16))

my_isoform_volcano_plot_EGF1min_vs_0min_STYcolor

ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/my_isoform_volcano_plot_EGF1min_vs_0min_STYcolor.png", plot = my_isoform_volcano_plot_EGF1min_vs_0min_STYcolor, width = 15, height = 10, scale = 0.4)

ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/my_isoform_volcano_plot_EGF1min_vs_0min_STYcolor.pdf", plot = my_isoform_volcano_plot_EGF1min_vs_0min_STYcolor, width = 15, height = 10, scale = 0.4)
```

## pheatmap (only diff > +-1 and  significant)
make matrix
```{r}
pheatmap_diff_pval_df <- isoform_diff_abundance_gene %>%
  filter(comparison %in% c("EGF15min_vs_EGF0min", "EGF5min_vs_EGF0min", "EGF3min_vs_EGF0min", "EGF1min_vs_EGF0min")) %>%
  
  # #flip the 0 min and 15 min comparison
  # mutate(diff = case_when(
  #   comparison == "EGF0min_vs_EGF15min" ~ -diff,
  #   TRUE ~ diff)) %>% 
  # mutate(comparison = case_when(
  #   comparison == "EGF0min_vs_EGF15min" ~ "EGF15min_vs_EGF0min", 
  #   TRUE ~ comparison)) %>% 
  
  mutate(
    isoform_cluster = str_sub(reference_isoform_cluster, start = 8L),
    gene_isoform = paste(gene, isoform_cluster, sep = " ")) %>% 
  filter(abs(diff) > 1) %>% 
  filter(adj_pval <= 0.05) %>% 
  pivot_wider(id_cols = c(gene_isoform), names_from = comparison, values_from = c(diff, adj_pval), values_fill = 0) %>% 
  select(gene_isoform, diff_EGF1min_vs_EGF0min, diff_EGF3min_vs_EGF0min, diff_EGF5min_vs_EGF0min, diff_EGF15min_vs_EGF0min, everything())
```


```{r}
# #add in control for range of log2FC to center colors
# control_rows_df <- data.frame(row.names = c("CTRL S0", "CTRL S100", "CTRL T0", "CTRL T100","CTRL Y0", "CTRL Y100"),
#                               diff_EGF1min_vs_EGF0min = c(-6, 6, -6, 6, -6, 6),
#                               diff_EGF3min_vs_EGF0min = c(-6, 6, -6, 6, -6, 6),
#                               diff_EGF5min_vs_EGF0min = c(-6, 6, -6, 6, -6, 6),
#                               diff_EGF15min_vs_EGF0min = c(-6, 6, -6, 6, -6, 6))
```

###---just one row per mod site
```{r}
#add in control for range of log2FC to center colors
control_rows_df <- data.frame(row.names = c("CTRL S0", "CTRL S100"),
                              diff_EGF1min_vs_EGF0min = c(-6, 6),
                              diff_EGF3min_vs_EGF0min = c(-6, 6),
                              diff_EGF5min_vs_EGF0min = c(-6, 6),
                              diff_EGF15min_vs_EGF0min = c(-6, 6))




# assign df columns to matrix values then row names:
# pheatmap_diff_pval_matrix <- as.matrix(pheatmap_diff_pval_matrix_wCTRL[,4:13]) #column range for all conditions

#may need to uncomment:
pheatmap_diff_pval_matrix_wCTRL <- as.matrix(pheatmap_diff_pval_df[,2:5])

rownames(pheatmap_diff_pval_matrix_wCTRL) <- as.matrix(pheatmap_diff_pval_df[,1])


pheatmap_diff_pval_matrix_wCTRL <- as.matrix(rbind(control_rows_df, pheatmap_diff_pval_matrix_wCTRL))

```

###---annotation df
####-----pSTY
```{r}
# annotate_row_pSTY <- pheatmap_diff_pval_df %>% select(gene_isoform, mod_res) %>% 
#   mutate(mod_res = case_when(
#     mod_res == "S" ~ "pSer",
#     mod_res == "T" ~ "pThr",
#     mod_res == "Y" ~ "pTyr" ))
```


####-----colors
```{r}
  my_colour = list(
    # sample = c(normal = "#5977ff", tumour = "#f74747"),
    mod_res = c(pSer = "purple3", pThr = "darkcyan", pTyr = "darkgoldenrod1" ))
    # random = c(random1 = "#82ed82", random2 = "#9e82ed"),
    # cluster = c(cluster1 = "#e89829", cluster2 = "#cc4ee0")
```

###---pheatmap plot
with fold change greater than 1 and adj pvalue below 0.05
```{r}

signif_change_pprotein_heatmap <- pheatmap(pheatmap_diff_pval_matrix_wCTRL,
                                           # annotation_row = annotate_row_pSTY,
                                           cluster_cols = FALSE,
                                           cluster_rows = TRUE,
                                           colorRampPalette(c("purple", "white", "orange"))(48),
                                           annotation_colors = my_colour,
                                           legend_breaks = c(-6,0, 6),
                                           display_numbers = TRUE,
                                           fontsize_number = 3,
                                           cutree_rows = 6, cutree_cols = 2,
                                           fontsize_row = 3, fontsize_col = 4, angle_col = 270,
                                           border_color = "black" )


ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/signif_change_isoform_cluster_heatmap.png", plot = signif_change_pprotein_heatmap, height = 40, width = 10, scale = 0.4)
```
## pheatmap all diff and significance
make matrix
```{r}
pheatmap_diff_pval_df_all <- isoform_diff_abundance_gene %>%
  filter(comparison %in% c("EGF15min_vs_EGF0min", "EGF5min_vs_EGF0min", "EGF3min_vs_EGF0min", "EGF1min_vs_EGF0min")) %>%
  #
  # #flip the 0 min and 15 min comparison
  # mutate(diff = case_when(
  #   comparison == "EGF0min_vs_EGF15min" ~ -diff,
  #   TRUE ~ diff)) %>%
  # mutate(comparison = case_when(
  #   comparison == "EGF0min_vs_EGF15min" ~ "EGF15min_vs_EGF0min",
  #   TRUE ~ comparison)) %>%


  mutate(
    isoform_cluster = str_sub(reference_isoform_cluster, start = 8L),
    gene_isoform = paste(gene, isoform_cluster, sep = " ")) %>%
  # filter(abs(diff) > 1) %>%
  # filter(adj_pval <= 0.05) %>%
  pivot_wider(id_cols = c(gene_isoform), names_from = comparison, values_from = c(diff, adj_pval), values_fill = 0) %>%
  #change NA to 0:
  mutate(
    diff_EGF1min_vs_EGF0min = ifelse(is.na(diff_EGF1min_vs_EGF0min), 0, diff_EGF1min_vs_EGF0min),
    diff_EGF3min_vs_EGF0min = ifelse(is.na(diff_EGF3min_vs_EGF0min), 0, diff_EGF3min_vs_EGF0min),
    diff_EGF5min_vs_EGF0min = ifelse(is.na(diff_EGF5min_vs_EGF0min), 0, diff_EGF5min_vs_EGF0min),
    diff_EGF15min_vs_EGF0min = ifelse(is.na(diff_EGF15min_vs_EGF0min), 0, diff_EGF15min_vs_EGF0min)) %>%
  select(gene_isoform, diff_EGF1min_vs_EGF0min, diff_EGF3min_vs_EGF0min, diff_EGF5min_vs_EGF0min, diff_EGF15min_vs_EGF0min, everything()) 


# assign df colums to matrix values then row names:
# pheatmap_diff_pval_matrix <- as.matrix(pheatmap_diff_pval_df[,4:13]) #column range for all conditions
pheatmap_diff_pval_matrix_all <- as.matrix(pheatmap_diff_pval_df_all[,2:5])

rownames(pheatmap_diff_pval_matrix_all) <- as.matrix(pheatmap_diff_pval_df_all[,1])

pheatmap_diff_pval_matrix_all_wCTRL <- as.matrix(rbind(control_rows_df, pheatmap_diff_pval_matrix_all))

```

annotation df
```{r}

```


pheatmap plot
with fold change greater than 1 and adj pvalue below 0.05
```{r}
signif_change_pprotein_heatmap_all <- pheatmap(pheatmap_diff_pval_matrix_all_wCTRL[c(1:2, 1006:1505),],
                                           cluster_cols = FALSE,
                                           cluster_rows = TRUE,
                                           colorRampPalette(c("purple", "white", "orange"))(48),
                                   legend = TRUE, legend_breaks = c(-6,0, 6),
                                   cutree_rows = 6, cutree_cols = 4,
                                   display_numbers = TRUE,
                                           fontsize_number = 3,
                                   fontsize_row = 3, fontsize_col = 3, angle_col = 270,
                                   border_color = "black" )

ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/signif_change_pprotein_heatmap_all_isoforms1001-1500.png", plot = signif_change_pprotein_heatmap_all, height = 80, width = 6, scale = 0.4)
```
So I see the EGFR pY isoforms, but they show minimal fold change. 
Make boxplots with points to see if minimal fold change is due to variation in intensity or low signal relative to noise, or in fact minimal change in intensity of those EGFR isoforms.


#5. BOXPLOTS + POINTS
##_a) df
```{r}
ascore_final_gene <- ascore_fasta_protein_mod_loc_sum_isoform %>%
  left_join(y = (fasta_gene_names %>% select(entry, gene, protein_names, keyword_id) %>% rename(reference = entry)), by = "reference") %>%
  mutate(
    gene_ref = paste(gene, isoform_cluster, sep = " "),
    condition = as.factor(condition),
    condition = fct_relevel(condition, "EGF0min", "EGF1min",  "EGF3min", "EGF5min", "EGF15min")) %>% 
  group_by(condition, gene_ref ) %>% 
  mutate(
    n_obs = n() ) %>%
  ungroup()


# ascore_final_gene <- isoform_diff_abundance_gene %>%
# 
#   filter(comparison %in% c("EGF15min_vs_EGF0min", "EGF5min_vs_EGF0min", "EGF3min_vs_EGF0min", "EGF1min_vs_EGF0min")) %>%
#   mutate(
#     isoform_cluster = str_sub(reference_isoform_cluster, start = 8L),
#     gene_isoform = paste(gene, isoform_cluster, sep = " "),
#     condition = as.factor(condition),
#     condition = fct_relevel(condition, "EGF0min", "EGF1min",  "EGF3min", "EGF5min", "EGF15min")) %>%
#   group_by(condition, isoform_cluster, ) %>% 
#   mutate(
#     n_obs = n() ) %>% 
#   ungroup()
```

####--- Save input data for SHINY
```{r}
most_common_protein_redundancy <- ascore_fasta_protein_mod_loc %>% 
  #first assign isoforms, but without summing protein intensities as done for ...sum_isoform dataframe.
  group_by(condition, replicate, sample_id, ascore_sequence, charge, ascore) %>% 
  filter(intensity == max(intensity)) %>% 
  ungroup() %>% 
  distinct(condition, replicate, sample_id, reference, ascore_sequence, mod_protein_location, intensity, proteins, redundancy) %>% 


  ##SUM PRECURSOR INTENSITIES TO P-isoforms clusters + median normalize

 
  group_by(condition, replicate,  sample_id, reference, ascore_sequence, redundancy, proteins) %>% #ascore_sequence = phospho peptide sequence = isoform
  mutate(
    isoform_in_peptide_number = paste("site" ,seq(1, n(), by = 1), sep = "" )) %>%
  ungroup %>% 
  
  pivot_wider(id_cols = c("condition", "replicate", "sample_id", "reference", "ascore_sequence", "intensity", "redundancy", "proteins"),
              values_from = mod_protein_location,
              names_from = isoform_in_peptide_number) %>% 
  mutate(
    isoform_cluster = paste(site1, site2, site3, sep = " "),
    isoform_cluster = str_squish(str_replace_all(str_replace_na(isoform_cluster), "NA", ""))) %>% 
  group_by(reference, isoform_cluster, proteins, redundancy) %>% 
  mutate(n_obs_each_redundancy = n()) %>% 
  ungroup() %>% 
  group_by(reference, isoform_cluster) %>% 
  filter(n_obs_each_redundancy == max(n_obs_each_redundancy)) %>% #keep most common redunancy assignment per isoform across all samples
  ungroup() %>% 
  distinct(sample_id, condition, replicate, reference, isoform_cluster, proteins, redundancy, n_obs_each_redundancy) #%>% 
  # group_by(condition, replicate, reference, isoform_cluster) %>%
  # mutate(distinct_redundancy = n_distinct(redundancy)) %>%
  # ungroup() #these three lines were just a QC for checking redundancy was successfully filtered to one instance per isoform.


global_isoform_shiny <- ascore_final_gene %>% 
  mutate(gene_ref = str_replace_all(gene_ref, " ", "_")) %>% 
  left_join(most_common_protein_redundancy) %>% 
  left_join(fasta_gene_names %>% distinct(entry, gene_names) %>% rename(reference = entry)) %>% 
  rename(gene_redundancy = gene_names) %>% 
  mutate(redundancy = as.numeric(redundancy),
         ambiguous = case_when(redundancy == 0 ~ "single protein assignment",
                               redundancy >=1 ~ paste("ambiguous protein assignment,", redundancy +1, "proteins", sep = " ")))
  
  
write_csv(x = global_isoform_shiny, file = "modified_data/DDA_GP/EGF_stim/GLOBAL_ISOFORM_input.csv")
```


```{r}
# write_csv(x = ascore_final_gene, file = "modified_data/DDA_GP/EGF_stim/GPisoform_SHINY_input.csv")
```



##_b) EGFR pSTY-isoforms
```{r}
pY_boxplot_points <- ggplot() + 
  # pY_boxplot_points <- ggplot(data = ascore_final_gene %>% filter(gene %in% c("EGFR", "MAPK1", "STAT3"))) +
  geom_boxplot(data = ascore_final_gene %>% filter(gene %in% c("EGFR")),
               mapping = aes(x = condition, y = median_norm_intensity), show.legend = FALSE, alpha = 0.3, outlier.color = NA) +
  geom_point(data = ascore_final_gene %>% filter(gene %in% c("EGFR")),
             mapping = aes(x = condition, y = median_norm_intensity, fill = sample_id), shape = 21, alpha = 0.6, show.legend = FALSE) +
  geom_text(data = ascore_final_gene %>% filter(gene %in% c("EGFR")) %>% distinct(condition, gene_ref, n_obs),
             aes(x = condition, y = 26, label = n_obs),size = 2.5, show.legend = FALSE, fontface = "bold") +
  alexis_theme() +
  facet_wrap(facets = vars(gene_ref), nrow = 3)

pY_boxplot_points


ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/pY_boxplot_points_EGFR.png", plot = pY_boxplot_points, height = 12, width = 20, scale = 0.4)
ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/pY_boxplot_points_EGFR.pdf", plot = pY_boxplot_points, height = 12, width = 20, scale = 0.4)
```
##_c) MAPK1 pSTY-isoforms
```{r}
pY_boxplot_points_MAPK1 <- ggplot() + 
  # pY_boxplot_points <- ggplot(data = ascore_final_gene %>% filter(gene %in% c("EGFR", "MAPK1", "STAT3"))) +
  geom_boxplot(data = ascore_final_gene %>% filter(gene %in% c("MAPK1")), 
               mapping = aes(x = condition, y = median_norm_intensity,), show.legend = FALSE, alpha = 0.3, outlier.color = NA) +
  geom_point(data = ascore_final_gene %>% filter(gene %in% c("MAPK1")),
             mapping = aes(x = condition, y = median_norm_intensity, fill = sample_id), shape = 21, alpha = 0.6, show.legend = FALSE) +
  
  alexis_theme() +
  theme(strip.text = element_text(size = 14)) +
  facet_wrap(facets = vars(gene_ref), nrow = 3) +
  geom_text(data = ascore_final_gene %>% filter(gene %in% c("MAPK1")) %>% distinct(condition, gene_ref, n_obs),
             aes(x = condition, y = 29, label = n_obs),size = 2.5, show.legend = FALSE, fontface = "bold") 

pY_boxplot_points_MAPK1


ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/pY_boxplot_points_MAPK1.png", plot = pY_boxplot_points_MAPK1, height = 8, width = 6, scale = 0.4)
```
AHA! so I can detect the dual phospho of MAPK1 (Thr and Tyr). The differential abundance came out as insignificant, but when looking at boxplot of intensities for these isoforms, we see an expected intensity pattern! :) 


##_d) function to iterate through my data
```{r}
boxplot_gene_psite_function <- function(vector_gene_refs, dataframe, output_dir = "output/DDA_GP/EGF_stim/", condition = condition, sample_id = sample_id){

  #create output directory if it doesn't exist already
  if (!dir.exists(output_dir)){
    dir.create(output_dir)}
  
  for (gene_var in vector_gene_refs) {
    if(gene_var %in% dataframe$gene_ref) {
      
      #create boxplot for any sites in each gene
      p <- ggplot() + 
        geom_boxplot(data = dataframe %>% filter(gene_ref == gene_var),
               mapping = aes(x = condition, y = median_norm_intensity), show.legend = FALSE, alpha = 0.3, outlier.color = NA, width = 0.5) +
        geom_point(data = dataframe %>% filter(gene_ref == gene_var),
                   mapping = aes(x = condition, y = median_norm_intensity, fill = sample_id), position = position_jitter( width = 0.3, height = 0), shape = 21, alpha = 0.4, size = 2, show.legend = FALSE) +
        geom_text(data = dataframe %>%
        filter(gene_ref == gene_var) %>%
        distinct(condition, gene_ref, n_obs),
        aes(x = condition, y = 34, label = n_obs),size = 4, show.legend = FALSE, fontface = "bold") +
        alexis_theme() +
        theme(strip.text = element_text(size = 14)) +
        # theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) +
        facet_wrap(facets = vars(gene_ref), nrow = 4) +
        scale_y_continuous( expand = c(0,0))  +
        expand_limits(y = c(10, 35)) 
      
      # save the plot as PNG & PDF
      ggsave(filename = file.path(output_dir, paste0(gene_var, "_boxplot_points.png")), plot = p, width = 6, height = 8, scale = 0.4)
      # ggsave(filename = file.path(output_dir, paste0(gene_var, "_boxplot_points.pdf")), plot = p, width = 6, height = 8, scale = 0.4)
      
    } else { 
      message (paste("Gene", gene_var, "not found in the dataframe."))
      
      }
    }
  }
```


###__i) run function on all psites
```{r}
boxplot_gene_psite_function(vector_gene_refs = (ascore_final_gene %>% distinct(gene_ref))$gene_ref, dataframe = ascore_final_gene %>% distinct(gene, gene_ref, condition, replicate, sample_id, median_norm_intensity, n_obs) %>% mutate(condition = fct_relevel(condition, "EGF0min", "EGF1min", "EGF3min", "EGF5min", "EGF15min")), output_dir = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/boxplot_points_scaled/")
```


#6. PCA
###__i) scree
```{r}
PCA_df <- ascore_fasta_protein_mod_loc_sum_isoform %>%
  # filter(!is.na(missingness)) %>% 
  filter(median_norm_intensity != -Inf)


all_isoform_scree  <- qc_pca(
  data = PCA_df,
  sample = sample_id,
  grouping = isoform_cluster,
  intensity = median_norm_intensity,
  condition = condition,
  digestion = NULL,
  plot_style = "scree")

all_isoform_scree
```

###__ii) PC1 vs. PC2
```{r}
all_isoform_PC_1_2  <- qc_pca(
  data = PCA_df,
  sample = sample_id,
  grouping = isoform_cluster,
  intensity = median_norm_intensity,
  condition = condition,
  digestion = NULL,
  plot_style = "pca",
  components = c("PC1", "PC2"))

all_isoform_PC_1_2

ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/all_isoform_PC_1_2.png", plot = all_isoform_PC_1_2, width = 24, height = 12, scale = 0.4)
```
that is cool grouping! I wonder if MS run order impacted the outliers of EGF0min rep3, EGF1min rep2, EGF3min rep2, EGF15min rep1? NEed to paint PCA by well number and therefore run order. See if these were the runs following washes. 
EGF1min rep2 = first run of my samples, immediately after wash, std, wash

(other post wash samples are not outliers, i.e. 15minrep5, 5minrep3, 3minrep6, 15minrep2)
###__iii) PC1 vs. PC3
```{r}
all_isoform_PC_1_3  <- qc_pca(
  data = PCA_df,
  sample = sample_id,
  grouping = isoform_cluster,
  intensity = median_norm_intensity,
  condition = condition,
  digestion = NULL,
  plot_style = "pca",
  components = c("PC1", "PC3"))

all_isoform_PC_1_3

ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/all_isoform_PC_1_3.png", plot = all_isoform_PC_1_3, width = 24, height = 12, scale = 0.4)
```



#5. PHOSPHO SIGNALING PATHWAY OVERLAPS
##EGFR
```{r}
# ascore_EGFR1_PTMSEA <- ascore_fasta_protein_mod_loc_sum_isoform %>%
#   separate(isoform_cluster, into = c("reference2", "isoform"), sep = "_", remove = FALSE) %>% 
#   select(-reference2) %>% 
#   left_join(y = fasta_gene_names %>% rename(reference = entry) %>% select(reference, gene, protein_names, keyword_id), by = "reference") %>% 
#   full_join(y = EGFR1_pathway , by = c("gene", "reference", "mod_res", "isoform"), relationship = "many-to-many") %>% 
# mutate(PTM_SEA = case_when(
#   is.na(PTM_SEA) ~ FALSE, 
#   TRUE ~ TRUE)) %>%   
# mutate(
#     overlap = case_when(
#     !is.na(condition) & PTM_SEA == TRUE ~ "both",
#     PTM_SEA == TRUE ~ "literature",
#     PTM_SEA == FALSE  ~ "my data")) %>%
#   distinct(overlap, gene, isoform, mod_res)
  
```

##--pSTY isoforms in EGFR annotated pathway from PTMSEA
```{r}
# plot_PTMSEA_EGFR_pathway_modres <- ggplot(data = EGFR1_pathway %>% distinct(gene, isoform, mod_res)) +
#   geom_bar(mapping = aes(x = mod_res, fill = mod_res), color = "black", size = 0.5, show.legend = FALSE) +
#   alexis_theme() +
#   theme(axis.text.x = element_text(angle = 0, size = 12, hjust = 0.5)) +
#   scale_fill_viridis_d() +
#   ggtitle(label = "PTM-SEA: EGFR pathway \np-site distribution") +
#   ylab("unique gene + p-site")
#   
# 
# plot_PTMSEA_EGFR_pathway_modres
# 
# ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/literature_PTMSEA_EGFR_pathway_modres_distribution.png", plot = plot_PTMSEA_EGFR_pathway_modres, width = 6, height = 6, scale = 0.4)
```


###---overlaps by protein + site
```{r}
# EGFR1_overlaps <- ascore_EGFR1_PTMSEA %>% 
#   distinct(overlap, gene, isoform, mod_res)
# 
# #plot overlaps count of pisoforms in my data vs. PTMSEA EGFR1 pathway
# EGFR1_isoform_overlaps <- ggplot(data = EGFR1_overlaps %>% filter(mod_res == "Y")) +
#   geom_bar(mapping = aes(x = overlap, fill = mod_res), color = "black", size = 0.5) +
#   scale_fill_viridis_d(direction = -1) +
#   alexis_theme() +
#   scale_y_continuous(expand = c(0,0)) +
#   # expand_limits(y = c(0, 500)) +
#   theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) +
#   ylab("unqiue p-isoforms")
# 
# 
# EGFR1_isoform_overlaps
# 
# 
# ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/EGFR1_isoform_overlaps_pY.png", plot = EGFR1_isoform_overlaps,
#        scale = 0.4, width = 8, height = 6)

```
still some pY isoforms to identify that we could not identify with just global phospho enrichment alone.

###---overlaps by protein
```{r}

ascore_EGFR1_PTMSEA_protein <- ascore_fasta_protein_mod_loc_sum_isoform %>% 
  left_join(y = fasta_gene_names %>% rename(reference = entry) %>% select(reference, gene, protein_names, keyword_id), by = "reference") %>%
  full_join(y = EGFR1_pathway %>% select(gene, PTM_SEA) , by = c("gene"), relationship = "many-to-many") %>%
  
  mutate(PTM_SEA = case_when(
  is.na(PTM_SEA) ~ FALSE, 
  TRUE ~ TRUE)) %>% 
  
  mutate(overlap = case_when(
    !is.na(condition) & PTM_SEA == TRUE ~ "both",
    is.na(condition) & PTM_SEA == TRUE ~ "literature",
    !is.na(condition) & PTM_SEA == FALSE  ~ "my data"))

EGFR1_overlaps_protein <- ascore_EGFR1_PTMSEA_protein %>% 
  distinct(overlap, gene)

#plot overlaps count of pisoforms in my data vs. PTMSEA EGFR1 pathway
EGFR1_overlaps_protein <- ggplot(data = EGFR1_overlaps_protein) +
  geom_bar(mapping = aes(x = overlap)) +
  alexis_theme() +
  scale_y_continuous(expand = c(0,0)) +
  expand_limits(y = c(0, 4000)) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) +
  ylab("unqiue p-proteins")


EGFR1_overlaps_protein


ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/EGFR1_overlaps_protein.png", plot = EGFR1_overlaps_protein,
       scale = 0.4, width = 6, height = 8)

#plot
# plot_intensity_pisoforms_EGFR1 <- ggplot(data = EGFR1_overlaps) +
#   geom_boxplot(mapping = aes())
```
###---overlaps by protein + site
```{r}
# EGFR1_overlaps <- ascore_EGFR1_PTMSEA %>% 
#   distinct(overlap, gene, isoform, mod_res)
# 
# #plot overlaps count of pisoforms in my data vs. PTMSEA EGFR1 pathway
# EGFR1_isoform_overlaps <- ggplot(data = EGFR1_overlaps) +
#   geom_bar(mapping = aes(x = overlap, fill = mod_res)) +
#   scale_fill_viridis_d() +
#   alexis_theme() +
#   scale_y_continuous(expand = c(0,0)) +
#   # expand_limits(y = c(0, 12000)) +
#   theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) +
#   ylab("unqiue p-isoforms")
# EGFR1_isoform_overlaps
# ggsave(filename = "output/MainFig3to6_SuppFig9to15/GlobalPhosphoIsoformLevel/EGFR1_isoform_overlaps.png", plot = EGFR1_isoform_overlaps,
#        scale = 0.4, width = 6, height = 8)
# 
# #plot
# plot_intensity_pisoforms_EGFR1 <- ggplot(data = EGFR1_overlaps) +
#   geom_boxplot(mapping = aes())
```


#6. PTM NAVIGATOR
##__a) input

###---write_csv for PTMNavigator

```{r}
# isoform_diff_abundance_gene %>%
#   filter(comparison %in% c("EGF0min_vs_EGF15min", "EGF5min_vs_EGF0min", "EGF3min_vs_EGF0min", "EGF1min_vs_EGF0min")) %>%
#   
#   #flip the 0 min and 15 min comparison
#   mutate(diff = case_when(
#     comparison == "EGF0min_vs_EGF15min" ~ -diff,
#     TRUE ~ diff)) %>% 
#   mutate(comparison = case_when(
#     comparison == "EGF0min_vs_EGF15min" ~ "EGF15min_vs_EGF0min", 
#     TRUE ~ comparison)) %>% 
```

```{r}
# PTMNav_isoform_input <- isoform_diff_abundance_gene %>%
#   
#   #select only comparisons relative to 0 min:
#   filter(comparison %in% c("EGF0min_vs_EGF15min",
#                            "EGF5min_vs_EGF0min",
#                            "EGF3min_vs_EGF0min",
#                            "EGF1min_vs_EGF0min")) %>%
#   
#   #flip direction of fold change and label for 15 min vs. 0 min:
#   mutate(diff = case_when(
#     comparison == "EGF0min_vs_EGF15min" ~ -diff,
#     TRUE ~ diff)) %>% 
#   mutate(comparison = case_when(
#     comparison == "EGF0min_vs_EGF15min" ~ "EGF15min_vs_EGF0min", 
#     TRUE ~ comparison)) %>% 
#   
#   distinct(reference, modres_position, comparison, diff, adj_pval) %>%
#   filter(!is.na(reference)) %>% 
#   mutate(
#     regulation = case_when(
#       adj_pval <= 0.05 & diff > 0 ~ "up",
#       adj_pval <= 0.05 & diff < 0 ~ "down",
#       adj_pval > 0.05 ~ "") )
# 
# 
# colnames(PTMNav_isoform_input) <- c("Protein IDs", "p-site", "Experiment", "Fold change", "adjusted pvalue", "Regulation")
# 
# write_csv(x = PTMNav_isoform_input, file = "modified_data/DDA_GP/PTMNav_isoform_input.csv")
```



