---
title: "BeadAge_4months"
author: "Alexis Chang"
date: "2024-04-25"
output: html_document
---


##Notes:
samples: 
x09013	4 mo.	rep 1	
x09014	new	rep 1	
x09018	new	rep 3	
x09019	4 mo.	rep 3	
x09020	4 mo.	rep 2	
x09021	new	rep 2

#0. GLOBAL THEME
alexis_theme()
```{r}
alexis_theme <- function() {
  theme(
    # panel.border = element_rect(colour = "blue", fill = NA, linetype = 2),
    panel.border = element_blank(),
    panel.background = element_blank(),
    panel.grid.major.x  = element_blank(),
    panel.grid.minor = element_blank(),
    # axis.text.x = element_text(angle = 90, vjust = 0.25, hjust = 1),
    axis.text.x = element_text(angle = -60, vjust = 0.9, hjust = 0.1),
    axis.line = element_line(colour = "black"),
    axis.text = element_text(colour = "black", face = "plain", family = "sans", size = 14),
    axis.title = element_text(colour = "black", family = "sans", size = 16),
    axis.ticks = element_line(colour = "black"),
    plot.title = element_text(size=10),
    # legend at the bottom 6)
    legend.position = "right")   
}
```

##_b) CVs
```{r}
my_qc_cvs <- function (data,
                       grouping,
                       condition,
                       intensity,
                       plot = TRUE,
                       plot_style = "density",
                       max_cv = 200,
                       xlab = "condition",
                       showlegend = FALSE,
                       yintercept = 25){
  
  
  protti_colours <- "placeholder"
  utils::data("protti_colours", envir = environment())
  
  
#-----------------------------------------------------------------------------------  
  if (plot == FALSE) {
    
    
    if (max(dplyr::pull(data, {{intensity}}), na.rm = TRUE) < 1000) {
      stop(strwrap("Please backtransform your data or use raw values.\nThe function does not handle log2 transformed data.", 
                   prefix = "\n", initial = ""))  }
    
    
    result <- data %>% dplyr::distinct({{grouping}}, {{condition}}, {{intensity}}) %>%
      tidyr::drop_na({{intensity}}) %>%
      dplyr::group_by({{grouping}}) %>%
      dplyr::mutate(cv_combined = (stats::sd({{intensity}})/mean({{intensity}})) * 100) %>%
      dplyr::group_by({{condition}}, {{grouping}}) %>%
      dplyr::mutate(cv = (stats::sd({{intensity}})/mean({{intensity}})) * 100) %>%
      dplyr::distinct({{condition}}, {{grouping}}, .data$cv_combined, .data$cv) %>%
      tidyr::drop_na() %>%
      dplyr::group_by({{condition}}) %>%
      dplyr::mutate(median_cv = stats::median(.data$cv)) %>% 
      dplyr::ungroup() %>%
      dplyr::mutate(median_cv_combined = stats::median(.data$cv_combined)) %>% 
      dplyr::select(-{{grouping}}, -c("cv_combined", "cv")) %>%
      dplyr::distinct() 
    return(result)  }  
  
  ##-----------------------------------------------------------------------------------   
  if (plot == TRUE) {
    if (max(dplyr::pull(data, {{intensity}}), na.rm = TRUE) < 1000) {
      stop(strwrap("Please backtransform your data or use raw values.\nThe function does not handle log2 transformed data.", 
                   prefix = "\n", initial = ""))}
    
    result <- data %>%
      dplyr::distinct({{grouping}}, {{condition}}, {{intensity}}) %>%
      tidyr::drop_na({{intensity}}) %>%
      
      dplyr::group_by({{grouping}}) %>%
      dplyr::mutate(
        cv_combined = (stats::sd({{intensity}})/mean({{intensity}})) * 100) %>%
      
      
      dplyr::group_by({{condition}}, {{grouping}}) %>%
      dplyr::mutate(cv = (stats::sd({{intensity}})/mean({{intensity}})) * 100) %>%
      dplyr::ungroup() %>%
      
      dplyr::distinct({{condition}}, {{grouping}}, .data$cv_combined, .data$cv) %>%
      tidyr::drop_na() %>%
      tidyr::pivot_longer(cols = starts_with("cv"), names_to = "type", values_to = "values") %>%
      dplyr::mutate(type = ifelse(.data$type =="cv", {{condition}}, "all")) %>%
      dplyr::mutate(type = forcats::fct_relevel(as.factor(.data$type),"all")) %>%
      dplyr::select(-{{condition}}) %>%
      dplyr::group_by(.data$type) %>%
      dplyr::mutate(median = stats::median(.data$values)) %>% 
      dplyr::distinct()
    
    if (max(result$values) > max_cv) {
      cv_too_high <- result %>%
        dplyr::filter(.data$values >max_cv) %>%
        nrow()
      
      warning(paste(cv_too_high), " values were exluded from the plot (CV > ",max_cv, " %)")  }
    
    
    
    ##-----------------------------------------------------------------------------------      
    if (plot_style == "boxplot") {
      plot <- ggplot2::ggplot(result) +
        ggplot2::geom_boxplot(aes(x = .data$type, y = .data$values, fill = .data$type), na.rm = TRUE, size = 1, show.legend = showlegend,alpha = 0.5, outliers = FALSE, width = 0.5) +
                              # alpha   = 0.5, outlier.alpha = 0.5, outlier.shape = 21) + 
        ggplot2::labs(
          # title = "Coefficients of variation",
          y = "Coefficient of variation [%]", fill = "Condition") + 
        ggplot2::geom_hline(yintercept = {{yintercept}}, linetype = 1, size = 0.75, alpha = 0.3) +
        ggplot2::scale_y_continuous(limits = c(0, max_cv)) + 
        # scale_fill_brewer(palette = "Dark2") +
        scale_color_manual(values = c("black","black","black","black")) +
  scale_fill_manual(values = c(rep("grey100", 1), rep("grey90", 1), rep("grey70", 1),rep("grey50",1),rep("grey30", 1))) +
        # ggplot2::scale_fill_manual(values = c("grey",protti_colours)) +
        alexis_theme() +
        xlab({{xlab}})
        # ggplot2::theme(plot.title = ggplot2::element_text(size = 20),
        #                axis.title.x = ggplot2::element_text(size = 15), 
        #                axis.text.y = ggplot2::element_text(size = 15),
        #                axis.text.x = ggplot2::element_text(size = 12,angle = 75, hjust = 1),
        #                axis.title.y = ggplot2::element_text(size = 15), 
        #                legend.title = ggplot2::element_text(size = 15),
        #                legend.text = ggplot2::element_text(size = 15))
      return(plot)
    }
    
    ##-----------------------------------------------------------------------------------    
    if (plot_style == "density") {
      plot <- ggplot2::ggplot(result) +
        ggplot2::geom_density(ggplot2::aes(x = .data$values, col = .data$type), size = 1, na.rm = TRUE, show.legend = showlegend) + 
        ggplot2::labs(
          # title = "Coefficients of variation",
          x = "Coefficient of variation [%]", y = "Density", color = "Condition") +
        ggplot2::scale_x_continuous(limits = c(0,max_cv)) +
        geom_vline(data = dplyr::distinct(result,  .data$median, .data$type),
                   ggplot2::aes(xintercept = median, col = .data$type),
                   size = 1,
                   linetype = "dashed", 
                   show.legend = FALSE) +
        scale_fill_brewer(palette = "Dark2") +
        # ggplot2::scale_color_manual(values = c("grey",protti_colours)) +
        alexis_theme() +
        xlab(xlab)
        # ggplot2::theme(plot.title = ggplot2::element_text(size = 20), 
        #                axis.title.x = ggplot2::element_text(size = 15),
        #                axis.text.y = ggplot2::element_text(size = 15),
        #                axis.text.x = ggplot2::element_text(size = 12, angle = 75, hjust = 1),
        #                axis.title.y = ggplot2::element_text(size = 15),
        #                legend.title = ggplot2::element_text(size = 15),
        #                legend.text = ggplot2::element_text(size = 15))
      
      return(plot)
      
    }
    
    ##-----------------------------------------------------------------------------------
    if (plot_style == "violin") {
      
      plot <- ggplot2::ggplot(result, aes(x = .data$type, 
                                          y = .data$values, color = .data$type)) +
        ggplot2::geom_violin(na.rm = TRUE, size = 1) + 
        ggplot2::geom_boxplot(width = 0.15, fill = "white", na.rm = TRUE, alpha = 0.6, size = 0.75, outlier.color = NA) +
        ggplot2::labs(
          # title = "Coefficients of variation",
                      x = "", y = "Coefficient of variation [%]",
                      fill = "Condition") +
        ggplot2::geom_hline(yintercept = {{yintercept}}, linetype = 1, size = 0.75, alpha = 0.3) +
        ggplot2::scale_y_continuous(limits = c(0, max_cv)) +
        # ggplot2::scale_fill_manual(values = c("grey",  protti_colours)) +
        # scale_fill_brewer(palette = "Dark2") +
        scale_color_manual(values = c("black","black","black","black")) +
  scale_fill_manual(values = c(rep("grey90", 1), rep("grey70", 1),rep("grey50",1),rep("grey30", 1))) +
        alexis_theme() +
        xlab(xlab)
        # ggplot2::theme_bw() +
        # ggplot2::theme(plot.title = ggplot2::element_text(size = 20),
        #                axis.title.x = ggplot2::element_text(size = 15),
        #                axis.text.y = ggplot2::element_text(size = 15), 
        #                axis.text.x = ggplot2::element_text(size = 12, angle = 75, hjust = 1), axis.title.y = ggplot2::element_text(size = 15),
        #                legend.title = ggplot2::element_text(size = 15), 
        #                legend.text = ggplot2::element_text(size = 15))
      
      return(plot)
      
    }
  }
}

```


#1. LOAD
##_a) libraries
```{r}
library(tidyverse)
library(janitor)
library(readr)
library(gridExtra)
library(ggrepel)
library(cowplot)
library(reshape2)
library(eulerr)
library(protti)
# library(diann)
library(iq)
library(RColorBrewer)
#for PCA loadings
library(FactoMineR)

#for protti sample correlation and volcano plots
library(dendextend)
library(pheatmap)
library(seriation)
library(UpSetR)
```


##_b) FASTA references
   -> prepare Yeast + Vsrc FASTA
   Downloaded from Zucchini on 1May2023, but pulled from uniprot by Bianca Ruiz in April 2017. K12 strain, swiss prot (reviewed entries only)
```{r}
Yeast_spurious <- read_csv(file = "raw_data/SuppFig8/Alex_yeast-and-spurious_20210219_forR.csv") %>%
  select(reference, full_protein_sequence) %>%
  mutate(organism = "Yeast") 

```


##_c) your data
###__i) comet
```{r}
##read in your data here, without ascore which filters for only phosphopeptides
comet <- read_csv("raw_data/SuppFig8/BeadAge_comet/result.csv") %>% 
  clean_names() %>%
  separate(sample_name, into = c("exp", "lysate","frxn", "condition", "well_replicate", "inj_vol"), sep = "_", remove = FALSE) %>% 
  mutate(condition = case_when(
    condition == "newHalo" ~ "new",
    condition == "oldHalo" ~ "4 mo.")) %>% 
  mutate(
    well = str_sub(well_replicate, start = 1L, end = 3L),
    replicate = as.numeric(str_sub(well_replicate, start = -1L)), 
    replicate = case_when(
      is.na(replicate) ~ 0,
      TRUE ~ as.numeric(replicate))) %>% 
  left_join(y = Yeast_spurious, by = c("reference")) %>% 
  
  ##rename column to 'intensity'
  rename(intensity = max_intensity_light_c2837) %>% 
  
  ##keep only forward hits
  filter(reverse == FALSE) %>% 
  
  ##classify phospho vs. no none
  mutate(phospho = case_when(
    grepl("\\@", sequence) ~ "phospho",
    TRUE~"none")) %>% 
  filter(replicate != 4) #dropped due to precipitate in old, and balanced replicates


  
```

```{r}
sample_key <- comet %>% 
  distinct(condition, replicate, raw_file_name)
```




###__i) ascore
```{r}
ascore <- read_csv("raw_data/SuppFig8/BeadAge_ascore/result.csv") %>% 
  clean_names() %>%
  separate(sample_name, into = c("exp", "lysate","frxn", "condition", "well_replicate", "inj_vol"), sep = "_", remove = FALSE) %>% 
  
  mutate(condition = case_when(
    condition == "newHalo" ~ "new",
    condition == "oldHalo" ~ "4 mo.")) %>%
  mutate(
    well = str_sub(well_replicate, start = 1L, end = 3L),
    replicate = as.numeric(str_sub(well_replicate, start = -1L)), 
    replicate = case_when(
      is.na(replicate) ~ 0,
      TRUE ~ as.numeric(replicate))) %>% 
  
  mutate(
    reference = case_when(
      reference == "P63185-V394A" ~ "P63185",
      reference == "P04049-D468N" ~ "P04049",
      TRUE ~ as.character(reference))) %>% 
  
  
  left_join(y = Yeast_spurious, by = c("reference")) %>% 
  
  ##rename column to 'intensity'
  rename(intensity = max_intensity_light_c2837) %>% 
  
  ##keep only forward hits
  filter(reverse == FALSE) %>% 
  
  ##classify phospho vs. no none
  mutate(phospho = case_when(
    grepl("\\@", sequence) ~ "phospho",
    TRUE~"none")) %>% 
  filter(replicate != 4) #dropped due to precipitate in old, and balanced replicates
  

sample_key <- ascore %>% distinct(raw_file_name, condition, replicate)

sample_key
```



#2. COMET QC
##_a) ENRICHMENT EFFICIENCY
###__i) counts
df
```{r}
enrichment_efficiency_count_df <- comet %>% 
  distinct( condition, replicate,sequence, phospho) %>% 
  group_by( condition, replicate, phospho) %>% 
  summarize(
    num_phospho = n()) %>% 
  pivot_wider(names_from = phospho, values_from = num_phospho) %>% 
  mutate(
    total_pept = none + phospho,
    individual_phosphopept_enrich_efficiency = phospho / total_pept) %>% 
  ungroup() %>% 
  group_by( condition) %>% 
  mutate(
    avg_phosphopeptide_enrichment_efficiency = mean(individual_phosphopept_enrich_efficiency)) %>% 
  ungroup() %>% 
  mutate(
    sample_id = paste(condition, replicate, sep = "_") )


##view df of phosphopeptide enrichment efficiency ratios
enrichment_efficiency_count_df
```


plot
```{r}
##relative to intensity of all peptides (with and without phospho)
set.seed(690)
plot_ratio_phos_count <- ggplot() + 
  geom_bar(data = (enrichment_efficiency_count_df %>%
                     distinct(condition, avg_phosphopeptide_enrichment_efficiency)),
           mapping = aes(x = condition, y = avg_phosphopeptide_enrichment_efficiency),
           stat = "identity",
           position = "dodge",
           fill = "grey30",
           alpha = 0.3, width = 0.8) +
  geom_point(data = enrichment_efficiency_count_df, mapping = aes(x = condition,
                 y = individual_phosphopept_enrich_efficiency,
                 fill = replicate),
             position = position_jitterdodge(dodge.width = 0.25), alpha = 1, size =1, shape = 1, show.legend =  FALSE)  +

  ylab("p- /all peptide (intensity)") +
  xlab ("bead age") +
  alexis_theme()+
  scale_y_continuous(expand = c(0,0), limits = c(0, 1.02), breaks = c(seq(0, 1, 0.25))) + #removes whitespace below bars!
  theme(axis.text.x = element_text(angle = -60, vjust = 0.1, hjust = 0.15, size = 10),
        axis.title = element_text(hjust = 0.5, size = 10))
  

plot_ratio_phos_count
ggsave("output/SuppFig8/plot_ratio_phos_count_ratio_style2.png", plot = plot_ratio_phos_count, width = 3, height = 6, scale = 0.4)
ggsave("output/SuppFig8/plot_ratio_phos_count_ratio_style2.pdf", plot = plot_ratio_phos_count, width = 4, height = 6, scale = 0.4)
```

###__ii) intensity
df
```{r}
enrichment_efficiency_intensity_df <- comet %>% 
  distinct(sequence,  condition, replicate, phospho, intensity) %>% 
  group_by(sequence,  condition, replicate, phospho) %>% 
  
##filter by max intensity for every peptide
  filter(intensity  == max(intensity)) %>% 
  ungroup() %>% 
  group_by(phospho, condition, replicate) %>% 
  
##summarize intensity of all peptides and just p-peptides
  summarise(all_peptides_intensity = sum(intensity)) %>% 
  pivot_wider(values_from = all_peptides_intensity, names_from = phospho) %>% 
  
##grouped mutate for taking mean of summed intensities for each replicate within each condition  
  group_by(condition) %>%   
  mutate(mean_phos = mean(phospho),
         mean_non = mean(none)) %>% 
  ungroup() %>% 
  mutate(mean_ratio = mean_phos / (mean_non + mean_phos),
         rep_ratio = phospho / (none + phospho)) %>% 
  ungroup() %>% 
  mutate(
    sample_id = paste(condition, replicate, sep = "_") )
  
  
  enrichment_efficiency_intensity_df
```





plot
```{r}
##relative to intensity of all peptides (with and without phospho)
set.seed(690)
plot_ratio_phos_intensity <- ggplot(data = (enrichment_efficiency_intensity_df)) + 
  geom_bar(aes(x = condition, y = mean_ratio),
           stat = "identity",
           position = "dodge",
           fill = "skyblue2",
           alpha = 0.3, width = 0.8) +
  geom_point(aes(x = condition,
                 y = rep_ratio,
                 fill = replicate),
             position = position_jitterdodge(dodge.width = 0.25), alpha = 1, size =1, shape = 1, show.legend =  FALSE)  +
  ylab("p- /all peptide (intensity)") +
  xlab ("bead age") +
  alexis_theme()+
  scale_y_continuous(expand = c(0,0), limits = c(0, 1.02), breaks = c(seq(0, 1, 0.25))) + #removes whitespace below bars!
  theme(axis.text.x = element_text(angle = -60, vjust = 0.1, hjust = 0.15, size = 10),
        axis.title = element_text(hjust = 0.5, size = 10))
  

plot_ratio_phos_intensity
ggsave("output/SuppFig8/phospho_pept_intensity_ratio_style2.png", plot = plot_ratio_phos_intensity, width = 3, height = 6, scale = 0.4)
ggsave("output/SuppFig8/phospho_pept_intensity_ratio_style2.pdf", plot = plot_ratio_phos_intensity, width = 4, height = 6, scale = 0.4)
```


##_b)  OVERALL ITENSITY
```{r}
plot_psm_intensity_distribution_reps <- ggplot(data = comet) +
  geom_boxplot(mapping = aes(x = condition, y = log2(intensity), fill = as.factor(replicate)),
               outlier.shape = 21,
               outlier.alpha = 0.3,
               outlier.size = 0.4) +
  alexis_theme() +
  scale_fill_brewer(palette = "Greys", name = "replicate") +
  ylab(expression(log[2]~(intensity))) + xlab("bead age") +
  theme(axis.title = element_text(hjust = 0.5, vjust = 1,  family = "sans", size = 14),
    axis.text.x = element_text(angle=0, vjust = 1, hjust = 0.5, family = "sans", size = 14),
        axis.text.y = element_text(vjust = 0.5, hjust = 1, family = "sans", size = 12),
        legend.title = element_text(hjust = 0.5, face = "bold", family = "sans")) +
  theme(axis.text.x = element_text(angle = -60, vjust = 0.1, hjust = 0.15, size = 12),
        axis.title = element_text(hjust = 0.5, size = 12)) +
  ggtitle("Intensity of all PSMs")

plot_psm_intensity_distribution_reps


ggsave("output/SuppFig8/all_psm_intensity_distribution_reps.png", plot = plot_psm_intensity_distribution_reps, width = 6, height = 7, scale = 0.4)
ggsave("output/SuppFig8/all_psm_intensity_distribution_reps.pdf", plot = plot_psm_intensity_distribution_reps, width = 6, height = 7, scale = 0.4)
```

#-----------------------------------------------

#3. ASCORE
##_a) parse p-site localization
Working from 'Ascore Extended' output from Zucchini. Avoids relying on "mod_location" from CometExtended for phospho localization.

###__i) index to peptide
```{r}
ascore_peptide_index <- ascore %>% 
  distinct() %>%
  
  mutate(seq_no_mods = str_remove_all(ascore_sequence, "[:punct:]"),
         seq_no_mods = str_remove_all(seq_no_mods, "[n]")) %>% ##remove n from n## mark of nAc modification. 
  mutate_at(c("a_score_1", "a_score_2", "a_score_3", "position_1", "position_2", "position_3"), funs(as.numeric)) %>% 
  mutate_at(c("a_score_1", "a_score_2", "a_score_3", "position_1", "position_2", "position_3"), ~replace_na(., 0)) %>%

  select(sample_name,  condition, replicate,  reference,
         ascore_sequence, num_sites, a_score_1, a_score_2, a_score_3, position_1, position_2, position_3,
         seq_no_mods, intensity, q_score_c2837, num_scans_light_c2837, charge, x_corr, 
         missed_cleavages, num_sites, redundancy) %>% 
 
##extract modified residue
  mutate(
    mod_res1 = str_sub(seq_no_mods, start = position_1+1L, end = position_1 + 1L),
    mod_res2 = str_sub(seq_no_mods, start = position_2+1L, end = position_2 + 1L),
    mod_res3 = str_sub(seq_no_mods, start = position_3+1L, end = position_3 + 1L)) %>% 


## pivot wider to match each p-site to an ascore and filter for STY

  unite(ascore_mod_res1, c(a_score_1, mod_res1, position_1), sep = "_") %>% 
  unite(ascore_mod_res2, c(a_score_2, mod_res2, position_2), sep = "_") %>%
  unite(ascore_mod_res3, c(a_score_3, mod_res3, position_3), sep = "_") %>%
  pivot_longer(cols = c("ascore_mod_res1", "ascore_mod_res2", "ascore_mod_res3")) %>% 
  separate(value, into = c("ascore", "mod_res", "mod_position"), sep = "_") %>% 
  mutate(ascore= as.numeric(ascore))


```

###__ii) filter Ascore > 13
```{r}
ascore_peptide_index_stringent <- ascore_peptide_index %>% 
  filter(ascore >= 13) %>% 
  filter(grepl("[STY]", mod_res) == TRUE)
```

###__iii) index to protein
JOIN TO FASTA here
```{r}
  #join to FASTA
ascore_stringent_fasta <- ascore_peptide_index_stringent %>% 
  left_join(y = Yeast_spurious, by = c("reference")) %>% 
  
  #keep detected proteins only
  filter(!is.na(ascore_sequence)) %>% 
  
  mutate(
    mod_position = as.numeric(mod_position), ##mod_position parsed from ascore output, not Comet_Extended
    z_peptide = str_replace_all(seq_no_mods, "[IL]", "Z"),
    z_protein = str_replace_all(full_protein_sequence, "[IL]", "Z"),
    z_pept_position_in_z_protein = str_locate(z_protein, pattern = z_peptide), ##get peptide position from z-substituted peptide and protein
    mod_position_in_protein = z_pept_position_in_z_protein[,"start"] + mod_position, ##extract mod position from true protein sequence
    test_mod_position = str_sub(full_protein_sequence, start = mod_position_in_protein, end = mod_position_in_protein)) %>% 
  select(test_mod_position, mod_res, mod_position, ascore_sequence, everything()) %>% 
  unite(col = "mod_protein_location", c(mod_res, mod_position_in_protein), sep = "", remove = FALSE) %>% 
  unite(col = "ref", c(reference, mod_protein_location), sep = "_", remove = FALSE)
  





#index to protein (account for equal I and L during search)


```

Check for unmapped phosphosites
```{r}
    ##tally up the number of rows that have NA vs. STY in test_mod_position
mod_pos_group <- ascore_stringent_fasta %>% 
  select(test_mod_position) %>% 
  group_by(test_mod_position) %>% 
   summarize(
    num_phospeptides = n())
mod_pos_group
```

Are all mods are on S, T or Y as hoped? If NAs exist, there is a mapping issue. 
For Yeast spurious, the mutant kinases did map back to my reference database. Just mapped to un-mutated protein instead.


###__iv) collapse PSM to precursor (max intensity)
keep single PSM per precursor
take max intensity PSM per precursor (peptide + charge)
```{r}
ascore_stringent_fasta_precursor <- ascore_stringent_fasta %>% 
  group_by(condition, replicate, ascore_sequence, charge, ascore) %>% 
  filter(intensity == max(intensity)) %>% 
  ungroup()
```

save lists of peptides for supplementary
```{r}
#replicate
write_csv(x = ascore_stringent_fasta_precursor %>% distinct(condition, replicate, ascore_sequence, charge, ascore, intensity, reference, sample_name,   mod_position_in_protein, ref), file = "modified_data/SuppFig8/distinct_pSTYpeptides_ascore13_replicate.csv", col_names = TRUE)


#condition
write_csv(x = ascore_stringent_fasta_precursor %>% distinct(condition,  ascore_sequence, charge, ascore, intensity, reference,   mod_position_in_protein, ref), file = "modified_data/SuppFig8/distinct_pSTYpeptides_ascore13_condition.csv", col_names = TRUE)
```


###__v) collapse precursor to p-site (by sum)
This combines intensities from different phosphoisoform peptides into single site measurements. 
Alternatives could be to differentiate phosphoisoform peptides and sum (i.e. add modifier to ref to indicate # of other sites co-detected)
Alternatively, I could also just take the monophosphorylated peptide if present and if no monophosphorylated peptide, I could average intensities summed to distinct isoforms (as in Krug et al 2019 for PTM SEA)

```{r}
##SUM PRECURSOR INTENSITIES TO P-SITES + median normalize

 ascore_stringent_fasta_psite <- ascore_stringent_fasta_precursor %>% 
  group_by(condition, replicate,  reference, mod_res,  ref) %>% #ref = unique p-site
  
  #sum PSM intensities of confident p-sites to individual p-sites
  mutate(
    sum_intensity_precursor_to_psite = sum(intensity),
    log2_psite_qty = log2(sum_intensity_precursor_to_psite)) %>% 
  ungroup() %>% 
  
  #keep single summed intensity per p-site prior to median normalization
  distinct(condition, replicate, reference, mod_res, ref, sum_intensity_precursor_to_psite, log2_psite_qty) 
  
  
```


###__vi) median normalize p-sites by sample
```{r}
ascore_stringent_fasta_psite <- ascore_stringent_fasta_psite %>% 
  #median normalize intensities summed to p-site
  mutate(
    global_median_intensity = median(log2_psite_qty)) %>% 
  group_by(condition, replicate) %>% 
  mutate(
    sample_median_intensity = median(log2_psite_qty)) %>% 
  ungroup() %>% 
  mutate(
    median_norm_intensity = log2_psite_qty - sample_median_intensity + global_median_intensity,
    raw_median_norm_intensity = 2^median_norm_intensity) %>% 
  mutate(sample_id = paste(condition, replicate, sep = "_"))
```



##_b) count p-sites



###__i) by condition

df
Use dataframe PRIOR TO collapsing PSMs to precursors, precursor to psites and median normalized intensity.
```{r}
distinct_p_sites_condition <- ascore_stringent_fasta %>% 
  distinct(reference, mod_protein_location, mod_res, condition)
##this collapses all  replicates into a single set of unique p sites.
```


write to csv
```{r}
write_csv(x = distinct_p_sites_condition, file = "modified_data/SuppFig8/distinct_pSTYsites_ascore13_condition.csv", col_names = TRUE)
```

plot
```{r}
plot_distinct_p_sites_condition <- ggplot(data = distinct_p_sites_condition) +
  geom_bar(mapping = aes(x = condition, fill = mod_res)) +
  theme_bw(18) +
  scale_fill_viridis_d() +
  ylab("unique phospho sites") #+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.25, hjust = 1))

plot_distinct_p_sites_condition

ggsave("output/SuppFig8/distinct_p_sites_condition.png", plot = plot_distinct_p_sites_condition, width = 16, height = 8, scale = 0.5)
ggsave("output/SuppFig8/distinct_p_sites_condition.pdf", plot = plot_distinct_p_sites_condition, width = 16, height = 8, scale = 0.5)
```


table summary of overall site counts
```{r}
summarize_distinct_psites_condition <- distinct_p_sites_condition %>%
  ungroup() %>% 
  group_by(condition, mod_res) %>% 
  summarize(
    n_psites = n())

summarize_distinct_psites_condition
```


###__ii) by replicate

df
```{r}
distinct_p_sites_reps <- ascore_stringent_fasta %>% 
  ungroup() %>% 
  distinct(reference, mod_protein_location, mod_res,  condition, replicate) %>% 
  mutate(
    sample_id = paste(condition, replicate, sep = "_"))
##this collapses all  replicates into a single set of unique p sites.
```


write to csv
```{r}
write_csv(x = distinct_p_sites_reps, file = "modified_data/SuppFig8/pSTYsites_ascore13_reps.csv", col_names = TRUE)
```

plot
```{r}
plot_distinct_p_sites_reps <- ggplot(data = distinct_p_sites_reps) +
  geom_bar(mapping = aes(x = sample_id, fill = mod_res)) +
  theme_bw(18) +
  scale_fill_viridis_d() +
  ylab("unique phospho sites") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.25, hjust = 1))

plot_distinct_p_sites_reps

ggsave("output/SuppFig8/distinct_p_sites_reps.png", plot = plot_distinct_p_sites_reps, width = 15, height = 15, scale = 0.5)
ggsave("output/SuppFig8/distinct_p_sites_reps.pdf", plot = plot_distinct_p_sites_reps, width = 15, height = 15, scale = 0.5)
```
I think this is on par with what I saw previously, replicates each have less p-sites, while combined conditions using DDA jump total unique p-site counts by ~ 1000.

table
```{r}
summarize_distinct_psites_replicates <- distinct_p_sites_reps %>%
  ungroup() %>% 
  group_by(condition, replicate, sample_id, mod_res) %>% 
  summarize(
    n_psites = n())

summarize_distinct_psites_replicates
```


#4. STYLE 2: ANALYZE ASCORE
##a. N DISTINCT P-SITES 

###__i. total:
```{r}
distinct_p_sites_total <- ascore_stringent_fasta %>% 
  distinct(reference, mod_position_in_protein, mod_res) %>%
  mutate(filter_level = "Ascore >= 13")
```


#### . . . . summarized count for text (PUB)
```{r}
distinct_pY_sites_total_summary <- distinct_p_sites_total %>% 
  group_by(mod_res) %>% 
  summarize(
    n_distinct_sites = n()) %>% 
  ungroup()

distinct_pY_sites_total_summary
```


#### . . . . plot
```{r}
plot_distinct_p_sites_total <- ggplot(data = distinct_p_sites_total) +
  geom_bar(mapping = aes(x = filter_level, fill = mod_res)) +
  theme_bw(18) +
  scale_fill_viridis_d() +
  ylab("unique phospho sites") +
  alexis_theme()+
  
  scale_y_continuous(expand = c(0,0), limits = c(0, 5000), breaks = c(seq(0, 5000, 1000))) +
  theme(axis.text.x = element_text(angle = 0, vjust = 0, hjust = 0.5))

plot_distinct_p_sites_total

ggsave("output/SuppFig8/distinct_p_sites_total.png", plot = plot_distinct_p_sites_total, width = 6, height = 8, scale = 0.4)
ggsave("output/SuppFig8/distinct_p_sites_total.pdf", plot = plot_distinct_p_sites_total, width = 8, height = 8, scale = 0.4)
```


###__ii. by condition:

#### . . . . filter for distinct phospho sites


```{r}
distinct_p_sites_condition <- ascore_stringent_fasta %>% 
  distinct(reference, mod_protein_location, mod_res, condition)
##this collapses all  replicates into a single set of unique p sites.
```

#### . . . . write to csv 
write to csv distinct pSTY sites per condition

```{r}
write_csv(x = distinct_p_sites_condition, file = "modified_data/SuppFig8/pSTYsites_ascore13_condition.csv", col_names = TRUE)
```

#### . . . . plot
plot number of distinct phospho sites 
-look at relative fraction of STY and compare to human ratios.

```{r}
plot_distinct_p_sites_condition <- ggplot(data = distinct_p_sites_condition ) +
  geom_bar(mapping = aes(x = condition, fill = mod_res)) +
  theme_bw(18) +
  scale_fill_viridis_d(name = "mod\nsite") +
  ylab("unique phospho sites") +
  alexis_theme()+
  # expand_limits(y = c(0, 4000)) +
  scale_y_continuous(expand = c(0,0), limits = c(0, 5000), breaks = c(seq(0, 5000, 1000))) +
  xlab("peptide concentration \n (0.5 mg input)")+
  theme(axis.text.x = element_text(angle = -60, vjust = 0.15, hjust = 0.1, ),
        axis.title.x = element_text(size = 15))

plot_distinct_p_sites_condition

ggsave("output/SuppFig8/distinct_p_sites_condition.png", plot = plot_distinct_p_sites_condition, width = 10, height = 12, scale = 0.4)
ggsave("output/SuppFig8/distinct_p_sites_condition.pdf", plot = plot_distinct_p_sites_condition, width = 8, height = 8, scale = 0.4)
```
####. . . . counts
```{r}
counts_distinct_psites_condition <- distinct_p_sites_condition %>% 
  distinct() %>% 
  group_by(condition, mod_res) %>% 
  summarize(
    n_distinct_sites = n()) %>% 
  ungroup() %>% 
  group_by(condition) %>% 
  mutate(total_sites = sum(n_distinct_sites)) %>% 
  ungroup() %>% 
  mutate(
    ratio_pY_sites = n_distinct_sites / total_sites * 100) %>% 
  group_by(mod_res) %>% 
  mutate(
    average_pYsite_percent = mean(ratio_pY_sites) ) %>% 
  ungroup()

counts_distinct_psites_condition
```
###__iii. by replicate

#### . . . . filter for distinct phospho sites

df
```{r}
distinct_p_sites_reps <- ascore_stringent_fasta %>% 
  ungroup() %>% 
  mutate(sample_id = paste(condition, replicate, sep = "\n")) %>% 
  distinct(condition, replicate, sample_id, reference, mod_protein_location, mod_res)
##this collapses all  replicates into a single set of unique p sites.
```


#### . . . . write to csv
```{r}
write_csv(x = distinct_p_sites_reps, file = "modified_data/SuppFig8/pSTYsites_ascore13_reps.csv", col_names = TRUE)
```

#### . . . . plot
```{r}
plot_distinct_p_sites_reps <- ggplot(data = distinct_p_sites_reps) + 
  geom_bar(mapping = aes(x = sample_id, fill = mod_res)) +
  theme_bw(10) +
  scale_fill_viridis_d() +
  ylab("unique phospho sites")  +
  xlab("1 mg peptide input") +
  alexis_theme()+
  scale_y_continuous(expand = c(0,0), limits = c(0, 4000), breaks = c(seq(0, 4000, 1000))) +
  theme(axis.text.x = element_text(angle = -60, vjust = 0.75, hjust = 0))

plot_distinct_p_sites_reps

ggsave("output/SuppFig8/distinct_p_sites_reps.png", plot = plot_distinct_p_sites_reps, width = 8, height = 10, scale = 0.4)
ggsave("output/SuppFig8/distinct_p_sites_reps.pdf", plot = plot_distinct_p_sites_reps, width = 8, height = 10, scale = 0.4)
```


###__iv.table and error bar df 
```{r}
summarize_distinct_psites_replicates <- distinct_p_sites_reps %>%
  ungroup() %>% 
  group_by(condition, replicate, sample_id, mod_res) %>% 
  summarize(
    n_psites = n()) %>% 
  ungroup()

summarize_distinct_psites_replicates



#errorbar ------------------------------------------------------------
df_geom_errorbar <- summarize_distinct_psites_replicates %>%
  group_by(condition, mod_res) %>%
  mutate(
    min_psites = min(n_psites), 
    max_psites = max(n_psites), 
    avg_psites = mean(n_psites)) %>% 
  ungroup() %>% 
  distinct(condition, mod_res, min_psites, max_psites, avg_psites) #could also include mid psites later if needed

df_geom_errorbar
```
###__iii) condition + replicate as dots


plot
```{r}
plot_distinct_p_sites_condition_reps_wPTS <- ggplot() +
  geom_bar(data = distinct_p_sites_condition,
                   mapping = aes(x = condition, fill = mod_res),show.legend = FALSE) +
  
  geom_errorbar(data = (df_geom_errorbar), 
                mapping = aes(x = condition, ymin = min_psites, ymax = max_psites),  show.legend = FALSE, linewidth = 0.5, width = 0.25, color = "grey90") +
  
  
  geom_jitter(data = (distinct_p_sites_reps), 
              mapping = aes(x = condition, color = sample_id, fill = mod_res),
              stat = "count", shape = 1, show.legend = FALSE, size = 1, stroke =0.5,
              width = 0.35, height = 0, alpha = 0.5) +
   
  theme_bw(10) +
  facet_wrap(facets = vars(mod_res), ncol = 3) +
  scale_fill_viridis_d() +
  scale_color_manual(values = rep(c("grey70" ),60)) +
  ylab("unique phospho sites")   +
  alexis_theme()+
  scale_y_continuous(expand = c(0,0), limits = c(0, 4500), breaks = c(seq(0, 4000, 1000)))


plot_distinct_p_sites_condition_reps_wPTS

ggsave("output/SuppFig8/distinct_p_sites_condition_wPTS.png", plot = plot_distinct_p_sites_condition_reps_wPTS, width = 6, height = 6, scale = 0.4)
ggsave("output/SuppFig8/distinct_p_sites_condition_wPTS.pdf", plot = plot_distinct_p_sites_condition_reps_wPTS, width = 6, height = 6, scale = 0.4)
```

####___iv) focus on pY
```{r}
set.seed(1333)
plot_distinct_p_sites_condition_reps_wPTS <- ggplot() +
  geom_bar(data = (distinct_p_sites_condition %>%
                     filter(mod_res == "Y")),
                    
                   mapping = aes(x = condition, fill = mod_res),
           show.legend = FALSE, color = "black", linewidth = 0.5, width = 0.75) +
  
   geom_errorbar(data = (df_geom_errorbar %>%
                           filter(mod_res == "Y")),
                mapping = aes(x = condition, ymin = min_psites, ymax = max_psites),  show.legend = FALSE, linewidth = 0.5, width = 0.2, color = "grey30") +
  
  
  geom_jitter(data = (distinct_p_sites_reps%>%
                        filter(mod_res == "Y")),
              mapping = aes(x = condition, color = sample_id, fill = mod_res),
              stat = "count", shape = 21, show.legend = FALSE, size = 1.2, stroke =0.5, fill = "white",
              width = 0.1, height = 0, alpha = 0.6) +
  geom_text(data = tibble(x = c(1, 2),y = c(4140, 4192)), mapping = aes( x = x, y = y +150, label = y),
            color = "black",
            size = 3,
            fontface = "bold",
            inherit.aes = FALSE) +
 
  scale_fill_viridis_d(direction = -1) +
  scale_color_manual(values = rep(c("grey30" ),60)) +
  ylab("unique pY sites")   +
  xlab("bead age") +
  alexis_theme()+
  scale_y_continuous(expand = c(0,0), limits = c(0, 5000), breaks = c(seq(0, 5000, 1000))) +
  theme(axis.text.x = element_text(angle = -60, vjust = 0.75, hjust = 0, size = 10),
        axis.text.y = element_text( size = 10))

plot_distinct_p_sites_condition_reps_wPTS

ggsave("output/SuppFig8/distinct_pY_sites_condition_wPTS.png", plot = plot_distinct_p_sites_condition_reps_wPTS, width = 4, height = 6, scale = 0.4)
ggsave("output/SuppFig8/distinct_pY_sites_condition_wPTS.pdf", plot = plot_distinct_p_sites_condition_reps_wPTS, width = 4, height = 6, scale = 0.4)
```
#5. pY intensity distributions
```{r}
alexis_theme <- function() {
  theme(
    # panel.border = element_rect(colour = "blue", fill = NA, linetype = 2),
    panel.border = element_blank(),
    panel.background = element_blank(),
    panel.grid.major.x  = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(angle = -60, vjust = 0.9, hjust = 0),
    axis.line = element_line(colour = "black"),
    axis.text = element_text(colour = "black", face = "plain", family = "sans"),
    axis.title = element_text(colour = "black", family = "sans"),
    axis.ticks = element_line(colour = "black"),
    plot.title = element_text(size=10),
    # legend at the bottom 6)
    legend.position = "right")   
}
```


```{r}
#dataframe -------------------------
ascore_stringent_fasta_pYsite <- ascore_stringent_fasta_psite %>%
  ungroup() %>% 
  filter(mod_res == "Y")

#plotting --------------------------
plot_pY_intensity_distribution_reps <- ggplot(data = ascore_stringent_fasta_pYsite %>% ungroup()) +

  geom_boxplot(mapping = aes(x = condition, y = log2_psite_qty, color = as.factor(replicate), fill = condition),
               
               outliers = FALSE,
               show.legend = FALSE,
               alpha = 0.5) +

  scale_color_manual(values = c("black","black","black","black")) +
  scale_fill_manual(values = c(rep("grey80", 1), rep("grey50",1))) +
  alexis_theme() +

  theme( legend.position = "none") +

  scale_y_continuous(expand = c(0,0),
                     breaks = c(12, 16, 20, 24),
                     limits = c(12, 24),
                     labels = c("12", "16", "20", "24")) +#removes whitespace below bars! custom ticks
  ylab(expression(log[2]~(intensity))) +
  xlab("bead age") 

plot_pY_intensity_distribution_reps


ggsave("output/SuppFig8/plot_pY_intensity_distribution_reps.png", plot = plot_pY_intensity_distribution_reps, width = 4, height = 6, scale = 0.4)

ggsave("output/SuppFig8/plot_pY_intensity_distribution_reps.pdf", plot = plot_pY_intensity_distribution_reps, width = 4, height = 6, scale = 0.4)
```

###__ii) pY
####___- not normalized
```{r}
psite_reps_CVs_NOTnorm_pY <- my_qc_cvs(
  data = ascore_stringent_fasta_pYsite, 
  grouping = ref, 
  condition = condition,
  intensity = sum_intensity_precursor_to_psite,
  plot = TRUE, 
  plot_style = "boxplot",
  xlab = expression(paste("bead age")),
  max_cv = 100)

psite_reps_CVs_NOTnorm_pY

ggsave(filename = "output/SuppFig8/CVs_not_normalized_pY.png", plot = psite_reps_CVs_NOTnorm_pY, width = 8, height = 10, scale = 0.4)
ggsave(filename = "output/SuppFig8/CVs_not_normalized_pY.pdf", plot = psite_reps_CVs_NOTnorm_pY, width = 8, height = 10, scale = 0.4)
```
##### _ _ _ Intensity (not normalized)
```{r}
plot_pY_intensity_distribution_reps <- ggplot(data = ascore_stringent_fasta_pYsite) +

  geom_boxplot(mapping = aes(x = condition, y = log2_psite_qty, color = as.factor(replicate), fill = condition),
               outliers = FALSE,
               outlier.shape = 21,
               outlier.alpha = 0.3,
               show.legend = FALSE,
               alpha = 0.5) +
  scale_color_manual(values = c("black","black","black","black")) +
  scale_fill_manual(values = c(rep("grey90", 1), rep("grey70", 1),rep("grey50",1),rep("grey30", 1))) +
  alexis_theme() +
  theme(axis.text.x = element_text(angle=0, hjust = 0.5), legend.position = "none") +
  scale_y_continuous(expand = c(0,0),
                     limits = c(12, 28),
                     breaks = c(seq(12, 28, 4)))+

  ylab(expression(log[2]~(intensity))) + xlab("bead age") 
 
plot_pY_intensity_distribution_reps


ggsave("output/SuppFig8/plot_pY_intensity_distribution_reps.png", plot = plot_pY_intensity_distribution_reps, width = 8, height = 8, scale = 0.4)

ggsave("output/SuppFig8/plot_pY_intensity_distribution_reps.pdf", plot = plot_pY_intensity_distribution_reps, width = 8, height = 8, scale = 0.4)
```


####___- normalized
```{r}
psite_reps_CVs_YESnorm_pY <- my_qc_cvs(
  data = ascore_stringent_fasta_pYsite, 
  grouping = ref, 
  condition = condition,
  intensity = raw_median_norm_intensity,
  plot = TRUE, 
  xlab = expression(paste("bead age")),
  plot_style = "boxplot",
  max_cv = 100)

psite_reps_CVs_YESnorm_pY

ggsave(filename = "output/SuppFig8/CVs_YES_normalized_pY.png", plot = psite_reps_CVs_YESnorm_pY, width = 5, height = 9, scale = 0.4)
ggsave(filename = "output/SuppFig8/CVs_YES_normalized_pY.pdf", plot = psite_reps_CVs_YESnorm_pY, width = 4, height = 8, scale = 0.4)
```

