---
title: "Halo_HeLa_pept_concn_A"
author: "Alexis Chang"
date: "2024-05-09"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#0. functions
##--alexis_theme()
```{r}
alexis_theme <- function() {
  theme(
    # panel.border = element_rect(colour = "blue", fill = NA, linetype = 2),
    panel.border = element_blank(),
    panel.background = element_blank(),
    panel.grid.major.x  = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(angle = 90, vjust = 0.25, hjust = 1),
    axis.line = element_line(colour = "black"),
    axis.text = element_text(colour = "black", face = "plain", family = "sans"),
    axis.title = element_text(colour = "black", family = "sans"),
    axis.ticks = element_line(colour = "black"),
    plot.title = element_text(size=10),
    # legend at the bottom 6)
    legend.position = "right")   
}
```

##_b) CVs
```{r}
my_qc_cvs <- function (data,
                       grouping,
                       condition,
                       intensity,
                       plot = TRUE,
                       plot_style = "density",
                       max_cv = 200,
                       xlab = "condition",
                       showlegend = FALSE,
                       yintercept = 25){
  
  
  protti_colours <- "placeholder"
  utils::data("protti_colours", envir = environment())
  
  
#-----------------------------------------------------------------------------------  
  if (plot == FALSE) {
    
    
    if (max(dplyr::pull(data, {{intensity}}), na.rm = TRUE) < 1000) {
      stop(strwrap("Please backtransform your data or use raw values.\nThe function does not handle log2 transformed data.", 
                   prefix = "\n", initial = ""))  }
    
    
    result <- data %>% dplyr::distinct({{grouping}}, {{condition}}, {{intensity}}) %>%
      tidyr::drop_na({{intensity}}) %>%
      dplyr::group_by({{grouping}}) %>%
      dplyr::mutate(cv_combined = (stats::sd({{intensity}})/mean({{intensity}})) * 100) %>%
      dplyr::group_by({{condition}}, {{grouping}}) %>%
      dplyr::mutate(cv = (stats::sd({{intensity}})/mean({{intensity}})) * 100) %>%
      dplyr::distinct({{condition}}, {{grouping}}, .data$cv_combined, .data$cv) %>%
      tidyr::drop_na() %>%
      dplyr::group_by({{condition}}) %>%
      dplyr::mutate(median_cv = stats::median(.data$cv)) %>% 
      dplyr::ungroup() %>%
      dplyr::mutate(median_cv_combined = stats::median(.data$cv_combined)) %>% 
      dplyr::select(-{{grouping}}, -c("cv_combined", "cv")) %>%
      dplyr::distinct() 
    return(result)  }  
  
  ##-----------------------------------------------------------------------------------   
  if (plot == TRUE) {
    if (max(dplyr::pull(data, {{intensity}}), na.rm = TRUE) < 1000) {
      stop(strwrap("Please backtransform your data or use raw values.\nThe function does not handle log2 transformed data.", 
                   prefix = "\n", initial = ""))}
    
    result <- data %>%
      dplyr::distinct({{grouping}}, {{condition}}, {{intensity}}) %>%
      tidyr::drop_na({{intensity}}) %>%
      
      dplyr::group_by({{grouping}}) %>%
      dplyr::mutate(
        cv_combined = (stats::sd({{intensity}})/mean({{intensity}})) * 100) %>%
      
      
      dplyr::group_by({{condition}}, {{grouping}}) %>%
      dplyr::mutate(cv = (stats::sd({{intensity}})/mean({{intensity}})) * 100) %>%
      dplyr::ungroup() %>%
      
      dplyr::distinct({{condition}}, {{grouping}}, .data$cv_combined, .data$cv) %>%
      tidyr::drop_na() %>%
      tidyr::pivot_longer(cols = starts_with("cv"), names_to = "type", values_to = "values") %>%
      dplyr::mutate(type = ifelse(.data$type =="cv", {{condition}}, "all")) %>%
      dplyr::mutate(type = forcats::fct_relevel(as.factor(.data$type),"all")) %>%
      dplyr::select(-{{condition}}) %>%
      dplyr::group_by(.data$type) %>%
      dplyr::mutate(median = stats::median(.data$values)) %>% 
      dplyr::distinct()
    
    if (max(result$values) > max_cv) {
      cv_too_high <- result %>%
        dplyr::filter(.data$values >max_cv) %>%
        nrow()
      
      warning(paste(cv_too_high), " values were exluded from the plot (CV > ",max_cv, " %)")  }
    
    
    
    ##-----------------------------------------------------------------------------------      
    if (plot_style == "boxplot") {
      plot <- ggplot2::ggplot(result) +
        ggplot2::geom_boxplot(aes(x = .data$type, y = .data$values, fill = .data$type), na.rm = TRUE, size = 0.5, show.legend = showlegend,
                              alpha   = 0.5, outlier.alpha = 0.5, outlier.shape = 21) + 
        ggplot2::labs(
          # title = "Coefficients of variation",
          y = "Coefficient of variation [%]", fill = "Condition") + 
        ggplot2::geom_hline(yintercept = {{yintercept}}, linetype = 1, size = 0.75, alpha = 0.3) +
        ggplot2::scale_y_continuous(limits = c(0, max_cv)) + 
        scale_fill_brewer(palette = "Purples") +
        # scale_fill_brewer(palette = "Dark2") +
        # ggplot2::scale_fill_manual(values = c("grey",protti_colours)) +
        alexis_theme() +
        xlab({{xlab}}) +
        ggplot2::theme(axis.title.y = element_text(size = 10, hjust = 0),
                       axis.text.x = element_text(angle = -60, hjust = 0, vjust = 0.5))
        # ggplot2::theme(plot.title = ggplot2::element_text(size = 20),
        #                axis.title.x = ggplot2::element_text(size = 15), 
        #                axis.text.y = ggplot2::element_text(size = 15),
        #                axis.text.x = ggplot2::element_text(size = 12,angle = 75, hjust = 1),
        #                axis.title.y = ggplot2::element_text(size = 15), 
        #                legend.title = ggplot2::element_text(size = 15),
        #                legend.text = ggplot2::element_text(size = 15))
      return(plot)
    }
    
    ##-----------------------------------------------------------------------------------    
    if (plot_style == "density") {
      plot <- ggplot2::ggplot(result) +
        ggplot2::geom_density(ggplot2::aes(x = .data$values, col = .data$type), size = 1, na.rm = TRUE, show.legend = showlegend) + 
        ggplot2::labs(
          # title = "Coefficients of variation",
          x = "Coefficient of variation [%]", y = "Density", color = "Condition") +
        ggplot2::scale_x_continuous(limits = c(0,max_cv)) +
        geom_vline(data = dplyr::distinct(result,  .data$median, .data$type),
                   ggplot2::aes(xintercept = median, col = .data$type),
                   size = 1,
                   linetype = "dashed", 
                   show.legend = FALSE) +
        scale_fill_brewer(palette = "Purples") +
        # scale_fill_brewer(palette = "Dark2") +
        # ggplot2::scale_color_manual(values = c("grey",protti_colours)) +
        alexis_theme() +
        xlab(xlab)
        # ggplot2::theme(plot.title = ggplot2::element_text(size = 20), 
        #                axis.title.x = ggplot2::element_text(size = 15),
        #                axis.text.y = ggplot2::element_text(size = 15),
        #                axis.text.x = ggplot2::element_text(size = 12, angle = 75, hjust = 1),
        #                axis.title.y = ggplot2::element_text(size = 15),
        #                legend.title = ggplot2::element_text(size = 15),
        #                legend.text = ggplot2::element_text(size = 15))
      
      return(plot)
      
    }
    
    ##-----------------------------------------------------------------------------------
    if (plot_style == "violin") {
      
      plot <- ggplot2::ggplot(result, aes(x = .data$type, 
                                          y = .data$values, fill = .data$type)) +
        ggplot2::geom_violin(na.rm = TRUE, size = 1) + 
        ggplot2::geom_boxplot(width = 0.15, fill = "white", na.rm = TRUE, alpha = 0.6, size = 0.75, outlier.color = NA) +
        ggplot2::labs(
          # title = "Coefficients of variation",
                      x = "", y = "Coefficient of variation [%]",
                      fill = "Condition") +
        ggplot2::geom_hline(yintercept = {{yintercept}}, linetype = 1, size = 0.75, alpha = 0.3) +
        ggplot2::scale_y_continuous(limits = c(0, max_cv)) +
        # ggplot2::scale_fill_manual(values = c("grey",  protti_colours)) +
        scale_fill_brewer(palette = "Purples") +
        # scale_fill_brewer(palette = "Dark2") +
        alexis_theme() +
        xlab(xlab)
        # ggplot2::theme_bw() +
        # ggplot2::theme(plot.title = ggplot2::element_text(size = 20),
        #                axis.title.x = ggplot2::element_text(size = 15),
        #                axis.text.y = ggplot2::element_text(size = 15), 
        #                axis.text.x = ggplot2::element_text(size = 12, angle = 75, hjust = 1), axis.title.y = ggplot2::element_text(size = 15),
        #                legend.title = ggplot2::element_text(size = 15), 
        #                legend.text = ggplot2::element_text(size = 15))
      
      return(plot)
      
    }
  }
}

```


##_c) Hierarchical clustering

Also, you can try other variants. Next two are slightly bigger the common asterisk.
https://stackoverflow.com/questions/54342274/pheatmap-display-numbers-argument

    Heavy asterisk
    test_labels[test_labels<=10] <- "\u2731\u2731"
    
    Full width asterisk
    test_labels[test_labels<=10] <- "\uFF0A\uFF0A"

      #EXAMPLE CODE WITH ASTERISKS -----------------------------------------------
      library(pheatmap)
      
      test_vals <- matrix(rnorm(20), 5, 4)
      test_labels <- matrix(1:20, 5, 4) 
      test_labels[test_labels <= 10] <- "\u2217\u2217"
      pheatmap(test_vals, display_numbers = test_labels, fontsize_number=20, cellheight=20)
      
      
      
      #EXAMPLE CODE WITH COLOR CHANGE OF NUMBERS IN CELLS ---------------------------
      pheatmap(m,
         display_numbers = TRUE,
         number_color = "black", 
         fontsize_number = 8)
```{r}
my_hier_cluster <- function (data,
          sample,
          grouping,
          intensity_log2,
          condition,
          digestion = NULL, 
          run_order = NULL, 
          method = "spearman", 
          fontsize_number = 14,
          number_color = "white",
          cell_height = 20, 
          interactive = FALSE) 
{
  
  protti_colours <- "placeholder"
  
  utils::data("protti_colours", envir = environment())
  
  viridis_colours <- "placeholder"
  
  utils::data("viridis_colours", envir = environment())
  
  
  
  
  correlation <- data %>%
    dplyr::distinct({{sample}}, {{grouping}}, {{intensity_log2}}) %>%
    tidyr::pivot_wider(names_from = { {sample}},
                       values_from = {{intensity_log2}}) %>%
    tibble::column_to_rownames(var = rlang::as_name(enquo(grouping))) %>% 
    stats::cor(method = {{method}}, use = "complete.obs")
  
  
  
  
  annotation <- data %>%
    dplyr::mutate(`:=`({{condition}}, as.character({{condition}}))) %>%
    dplyr::distinct({{sample}}, {{condition}}, {{digestion}}, {{run_order}}) %>%
    tibble::column_to_rownames(var = rlang::as_name(enquo(sample)))
  
  n_conditions <- 0
  n_digest <- 0
  n_run_ord <- 0
  conditions_colour <- c()
  digest_colours <- c()
  run_ord_colours <- c()
  
  
  
  
  if (!missing(condition)) {
    conditions <- unique(dplyr::pull(annotation, {{condition}}))
    n_conditions <- length(conditions)
    conditions_colours <- protti_colours[1:n_conditions]
    names(conditions_colours) <- conditions
  }
  
  
  
  
  
  
  if (!missing(digestion)) {
    digest <- unique(dplyr::pull(annotation, {{digestion}}))
    n_digest <- length(digest)
    digest_colours <- protti_colours[(n_conditions + 1):(n_digest + 
                                                           n_conditions)]
    names(digest_colours) <- digest
  }
  
  
  
  
  
  if (!missing(run_order)) {
    colfunc <- grDevices::colorRampPalette(c("#0D0887", 
                                             "#2E0595", "#46039F", "#5C01A6", "#7201A8", "#8707A6", 
                                             "#9A169F", "#AC2694", "#BC3587", "#CA457A", "#D6556D", 
                                             "#E26561", "#EB7655", "#F48849", "#FA9B3D", "#FDAF31", 
                                             "#FDC527", "#F9DC24", "#F0F921"))
    run_ord <- unique(dplyr::pull(annotation, {
      {
        run_order
      }
    }))
    n_run_ord <- length(run_ord)
    run_ord_colours <- colfunc(n_run_ord)
    names(run_ord_colours) <- run_ord
  }
  
  
  
  
  annotation_colours <- list(conditions_colours, digest_colours, run_ord_colours)
  
  
  names(annotation_colours) <- c(if (!missing(condition)) {
    rlang::as_name(enquo(condition))
  } else {
    "condition"
  }, if (!missing(digestion)) {
    rlang::as_name(enquo(digestion))
  } else {
    "digestion"
  }, if (!missing(run_order)) {
    rlang::as_name(enquo(run_order))
  } else {
    "run_order"
  })
  
  
  
  
  
  #interactive -----------------------------------------------------------------------------------
  
  if (interactive == TRUE) {
    if (!requireNamespace("heatmaply", quietly = TRUE)) {
      message("Package \"heatmaply\" is needed for this function to work. Please install it.", 
              call. = FALSE)
      return(invisible(NULL))
    }
    heatmap_interactive <- heatmaply::heatmaply(correlation, 
                                                main = "Correlation based hirachical clustering of samples", 
                                                col_side_colors = annotation, col_side_palette = c(annotation_colours[[1]], 
                                                                                                   annotation_colours[[2]], annotation_colours[[3]]),
                                                display_numbers(round(correlation, 2)),
                                                k_col = NA, k_row = NA, plot_method = "plotly")
    return(heatmap_interactive)
  }
  
  
  
  
  
  #not interactive ----------------------------------------------------------------------------------
  
  if (interactive == FALSE) {
    dependency_test <- c(dendextend = !requireNamespace("dendextend", 
                                                        quietly = TRUE),
                         pheatmap = !requireNamespace("pheatmap", 
                                                      quietly = TRUE), seriation = !requireNamespace("seriation", quietly = TRUE))
    if (any(dependency_test)) {
      
      dependency_name <- names(dependency_test[dependency_test ==  TRUE])
      
      if (length(dependency_name) == 1) {
        message("Package \"", paste(dependency_name), "\" is needed for this function to work. Please install it.", call. = FALSE)
        return(invisible(NULL))
      }
      
      else {
        message("Packages \"", paste(dependency_name, collapse = "\" and \""), "\" are needed for this function to work. Please install them.", call. = FALSE)
        return(invisible(NULL))
      }
    }
    
    
    distance <- stats::dist(correlation)
    
    hierachical_clustering <- stats::hclust(distance)
    
    dendrogram <- stats::as.dendrogram(hierachical_clustering)
    
    dendrogram_row <- dendextend::seriate_dendrogram(dendrogram, 
                                                     distance, method = "OLO")
    
    dendrogram_column <- dendextend::rotate(dendrogram_row, 
                                            order = rev(labels(distance)[seriation::get_order(stats::as.hclust(dendrogram_row))]))
    
    heatmap_static <- pheatmap::pheatmap(correlation,
                                         cluster_rows = stats::as.hclust(dendrogram_row), 
                                         cluster_cols = stats::as.hclust(dendrogram_column), 
                                         display_numbers = round(correlation,2),
                                         number_color = {{number_color}},
                                         fontsize_number = {{fontsize_number}},
                                         annotation = annotation,
                                         annotation_colors = annotation_colours, 
                                         main = "Correlation based hierachical clustering of samples", 
                                         color = viridis_colours, silent = TRUE)
    return(heatmap_static)
  }
}

```

#1. LOAD
##_a) libraries
```{r}
library(tidyverse)
library(janitor)
library(readr)
library(gridExtra)
library(ggrepel)
library(cowplot)
library(reshape2)
library(eulerr)
library(protti)
library(diann)
library(iq)
library(RColorBrewer)
#for PCA loadings
library(FactoMineR)

#for protti sample correlation and volcano plots
library(dendextend)
library(pheatmap)
library(seriation)
library(UpSetR)
```

##_b) FASTA references
   -> prepare Yeast + Vsrc FASTA
   Downloaded from Zucchini on 1May2023, but pulled from uniprot by Bianca Ruiz in April 2017. K12 strain, swiss prot (reviewed entries only)
```{r}
human_fasta_2024 <- read_csv("raw_data/sp_reviewed_UP000005640_2024_03_21_forR.csv")


haloTEVsSrc <- read_csv("raw_data/haloTEVsSrc_forR.csv") %>%
  clean_names() %>% 
  mutate(
    reference = "HalosSrcV2",
    organism = "bait",
    length = str_length(full_protein_sequence),
    gene = NA, 
    protein_names = NA)


Ecoli_fasta_2017 <- read_csv("raw_data/Ecoli_K12_uniprot_April2017_BR_forR.csv") %>%
  clean_names() %>% 
  mutate(
    organism = "E.coli",
    length = str_length(full_protein_sequence),
    gene = NA, 
    protein_names = description) %>% 
  select(-description)

human_fasta_w_bg <- rbind(human_fasta_2024, haloTEVsSrc, Ecoli_fasta_2017)
```



##_c) your data
###__i) comet
```{r}

comet <- read_csv("raw_data/MainFig2/AC078_hiRes_comet/result.csv") %>%
  clean_names() %>%
  separate(sample_name, into = c("exp", "lysate", "well", "pept_amt","condition", "inj_vol", "rep"), sep = "_", remove = FALSE) %>% 
  filter(lysate == "HFPV") %>%
  filter(well != "pool") %>%
  
  # filter(raw_file_name %in% c("x09135", # 4mg/mL rep 1
  #                             "x09136", # 1 mg/mL rep 1
  #                             "x09141", # 4 mg/mL rep 3
  #                             "x09137")) %>%  # 1 mg/mL rep 2
  mutate(
    replicate = as.numeric(str_sub(rep, start = -1L)),
    # replicate = case_when(
    #   raw_file_name == "x09139" ~ 3, #fix incorrect sample names from Xcalibur queue
    #   raw_file_name == "x09140" ~ 2,
    #   raw_file_name == "x09141" ~ 3,
    #   TRUE ~ replicate),
    condition = case_when(
      condition == "05mgml" ~ "0.5 mg/mL",
      condition == "1mgml" ~ "1 mg/mL",
      condition == "2mgml" ~ "2 mg/mL",
      condition == "4mgml" ~ "4 mg/mL")) %>% 
  
  filter(replicate %in% c(1, 2, 3)) %>%   #had extra replicate for 2 mg/mL, remove this replicate 4
  left_join(y = human_fasta_w_bg, by = c("reference")) %>% 
  
  ##rename column to 'intensity'
  rename(intensity = max_intensity_light_c2837) %>% 
  
  ##keep only forward hits
  filter(reverse == FALSE) %>% 
  
  ##classify phospho vs. no none
  mutate(phospho = case_when(
    grepl("\\@", sequence) ~ "phospho",
    TRUE~"none"))
  

comet %>% distinct(raw_file_name, condition, replicate)  
```




###__i) ascore
```{r}

  ascore <- read_csv("raw_data/MainFig2/AC078_hiRes_ascore/result.csv") %>%
  clean_names() %>%
  separate(sample_name, into = c("exp", "lysate", "well", "pept_amt","condition", "inj_vol", "rep"), sep = "_", remove = FALSE) %>% 
  filter(lysate == "HFPV") %>%
  filter(well != "pool") %>%
  # filter(raw_file_name %in% c("x09135", # 4mg/mL rep 1
  #                             "x09136", # 1 mg/mL rep 1
  #                             "x09141", # 4 mg/mL rep 3
  #                             "x09137")) %>%  #1 mg/mL rep 2
  
  mutate(
    replicate = as.numeric(str_sub(rep, start = -1L)),
    # replicate = case_when( 
    #   raw_file_name == "x09139" ~ 3, #fix incorrect sample names from Xcalibur queue
    #   raw_file_name == "x09140" ~ 2,
    #   raw_file_name == "x09141" ~ 3,
    #   TRUE ~ replicate),
    condition = case_when(
      condition == "05mgml" ~ "0.5 mg/mL",
      condition == "1mgml" ~ "1 mg/mL",
      condition == "2mgml" ~ "2 mg/mL",
      condition == "4mgml" ~ "4 mg/mL")) %>% 
  
  filter(replicate %in% c(1, 2, 3)) %>%   #had extra replicate for 2 mg/mL, remove this replicate 4
  
  ##join to FASTA
  left_join(y = human_fasta_2024, by = c("reference")) %>% 
  
  ##rename column to 'intensity'
  rename(intensity = max_intensity_light_c2837) %>% 
  
  ##keep only forward hits
  filter(reverse == FALSE) %>% 
  
  ##classify phospho vs. no none
  mutate(phospho = case_when(
    grepl("\\@", sequence) ~ "phospho",
    TRUE~"none"))
  
```





#2. COMET QC
##_a) ENRICHMENT EFFICIENCY
###__i) counts
df
```{r}
enrichment_efficiency_count_df <- comet %>% 
  distinct( condition, replicate,sequence, phospho) %>% 
  group_by( condition, replicate, phospho) %>% 
  summarize(
    num_phospho = n()) %>% 
  pivot_wider(names_from = phospho, values_from = num_phospho) %>% 
  mutate(
    total_pept = none + phospho,
    individual_phosphopept_enrich_efficiency = phospho / total_pept) %>% 
  ungroup() %>% 
  group_by( condition) %>% 
  mutate(
    avg_phosphopeptide_enrichment_efficiency = mean(individual_phosphopept_enrich_efficiency)) %>% 
  ungroup() %>% 
  mutate(
    sample_id = paste(condition, replicate, sep = "_") )


##view df of phosphopeptide enrichment efficiency ratios
enrichment_efficiency_count_df
```

plot
```{r}
plot_ppept_enrich_efficiency <- ggplot() +
  geom_bar(data = (enrichment_efficiency_count_df %>%
                     distinct(condition, avg_phosphopeptide_enrichment_efficiency)),
           mapping = aes(x = condition, y = avg_phosphopeptide_enrichment_efficiency),
           stat = "identity", alpha = 0.7) + 
  geom_jitter(data = enrichment_efficiency_count_df,
              mapping = aes(x = condition, y = individual_phosphopept_enrich_efficiency, fill = replicate), shape = 1, show.legend = FALSE, height = 0, width = 0.2) +
  # theme_bw(18) +
  # theme(legend.position = "none") +
  ylab("p-/all peptide (count)") +
  xlab ("") +
  # xlab("sample concentration") +
  
  alexis_theme()+
  scale_y_continuous(expand = c(0,0), limits = c(0, 1.01), breaks = c(seq(0, 1, 0.25))) + #removes whitespace below bars!
  theme(axis.text.x = element_text(angle = -60, vjust = 0.15, hjust = 0.1, size = 8),
        axis.title = element_text(hjust = 0))
  ## ggtitle("Phosphopeptide enrichment efficiency")

plot_ppept_enrich_efficiency
ggsave("output/MainFig2/phospho_pept_enrich_efficiency_counts.png", plot = plot_ppept_enrich_efficiency, width = 4, height = 6, scale = 0.4)
ggsave("output/MainFig2/phospho_pept_enrich_efficiency_counts.pdf", plot = plot_ppept_enrich_efficiency, width = 4, height = 6, scale = 0.4)
```


###__ii) intensity
df
```{r}
enrichment_efficiency_intensity_df <- comet %>% 
  distinct(sequence,  condition, replicate, phospho, intensity) %>% 
  group_by(sequence,  condition, replicate, phospho) %>% 
  
##filter by max intensity for every peptide
  filter(intensity  == max(intensity)) %>% 
  ungroup() %>% 
  group_by(phospho, condition, replicate) %>% 
  
##summarize intensity of all peptides and just p-peptides
  summarise(all_peptides_intensity = sum(intensity)) %>% 
  pivot_wider(values_from = all_peptides_intensity, names_from = phospho) %>% 
  
##grouped mutate for taking mean of summed intensities for each replicate within each condition  
  group_by(condition) %>%   
  mutate(mean_phos = mean(phospho),
         mean_non = mean(none)) %>% 
  ungroup() %>% 
  mutate(mean_ratio = mean_phos / (mean_non + mean_phos),
         rep_ratio = phospho / (none + phospho)) %>% 
  ungroup() %>% 
  mutate(
    sample_id = paste(condition, replicate, sep = "_") )
  
  
  enrichment_efficiency_intensity_df
```



plot
```{r}
##relative to intensity of all peptides (with and without phospho)

plot_ratio_phos_intensity <- ggplot(data = (enrichment_efficiency_intensity_df)) + 
  geom_bar(aes(x = condition, y = mean_ratio),
           stat = "identity",
           position = "dodge",
           fill = "skyblue2",
           alpha = 0.3) +
  geom_point(aes(x = condition,
                 y = rep_ratio,
                 fill = replicate),
             position = position_jitterdodge(dodge.width = 0.2), alpha = 1, size =1, shape = 1, show.legend =  FALSE)  +
  # theme_bw(18) +
  # theme(axis.text.x = element_text(angle=90, vjust = 0.25, hjust = 1), legend.position = "none") +
  # theme(legend.position = "none") +
  ylab("p- /all peptide (intensity)") +
  xlab ("") +
  # xlab("sample concentration") +
  
  alexis_theme()+
  scale_y_continuous(expand = c(0,0), limits = c(0, 1.02), breaks = c(seq(0, 1, 0.25))) + #removes whitespace below bars!
  theme(axis.text.x = element_text(angle = -60, vjust = 0.1, hjust = 0.15, size = 8),
        axis.title = element_text(hjust = 0, size = 10))
  

plot_ratio_phos_intensity
ggsave("output/MainFig2/phospho_pept_intensity_ratio.png", plot = plot_ratio_phos_intensity, width = 3.9, height = 6, scale = 0.4)
ggsave("output/MainFig2/phospho_pept_intensity_ratio.pdf", plot = plot_ratio_phos_intensity, width = 4, height = 6, scale = 0.4)
```

##_b)  OVERALL ITENSITY
```{r}
plot_psm_intensity_distribution_reps <- ggplot(data = comet) +
  geom_boxplot(mapping = aes(x = condition, y = log2(intensity),
                             fill = as.factor(replicate)),
               alpha = 1,
               outlier.shape = 21,
               outlier.alpha = 0.3,
               show.legend = FALSE) +
  # theme_bw(18) +
  alexis_theme()+
  # scale_y_continuous(expand = c(0,0)) + #removes whitespace below bars!
  scale_fill_brewer(palette = "Dark2") +
  # theme(axis.text.x = element_text(angle=90, vjust = 0.25, hjust = 1), legend.position = "none") +
  # ylab(expression(log[2]~(intensity))) + xlab("peptide concentration") +
  ggtitle("Intensity of all PSMs")

plot_psm_intensity_distribution_reps


ggsave("output/MainFig2/all_psm_intensity_distribution_reps.png", plot = plot_psm_intensity_distribution_reps, width = 6, height = 8, scale = 0.4)
ggsave("output/MainFig2/all_psm_intensity_distribution_reps.pdf", plot = plot_psm_intensity_distribution_reps, width = 6, height = 8, scale = 0.4)
```
#-----------------------------------------------

#3. ASCORE
##_a) parse p-site localization
Working from 'Ascore Extended' output from Zucchini. Avoids relying on "mod_location" from CometExtended for phospho localization.

###__i) index to peptide
```{r}
ascore_peptide_index <- ascore %>%
  filter(replicate != 4) %>% 
  distinct() %>%
  
  mutate(seq_no_mods = str_remove_all(ascore_sequence, "[:punct:]"),
         seq_no_mods = str_remove_all(seq_no_mods, "[n]")) %>% ##remove n from n## mark of nAc modification. 
  mutate_at(c("a_score_1", "a_score_2", "a_score_3", "position_1", "position_2", "position_3"), funs(as.numeric)) %>% 
  mutate_at(c("a_score_1", "a_score_2", "a_score_3", "position_1", "position_2", "position_3"), ~replace_na(., 0)) %>%

  select(sample_name,  condition, replicate,  reference,
         ascore_sequence, num_sites, a_score_1, a_score_2, a_score_3, position_1, position_2, position_3,
         seq_no_mods, intensity, q_score_c2837, num_scans_light_c2837, charge, x_corr, 
         missed_cleavages, num_sites, redundancy) %>% 
 
##extract modified residue
  mutate(
    mod_res1 = str_sub(seq_no_mods, start = position_1+1L, end = position_1 + 1L),
    mod_res2 = str_sub(seq_no_mods, start = position_2+1L, end = position_2 + 1L),
    mod_res3 = str_sub(seq_no_mods, start = position_3+1L, end = position_3 + 1L)) %>% 


## pivot wider to match each p-site to an ascore and filter for STY

  unite(ascore_mod_res1, c(a_score_1, mod_res1, position_1), sep = "_") %>% 
  unite(ascore_mod_res2, c(a_score_2, mod_res2, position_2), sep = "_") %>%
  unite(ascore_mod_res3, c(a_score_3, mod_res3, position_3), sep = "_") %>%
  pivot_longer(cols = c("ascore_mod_res1", "ascore_mod_res2", "ascore_mod_res3")) %>% 
  separate(value, into = c("ascore", "mod_res", "mod_position"), sep = "_") %>% 
  mutate(ascore= as.numeric(ascore))


```

###__ii) filter Ascore > 13
```{r}
ascore_peptide_index_stringent <- ascore_peptide_index %>% 
  filter(ascore >= 13) %>% 
  filter(grepl("[STY]", mod_res) == TRUE)
```

###__iii) index to protein
JOIN TO FASTA here
```{r}
  #join to FASTA
ascore_stringent_fasta <- ascore_peptide_index_stringent %>% 
  left_join(y = human_fasta_w_bg, by = c("reference")) %>% 
  
  #keep detected proteins only
  filter(!is.na(ascore_sequence)) %>% 
  
  mutate(
    mod_position = as.numeric(mod_position), ##mod_position parsed from ascore output, not Comet_Extended
    z_peptide = str_replace_all(seq_no_mods, "[IL]", "Z"),
    z_protein = str_replace_all(full_protein_sequence, "[IL]", "Z"),
    z_pept_position_in_z_protein = str_locate(z_protein, pattern = z_peptide), ##get peptide position from z-substituted peptide and protein
    mod_position_in_protein = z_pept_position_in_z_protein[,"start"] + mod_position, ##extract mod position from true protein sequence
    test_mod_position = str_sub(full_protein_sequence, start = mod_position_in_protein, end = mod_position_in_protein)) %>% 
  select(test_mod_position, mod_res, mod_position, ascore_sequence, everything()) %>% 
  unite(col = "mod_protein_location", c(mod_res, mod_position_in_protein), sep = "", remove = FALSE) %>% 
  unite(col = "ref", c(reference, mod_protein_location), sep = "_", remove = FALSE)
  





#index to protein (account for equal I and L during search)


```

Check for unmapped phosphosites
```{r}
    ##tally up the number of rows that have NA vs. STY in test_mod_position
mod_pos_group <- ascore_stringent_fasta %>% 
  select(test_mod_position) %>% 
  group_by(test_mod_position) %>% 
   summarize(
    num_phospeptides = n())
mod_pos_group
```

Are all mods are on S, T or Y as hoped? If NAs exist, there is a mapping issue. 
For Yeast spurious, the mutant kinases did map back to my reference database. Just mapped to un-mutated protein instead.


###__iv) collapse PSM to precursor (max intensity)
keep single PSM per precursor
take max intensity PSM per precursor (peptide + charge)
```{r}
ascore_stringent_fasta_precursor <- ascore_stringent_fasta %>% 
  group_by(condition, replicate, ascore_sequence, charge, ascore) %>% 
  filter(intensity == max(intensity)) %>% 
  ungroup()
```

_save lists of peptides for supplementary_
write to csv
```{r}
#replicate
write_csv(x = ascore_stringent_fasta_precursor %>% distinct(condition, replicate, ascore_sequence, charge, ascore, intensity, reference, sample_name, gene, protein_names, mod_position_in_protein, ref), file = "modified_data/MainFig2/distinct_pSTYpeptides_ascore13_replicate.csv", col_names = TRUE)


#condition
write_csv(x = ascore_stringent_fasta_precursor %>% distinct(condition,  ascore_sequence, charge, ascore, intensity, reference, gene, protein_names, mod_position_in_protein, ref), file = "modified_data/MainFig2/distinct_pSTYpeptides_ascore13_condition.csv", col_names = TRUE)
```



###__v) collapse precursor to p-site (by sum)
This combines intensities from different phosphoisoform peptides into single site measurements. 
Alternatives could be to differentiate phosphoisoform peptides and sum (i.e. add modifier to ref to indicate # of other sites co-detected)
Alternatively, I could also just take the monophosphorylated peptide if present and if no monophosphorylated peptide, I could average intensities summed to distinct isoforms (as in Krug et al 2019 for PTM SEA)

```{r}
##SUM PRECURSOR INTENSITIES TO P-SITES + median normalize

 ascore_stringent_fasta_psite <- ascore_stringent_fasta_precursor %>% 
  group_by(condition, replicate,  reference, mod_res,  ref) %>% #ref = unique p-site
  
  #sum PSM intensities of confident p-sites to individual p-sites
  mutate(
    sum_intensity_precursor_to_psite = sum(intensity),
    log2_psite_qty = log2(sum_intensity_precursor_to_psite)) %>% 
  ungroup() %>% 
  
  #keep single summed intensity per p-site prior to median normalization
  distinct(condition, replicate, reference, mod_res, ref, sum_intensity_precursor_to_psite, log2_psite_qty) 
  
  
```

###__vi) median normalize pSTY-sites by sample
```{r}
ascore_stringent_fasta_pSTYsite <- ascore_stringent_fasta_psite %>%
  # filter(mod_res == "Y") %>% 
  #median normalize intensities summed to p-site
  mutate(
    global_median_intensity = median(log2_psite_qty)) %>% 
  group_by(condition, replicate) %>% 
  mutate(
    sample_median_intensity = median(log2_psite_qty)) %>% 
  ungroup() %>% 
  mutate(
    median_norm_intensity = log2_psite_qty - sample_median_intensity + global_median_intensity,
    raw_median_norm_intensity = 2^median_norm_intensity) %>% 
  mutate(sample_id = paste(condition, replicate, sep = "_"))
```

###__vi) median normalize pY-sites by sample
```{r}
ascore_stringent_fasta_pYsite <- ascore_stringent_fasta_psite %>%
  filter(mod_res == "Y") %>% 
  #median normalize intensities summed to p-site
  mutate(
    global_median_intensity = median(log2_psite_qty)) %>% 
  group_by(condition, replicate) %>% 
  mutate(
    sample_median_intensity = median(log2_psite_qty)) %>% 
  ungroup() %>% 
  mutate(
    median_norm_intensity = log2_psite_qty - sample_median_intensity + global_median_intensity,
    raw_median_norm_intensity = 2^median_norm_intensity) %>% 
  mutate(sample_id = paste(condition, replicate, sep = "_"))
```



##_b) count p-sites



###__i) by condition

df
Use dataframe PRIOR TO collapsing PSMs to precursors, precursor to psites and median normalized intensity.
```{r}
distinct_p_sites_condition <- ascore_stringent_fasta %>% 
  distinct(reference, mod_protein_location, mod_res, condition)
##this collapses all  replicates into a single set of unique p sites.
```


write to csv
```{r}
write_csv(x = distinct_p_sites_condition, file = "modified_data/MainFig2/distinct_pSTYsites_ascore13_condition.csv", col_names = TRUE)
```

plot
```{r}
plot_distinct_p_sites_condition_wNums <- ggplot(data = distinct_p_sites_condition) +
  geom_bar(mapping = aes(x = condition, fill = mod_res)) +
  geom_text(data = tibble(x = c(1, 2, 3, 4),y = c(4263, 4364, 4462, 4441)), mapping = aes( x = x, y = y -500, label = paste0(y, "\npY")),
            color = "black",
            size = 3.5,
            fontface = "bold",
            inherit.aes = FALSE) +
  alexis_theme()+
  theme(axis.text.x = element_text(angle = -60, hjust = 0.15, vjust = 0.1)) +
  scale_y_continuous(expand = c(0,0)) + #removes whitespace below bars!
  scale_fill_viridis_d(name = "mod\nsite") +
  ylab("unique phospho sites") +
  xlab("peptide concentration")
  # xlab(expression(paste("Bead slurry volume (", mu, "L", ")"))) 

plot_distinct_p_sites_condition_wNums

ggsave("output/MainFig2/distinct_p_sites_condition_wNumbers.png", plot = plot_distinct_p_sites_condition_wNums, width = 8, height = 8, scale = 0.4)
ggsave("output/MainFig2/distinct_p_sites_condition_wNumbers.pdf", plot = plot_distinct_p_sites_condition_wNums, width = 8, height = 8, scale = 0.4)
```
check how many of the pS sites are on pY isoforms, or are separate...This could suggest non-specific capture if on different peptides, or phosphopriming on the same peptide.

table summary of overall site counts
```{r}
summarize_distinct_psites_condition <- distinct_p_sites_condition %>%
  ungroup() %>% 
  group_by(condition, mod_res) %>% 
  summarize(
    n_psites = n())

summarize_distinct_psites_condition
```


###__ii) by replicate

df
```{r}
distinct_p_sites_reps <- ascore_stringent_fasta %>% 
  ungroup() %>% 
  distinct(reference, mod_protein_location, mod_res,  condition, replicate) %>% 
  mutate(
    sample_id = paste(condition, replicate, sep = "_"))
##this collapses all  replicates into a single set of unique p sites.
```


write to csv
```{r}
write_csv(x = distinct_p_sites_reps, file = "modified_data/MainFig2/pSTYsites_ascore13_reps.csv", col_names = TRUE)
```

plot

```{r}
plot_distinct_p_sites_reps <- ggplot(data = distinct_p_sites_reps) +
  geom_bar(mapping = aes(x = sample_id, fill = mod_res)) +
  geom_text(data = tibble(x = c(seq(1, 12, 1)),
                          
                          # tibble_pY_reps <- (summarize_distinct_psites_replicates %>% filter(mod_res == "Y))$n_psites
                          y = c(3475, 3598, 3488, 3621, 3748, 3478 ,3306 , 3862,3598 ,3658 , 3564,3716 )),
            
            mapping = aes( x = x, y = y -400, label = y, angle = -90),
            color = "black",
            size = 3,
            fontface = "bold",
            inherit.aes = FALSE) +
  alexis_theme()+
  scale_y_continuous(expand = c(0,0)) + #removes whitespace below bars!
  scale_fill_viridis_d() +
  ylab("unique phospho sites") +
  xlab(expression(paste("Bead slurry volume (", mu, "L", ")")))

plot_distinct_p_sites_reps

ggsave("output/MainFig2/distinct_p_sites_reps.png", plot = plot_distinct_p_sites_reps, width = 8, height = 8, scale = 0.4)
ggsave("output/MainFig2/distinct_p_sites_reps.pdf", plot = plot_distinct_p_sites_reps, width = 8, height = 8, scale = 0.4)
```
the max IT 86 ms and AGC200 provided more unique confidently localized pY sites. This could have been from column conditioning or the method difference. Either way the pY site counts are comparable.




table
```{r}
summarize_distinct_psites_replicates <- distinct_p_sites_reps %>%
  ungroup() %>% 
  group_by(condition, replicate, sample_id, mod_res) %>% 
  summarize(
    n_psites = n())

summarize_distinct_psites_replicates
```
#==============================

#2. ANALYZE ASCORE
##a. N DISTINCT P-SITES 

###__i. total:
```{r}
distinct_p_sites_total <- ascore_stringent_fasta %>% 
  distinct(reference, mod_position_in_protein, mod_res) %>%
  mutate(filter_level = "Ascore >= 13")
```


#### . . . . summarized count for text (PUB)
```{r}
distinct_pY_sites_total_summary <- distinct_p_sites_total %>% 
  group_by(mod_res) %>% 
  summarize(
    n_distinct_sites = n()) %>% 
  ungroup()

distinct_pY_sites_total_summary
```


#### . . . . plot
```{r}
plot_distinct_p_sites_total <- ggplot(data = distinct_p_sites_total) +
  geom_bar(mapping = aes(x = filter_level, fill = mod_res)) +
  theme_bw(18) +
  scale_fill_viridis_d() +
  ylab("unique phospho sites") +
  alexis_theme()+
  # expand_limits(y = c(0, 2000)) +
  scale_y_continuous(expand = c(0,0))
  # theme(axis.text.x = element_text(angle = 90, vjust = 0.25, hjust = 1))

plot_distinct_p_sites_total

ggsave("output/MainFig2/distinct_p_sites_total.png", plot = plot_distinct_p_sites_total, width = 6, height = 8, scale = 0.4)
ggsave("output/MainFig2/distinct_p_sites_total.pdf", plot = plot_distinct_p_sites_total, width = 8, height = 8, scale = 0.4)
```


###__ii. by condition:

#### . . . . filter for distinct phospho sites


```{r}
distinct_p_sites_condition <- ascore_stringent_fasta %>% 
  distinct(reference, mod_protein_location, mod_res, condition)
##this collapses all  replicates into a single set of unique p sites.
```

#### . . . . write to csv 
write to csv distinct pSTY sites per condition

```{r}
write_csv(x = distinct_p_sites_condition, file = "modified_data/MainFig2/pSTYsites_ascore13_condition.csv", col_names = TRUE)
```

#### . . . . plot
plot number of distinct phospho sites 
-look at relative fraction of STY and compare to human ratios.

```{r}
plot_distinct_p_sites_condition <- ggplot(data = distinct_p_sites_condition ) +
  geom_bar(mapping = aes(x = condition, fill = mod_res)) +
  geom_text(data = tibble(x = c(1, 2, 3, 4),y = c(4263, 4364, 4462, 4441)), mapping = aes( x = x, y = y -500, label = paste0(y, "\npY")),
            color = "black",
            size = 3.5,
            fontface = "bold",
            inherit.aes = FALSE) +
  theme_bw(18) +
  scale_fill_viridis_d(name = "mod\nsite") +
  ylab("unique phospho sites") +
  alexis_theme()+
  # expand_limits(y = c(0, 4000)) +
  scale_y_continuous(expand = c(0,0)) +
  xlab("peptide concentration \n (0.5 mg input)")+
  theme(axis.text.x = element_text(angle = -60, vjust = 0.15, hjust = 0.1, ),
        axis.title.x = element_text(size = 15))

plot_distinct_p_sites_condition

ggsave("output/MainFig2/distinct_p_sites_condition.png", plot = plot_distinct_p_sites_condition, width = 10, height = 12, scale = 0.4)
ggsave("output/MainFig2/distinct_p_sites_condition.pdf", plot = plot_distinct_p_sites_condition, width = 8, height = 8, scale = 0.4)
```
####. . . . counts
```{r}
counts_distinct_psites_condition <- distinct_p_sites_condition %>% 
  distinct() %>% 
  group_by(condition, mod_res) %>% 
  summarize(
    n_distinct_sites = n()) %>% 
  ungroup() %>% 
  group_by(condition) %>% 
  mutate(total_sites = sum(n_distinct_sites)) %>% 
  ungroup() %>% 
  mutate(
    ratio_pY_sites = n_distinct_sites / total_sites * 100) %>% 
  group_by(mod_res) %>% 
  mutate(
    average_pYsite_percent = mean(ratio_pY_sites) ) %>% 
  ungroup()

counts_distinct_psites_condition
```
###__iii. by replicate

#### . . . . filter for distinct phospho sites

df
```{r}
distinct_p_sites_reps <- ascore_stringent_fasta %>% 
  ungroup() %>% 
  mutate(sample_id = paste(condition, replicate, sep = "\n")) %>% 
  # filter(replicate %in% c("01", "02", "03", "1", "2", "3")) %>% #distinct(replicate) - successfully downsampled to 3 reps 
  distinct(condition, replicate, sample_id, reference, mod_protein_location, mod_res)
##this collapses all  replicates into a single set of unique p sites.
```


#### . . . . write to csv
```{r}
write_csv(x = distinct_p_sites_reps, file = "modified_data/MainFig2/pSTYsites_ascore13_reps.csv", col_names = TRUE)
```

#### . . . . plot
```{r}
plot_distinct_p_sites_reps <- ggplot(data = distinct_p_sites_reps) + #%>% 
                                       # mutate(sample_id = fct_relevel(sample_id,
                                       #                                "NHS 1", "NHS 2", "NHS 3", "Halo 1", "Halo 2", "Halo 3")))+
  geom_bar(mapping = aes(x = sample_id, fill = mod_res)) +
  theme_bw(10) +
  scale_fill_viridis_d() +
  ylab("unique phospho sites")  +
  xlab("1 mg peptide input") +
  alexis_theme()+
  expand_limits(y = c(0, 3000)) +
  scale_y_continuous(expand = c(0,0)) +
  theme(axis.text.x = element_text(angle = -60, vjust = 0.75, hjust = 0))

plot_distinct_p_sites_reps

ggsave("output/MainFig2/distinct_p_sites_reps.png", plot = plot_distinct_p_sites_reps, width = 8, height = 10, scale = 0.4)
ggsave("output/MainFig2/distinct_p_sites_reps.pdf", plot = plot_distinct_p_sites_reps, width = 8, height = 10, scale = 0.4)
```


###__iv.table and error bar df 
```{r}
summarize_distinct_psites_replicates <- distinct_p_sites_reps %>%
  ungroup() %>% 
  group_by(condition, replicate, sample_id, mod_res) %>% 
  summarize(
    n_psites = n()) %>% 
  ungroup()

summarize_distinct_psites_replicates



#errorbar ------------------------------------------------------------
df_geom_errorbar <- summarize_distinct_psites_replicates %>%
  group_by(condition, mod_res) %>%
  mutate(
    min_psites = min(n_psites), 
    max_psites = max(n_psites), 
    mid_psites = mean(n_psites)) %>% 
  ungroup() %>% 
  distinct(condition, mod_res, min_psites, max_psites) #could also include mid psites later if needed

df_geom_errorbar
```
###__iii) condition + replicate as dots


plot
```{r}
plot_distinct_p_sites_condition_reps_wPTS <- ggplot() +
  geom_bar(data = distinct_p_sites_condition,
             # (distinct_p_sites_condition %>%
             #         filter(grepl("pool", condition) == FALSE) %>%
             #         mutate(condition = fct_relevel(condition, "NHS", "Halo"))),
                   mapping = aes(x = condition, fill = mod_res),show.legend = FALSE) +
  
  geom_errorbar(data = (df_geom_errorbar), #%>%
                          # filter(grepl("pool", condition) == FALSE) %>% 
                          # mutate(condition = fct_relevel(condition, "NHS", "Halo"))),
                mapping = aes(x = condition, ymin = min_psites, ymax = max_psites),  show.legend = FALSE, linewidth = 0.5, width = 0.25, color = "grey90") +
  
  
  geom_jitter(data = (distinct_p_sites_reps), #%>%
                        # filter(grepl("pool", condition) == FALSE) %>% 
                        # mutate(condition = fct_relevel(condition, "NHS", "Halo"))),
              mapping = aes(x = condition, color = sample_id, fill = mod_res),
              stat = "count", shape = 1, show.legend = FALSE, size = 1, stroke =0.5,
              width = 0.35, height = 0, alpha = 0.5) +
   
  theme_bw(10) +
  facet_wrap(facets = vars(mod_res), ncol = 3) +
  scale_fill_viridis_d() +
  scale_color_manual(values = rep(c("grey70" ),60)) +
  ylab("unique phospho sites")   +
  alexis_theme()+
  scale_y_continuous(expand = c(0,0))
  # theme(axis.text.x = element_text(angle = 30, vjust = 0.99, hjust = 1, size = 10))

plot_distinct_p_sites_condition_reps_wPTS

ggsave("output/MainFig2/distinct_p_sites_condition_wPTS.png", plot = plot_distinct_p_sites_condition_reps_wPTS, width = 6, height = 6, scale = 0.4)
ggsave("output/MainFig2/distinct_p_sites_condition_wPTS.pdf", plot = plot_distinct_p_sites_condition_reps_wPTS, width = 6, height = 6, scale = 0.4)
```

####___iv) focus on pY
```{r}
set.seed(11)
plot_distinct_p_sites_condition_reps_wPTS <- ggplot() +
  geom_bar(data = (distinct_p_sites_condition %>%
                     
                     # mutate(condition = fct_relevel(condition, "NHS", "Halo"))%>%
                     filter(mod_res == "Y")),
                    
                   mapping = aes(x = condition, fill = mod_res),
           show.legend = FALSE, color = "black", size = 0.5) +
  
   geom_errorbar(data = (df_geom_errorbar %>%
                          
                          # mutate(condition = fct_relevel(condition, "NHS", "Halo"))%>%
                           filter(mod_res == "Y")),
                mapping = aes(x = condition, ymin = min_psites, ymax = max_psites),  show.legend = FALSE, linewidth = 0.5, width = 0.2, color = "grey30") +
  
  
  geom_jitter(data = (distinct_p_sites_reps%>%
                        # mutate(condition = fct_relevel(condition, "NHS", "Halo"))%>%
                        filter(mod_res == "Y")),
              mapping = aes(x = condition, color = sample_id, fill = mod_res),
              stat = "count", shape = 21, show.legend = FALSE, size = 1.5, stroke =0.5, fill = "white",
              width = 0, height = 0, alpha = 0.75) +
  geom_text(data = tibble(x = c(1, 2, 3, 4),y = c(4263, 4364, 4462, 4441)), mapping = aes( x = x, y = y +150, label = y),
            color = "black",
            size = 3,
            fontface = "bold",
            inherit.aes = FALSE) +
 
   
  # theme_bw(10) +
  # facet_wrap(facets = vars(mod_res), ncol = 3) +
  scale_fill_viridis_d(direction = -1) +
  scale_color_manual(values = rep(c("grey30" ),60)) +
  ylab("unique pY sites")   +
  xlab("bead type") +
  # xlab("1 mg peptide input") +
  alexis_theme()+
  scale_y_continuous(expand = c(0,0), limits = c(0, 5000), breaks = c(seq(0, 5000, 1000))) +
  # expand_limits(y = c(0, 3000)) +
  theme(axis.text.x = element_text(angle = -60, vjust = 0.75, hjust = 0, size = 10),
        axis.text.y = element_text( size = 10))

plot_distinct_p_sites_condition_reps_wPTS

ggsave("output/MainFig2/distinct_pY_sites_condition_wPTS.png", plot = plot_distinct_p_sites_condition_reps_wPTS, width = 5.8, height = 8, scale = 0.4)
ggsave("output/MainFig2/distinct_pY_sites_condition_wPTS.pdf", plot = plot_distinct_p_sites_condition_reps_wPTS, width = 5.8, height = 8, scale = 0.4)
```

###### . .. . t-test on pY site counts
```{r}
stats_pYsite_counts <- summarize_distinct_psites_replicates %>% 
  filter(mod_res == "Y") 
  
halfmgmL <- (stats_pYsite_counts %>% filter(grepl("0.5 mg/mL", sample_id) == TRUE))$n_psites

onemgmL <- (stats_pYsite_counts %>% filter(grepl("1 mg/mL", sample_id) == TRUE))$n_psites
twomgmL <- (stats_pYsite_counts %>% filter(grepl("2 mg/mL", sample_id) == TRUE))$n_psites
fourmgmL <- (stats_pYsite_counts %>% filter(grepl("4 mg/mL", sample_id) == TRUE))$n_psites


t.test(x = halfmgmL, y = fourmgmL, var.equal = FALSE)

t.test(x = onemgmL, y = fourmgmL, var.equal = FALSE)

t.test(x = twomgmL, y = fourmgmL, var.equal = FALSE)


```
no statistical difference in pY site counts of replicates between peptide input concentrations.

#==============================



##c) UPSET overlaps
###__i) replicate
```{r}
# need each row to be a protein id and each column to be a cell type or sample.
df_psite_overlaps_reps <- distinct_p_sites_reps %>% 
  mutate(
    ref = paste(reference, mod_protein_location, sep = "_")) %>% 
  distinct(sample_id, ref) %>% 
  mutate(
    psite_present = 1) %>% 
  pivot_wider(
    names_from = sample_id, values_fill = 0, values_from = psite_present, id_cols = ref)

#unselect some columns
df_psite_overlaps_reps_less <- df_psite_overlaps_reps %>% 
  select(-ref) %>% 
  mutate_all(.funs = as.numeric) %>% 
  as.matrix() %>% #trick for formatting fxns within protti.
  as.data.frame()

#upset plot
upset_plot_psites_reps <- upset(data = df_psite_overlaps_reps_less, nsets = 16, order.by = "freq", text.scale = 1.5)
upset_plot_psites_reps

#save plot

png("output/MainFig2/upset_plot_psites_reps.png", width = 800, height = 600)
print(upset_plot_psites_reps)
dev.off()


pdf("output/MainFig2/upset_plot_psites_reps.pdf", width = 12, height = 8)
print(upset_plot_psites_reps)
dev.off()
```

####___- pY only (replicate)
```{r}
# need each row to be a protein id and each column to be a cell type or sample.
df_pYsite_overlaps_reps <- distinct_p_sites_reps %>% 
  filter(mod_res == "Y") %>% 
  mutate(
    ref = paste(reference, mod_protein_location, sep = "_")) %>% 
  distinct(sample_id, ref) %>% 
  mutate(
    psite_present = 1) %>% 
  pivot_wider(
    names_from = sample_id, values_fill = 0, values_from = psite_present, id_cols = ref)

#unselect some columns
df_pYsite_overlaps_reps_less <- df_pYsite_overlaps_reps %>% 
  select(-ref) %>% 
  mutate_all(.funs = as.numeric) %>% 
  as.matrix() %>% #trick for formatting fxns within protti.
  as.data.frame()

#upset plot
upset_plot_pYsites_reps <- upset(data = df_pYsite_overlaps_reps_less, nsets = 16, order.by = "freq", text.scale = 1.5)
upset_plot_pYsites_reps

#save plot

png("output/MainFig2/upset_plot_pYsites_reps.png", width = 800, height = 600)
print(upset_plot_pYsites_reps)
dev.off()


pdf("output/MainFig2/upset_plot_pYsites_reps.pdf", width = 12, height = 8)
print(upset_plot_pYsites_reps)
dev.off()
```

###__ii) condition
```{r}
# need each row to be a protein id and each column to be a cell type or sample.
df_psite_overlaps_condition <- distinct_p_sites_condition %>% 
  mutate(
    ref = paste(reference, mod_protein_location, sep = "_")) %>% 
  distinct(condition, ref) %>% 
  mutate(
    psite_present = 1) %>% 
  pivot_wider(
    names_from = condition, values_fill = 0, values_from = psite_present, id_cols = ref)

#unselect some columns
df_psite_overlaps_condition_less <- df_psite_overlaps_condition %>% 
  select(-ref) %>% 
  mutate_all(.funs = as.numeric) %>% 
  as.matrix() %>% #trick for formatting fxns within protti.
  as.data.frame()

#upset plot
upset_plot_psites_condition <- upset(data = df_psite_overlaps_condition_less, nsets = 16, order.by = "freq", text.scale = 1.5)
upset_plot_psites_condition

#save plot

png("output/MainFig2/upset_plot_psites_condition.png", width = 800, height = 600)
print(upset_plot_psites_condition)
dev.off()


pdf("output/MainFig2/upset_plot_psites_condition.pdf", width = 12, height = 8)
print(upset_plot_psites_condition)
dev.off()
```

####___- pY only (condition)
```{r}
# need each row to be a protein id and each column to be a cell type or sample.
df_pYsite_overlaps_condition <- distinct_p_sites_condition %>% 
  filter(mod_res == "Y") %>% 
  mutate(
    ref = paste(reference, mod_protein_location, sep = "_")) %>% 
  distinct(condition, ref) %>% 
  mutate(
    psite_present = 1) %>% 
  pivot_wider(
    names_from = condition, values_fill = 0, values_from = psite_present, id_cols = ref)

#unselect some columns
df_pYsite_overlaps_condition_less <- df_pYsite_overlaps_condition %>% 
  select(-ref) %>% 
  mutate_all(.funs = as.numeric) %>% 
  as.matrix() %>% #trick for formatting fxns within protti.
  as.data.frame()

#upset plot
upset_plot_pYsites_condition <- upset(data = df_pYsite_overlaps_condition_less, nsets = 16, order.by = "freq", text.scale = 2.5)
upset_plot_pYsites_condition

#save plot

png("output/MainFig2/upset_plot_pYsites_condition.png", width = 700, height = 500)
print(upset_plot_pYsites_condition)
dev.off()


pdf("output/MainFig2/upset_plot_pYsites_condition.pdf", width = 12, height = 8)
print(upset_plot_pYsites_condition)
dev.off()
```



#4. PROTTI
##_a) CVs
###__i) all pSTY
####___- not normalized
```{r}
psite_reps_CVs_NOTnorm_pSTY <- my_qc_cvs(
  data = ascore_stringent_fasta_pSTYsite, 
  grouping = ref, 
  condition = condition,
  intensity = sum_intensity_precursor_to_psite,
  plot = TRUE, 
  plot_style = "boxplot",
  xlab = expression(paste("Bead slurry volume (", mu, "L", ")")),
  max_cv = 100)

psite_reps_CVs_NOTnorm_pSTY

ggsave(filename = "output/MainFig2/CVs_not_normalized_pSTY.png", plot = psite_reps_CVs_NOTnorm_pSTY, width = 10, height = 10, scale = 0.4)
ggsave(filename = "output/MainFig2/CVs_not_normalized_pSTY.pdf", plot = psite_reps_CVs_NOTnorm_pSTY, width = 10, height = 10, scale = 0.4)
```


####___- normalized
```{r}
# stats ----------------------------------------
psite_reps_CVs_YESnorm_stats_pSTY <- qc_cvs(
  data = ascore_stringent_fasta_pSTYsite, 
  grouping = ref, 
  condition = condition,
  intensity = raw_median_norm_intensity,
  plot = FALSE, 
  plot_style = "boxplot",
  max_cv = 100)

psite_reps_CVs_YESnorm_stats_pSTY





#plot ------------------------------------------
psite_reps_CVs_YESnorm_pSTY <- my_qc_cvs(
  data = ascore_stringent_fasta_pSTYsite, 
  grouping = ref, 
  condition = condition,
  intensity = raw_median_norm_intensity,
  plot = TRUE, 
  xlab = expression(paste("Bead slurry volume (", mu, "L", ")")),
  plot_style = "boxplot",
  max_cv = 100)

psite_reps_CVs_YESnorm_pSTY

ggsave(filename = "output/MainFig2/CVs_YES_normalized_pSTY.png", plot = psite_reps_CVs_YESnorm_pSTY, width = 10, height = 10, scale = 0.4)
ggsave(filename = "output/MainFig2/CVs_YES_normalized_pSTY.pdf", plot = psite_reps_CVs_YESnorm_pSTY, width = 10, height = 10, scale = 0.4)
```
Only the 20 uL bead condition becomes lower CV distribution upon median normalization.


###__ii) pY
####___- not normalized
```{r}
psite_reps_CVs_NOTnorm_pY <- my_qc_cvs(
  data = ascore_stringent_fasta_pYsite, 
  grouping = ref, 
  condition = condition,
  intensity = sum_intensity_precursor_to_psite,
  plot = TRUE, 
  plot_style = "boxplot",
  xlab = expression(paste("Bead slurry volume (", mu, "L", ")")),
  max_cv = 100)

psite_reps_CVs_NOTnorm_pY

ggsave(filename = "output/MainFig2/CVs_not_normalized_pY.png", plot = psite_reps_CVs_NOTnorm_pY, width = 10, height = 10, scale = 0.4)
ggsave(filename = "output/MainFig2/CVs_not_normalized_pY.pdf", plot = psite_reps_CVs_NOTnorm_pY, width = 10, height = 10, scale = 0.4)
```


####___- normalized
```{r}
psite_reps_CVs_YESnorm_pY <- my_qc_cvs(
  data = ascore_stringent_fasta_pYsite, 
  grouping = ref, 
  condition = condition,
  intensity = raw_median_norm_intensity,
  plot = TRUE, 
  xlab = "peptide concentration",
  # xlab = expression(paste("Bead slurry volume (", mu, "L", ")")),
  plot_style = "boxplot",
  max_cv = 100)

psite_reps_CVs_YESnorm_pY

ggsave(filename = "output/MainFig2/CVs_YES_normalized_pY.png", plot = psite_reps_CVs_YESnorm_pY, width = 5.5, height = 8, scale = 0.38)
ggsave(filename = "output/MainFig2/CVs_YES_normalized_pY.pdf", plot = psite_reps_CVs_YESnorm_pY, width = 5.5, height = 8, scale = 0.38)
```



##_b) hierarchical clustering
```{r}
pYsite_correlation <- qc_sample_correlation(
  data = ascore_stringent_fasta_pYsite,
  sample = sample_id, 
  grouping = ref, 
  intensity_log2 = median_norm_intensity,
  condition = condition,
  # method = "pearson",
  interactive =  FALSE)

pYsite_correlation


ggsave(filename = "output/MainFig2/pYsite_correlation.png", plot = pYsite_correlation, width = 14, height = 12, scale = 0.4)
ggsave(filename = "output/MainFig2/pYsite_correlation.pdf", plot = pYsite_correlation, width = 14, height = 12, scale = 0.4)
```

```{r}
my_pYsite_correlation <- my_hier_cluster(
  data = ascore_stringent_fasta_pYsite,
  sample = sample_id, 
  grouping = ref, 
  intensity_log2 = median_norm_intensity,
  condition = condition,
  # method = "pearson",
  fontsize_number = 6, 
  number_color = "white",
  # display_numbers = TRUE,
  interactive =  FALSE)

my_pYsite_correlation


ggsave(filename = "output/MainFig2/my_pYsite_correlation.png", plot = my_pYsite_correlation, width = 14, height = 12, scale = 0.4)
ggsave(filename = "output/MainFig2/my_pYsite_correlation.pdf", plot = my_pYsite_correlation, width = 14, height = 12, scale = 0.4)
```

```{r}
# pYsite_correlation_interactive <- qc_sample_correlation(
#   data = ascore_stringent_fasta_pYsite,
#   sample = sample_id, 
#   grouping = ref, 
#   intensity_log2 = median_norm_intensity,
#   condition = condition,
#   # method = "pearson",
#   interactive =  TRUE)
# 
# pYsite_correlation_interactive
# 
# # install.packages("heatmaply")
# # library(heatmaply)

```


